// tdesign-vue-next/cjs/radio/radio.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$5 = require('../_chunks/dep-1862d466.js');
var index$3 = require('../_chunks/dep-38acad3d.js');
var index$4 = require('../_chunks/dep-0238ed9a.js');
var index$1 = require('../_chunks/dep-362c5212.js');
require('../_chunks/dep-0989c3be.js');
var index$2 = require('../_chunks/dep-9dbade68.js');
var index = require('../_chunks/dep-ce9c15f5.js');
var radio_props = require('./props.js');
var radio_constants_index = require('./constants/index.js');
var helper = require('../_chunks/dep-2aa3d3cf.js');
var omit = require('../_chunks/dep-5c79e9ef.js');
var isString = require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-74b143e0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

var _Radio = Vue.defineComponent({
  name: "TRadio",
  inheritAttrs: false,
  props: radio_props["default"],
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var inputRef = Vue.ref();
    var _toRefs = Vue.toRefs(props2),
      checked = _toRefs.checked,
      modelValue = _toRefs.modelValue;
    var _useVModel = index.useVModel(checked, modelValue, props2.defaultChecked, props2.onChange, "checked"),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      innerChecked = _useVModel2[0],
      setInnerChecked = _useVModel2[1];
    var radioChecked = Vue.computed(function () {
      return radioGroup ? props2.value === radioGroup.value : innerChecked.value;
    });
    var radioGroup = Vue.inject(radio_constants_index.RadioGroupInjectionKey, void 0);
    var allowUncheck = Vue.computed(function () {
      return Boolean(props2.allowUncheck || (radioGroup === null || radioGroup === void 0 ? void 0 : radioGroup.allowUncheck));
    });
    var handleClick = function handleClick(e) {
      e.stopPropagation();
    };
    var onLabelClick = function onLabelClick(e) {
      var _props2$onClick;
      if (isDisabled.value || isReadonly.value) return;
      (_props2$onClick = props2.onClick) === null || _props2$onClick === void 0 || _props2$onClick.call(props2, {
        e: e
      });
      if (radioChecked.value && !allowUncheck.value) return;
      if (radioGroup) {
        var value = radioChecked.value && allowUncheck.value ? void 0 : props2.value;
        radioGroup.setValue(value, {
          e: e,
          name: radioGroup.name
        });
      } else {
        var _value = allowUncheck.value ? !radioChecked.value : true;
        setInnerChecked(_value, {
          e: e
        });
      }
    };
    var inputEvents = Vue.computed(function () {
      return helper.getValidAttrs({
        focus: attrs.onFocus,
        blur: attrs.onBlur,
        keydown: attrs.onKeydown,
        keyup: attrs.onKeyup,
        keypresss: attrs.onKeypresss
      });
    });
    var wrapperAttrs = Vue.computed(function () {
      var events = [].concat(_toConsumableArray__default["default"](Object.keys(inputEvents.value)), ["input", "change"]).map(function (str) {
        return "on".concat(str[0].toUpperCase()).concat(str.slice(1));
      });
      return omit.omit(attrs, events);
    });
    var groupDisabled = Vue.computed(function () {
      return radioGroup === null || radioGroup === void 0 ? void 0 : radioGroup.disabled;
    });
    var groupReadonly = Vue.computed(function () {
      return radioGroup === null || radioGroup === void 0 ? void 0 : radioGroup.readonly;
    });
    var isDisabled = index$1.useDisabled({
      afterDisabled: groupDisabled
    });
    var isReadonly = index$2.useReadonly({
      afterReadonly: groupReadonly
    });
    var inputProps = Vue.computed(function () {
      return {
        name: radioGroup ? radioGroup.name : props2.name,
        checked: radioChecked.value,
        disabled: isDisabled.value,
        readonly: isReadonly.value,
        value: props2.value
      };
    });
    var _useCommonClassName = index$3.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var radioButton = Vue.inject(radio_constants_index.RadioButtonInjectionKey, void 0);
    var radioBtnName = index$4.usePrefixClass("radio-button");
    var COMPONENT_NAME = index$4.usePrefixClass("radio");
    var prefixCls = Vue.computed(function () {
      return radioButton ? radioBtnName.value : COMPONENT_NAME.value;
    });
    var inputClass = Vue.computed(function () {
      return ["".concat(prefixCls.value), _defineProperty__default["default"](_defineProperty__default["default"]({}, STATUS.value.checked, inputProps.value.checked), STATUS.value.disabled, inputProps.value.disabled)];
    });
    var renderContent = index$5.useContent();
    return function () {
      return Vue.createVNode("label", Vue.mergeProps({
        "ref": inputRef,
        "class": inputClass.value
      }, wrapperAttrs.value, {
        "tabindex": isDisabled.value ? void 0 : "0",
        "onClick": onLabelClick
      }), [Vue.createVNode("input", Vue.mergeProps({
        "type": "radio",
        "class": "".concat(prefixCls.value, "__former")
      }, inputEvents.value, inputProps.value, {
        "onClick": handleClick,
        "tabindex": "-1",
        "data-value": isString.isString(props2.value) ? "'".concat(props2.value, "'") : props2.value,
        "data-allow-uncheck": allowUncheck.value || void 0
      }), null), Vue.createVNode("span", {
        "class": "".concat(prefixCls.value, "__input")
      }, null), Vue.createVNode("span", {
        "class": "".concat(prefixCls.value, "__label")
      }, [renderContent("default", "label")])]);
    };
  }
});

exports["default"] = _Radio;
