// tdesign-vue-next/cjs/date-picker/components/panel/RangePanel.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-35e23c2b.js');
var index = require('../../../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
var log = require('../../../_chunks/dep-692bdebe.js');
require('../../../_chunks/dep-0989c3be.js');
var datePicker_components_panel_PanelContent = require('./PanelContent.js');
var datePicker_components_panel_ExtraContent = require('./ExtraContent.js');
var format = require('../../../_chunks/dep-4d6191e3.js');
var datePicker_hooks_useDisableDate = require('../../hooks/useDisableDate.js');
require('tdesign-icons-vue-next');
require('dayjs');
var datePicker_hooks_useTableData = require('../../hooks/useTableData.js');
require('../../../_chunks/dep-1035a9b6.js');
var isArray = require('../../../_chunks/dep-4cb26289.js');
var isFunction = require('../../../_chunks/dep-2dcf9237.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
require('../../../_chunks/dep-4a7162e7.js');
require('../../../_chunks/dep-be1af85d.js');
require('../../../_chunks/dep-db9c85b1.js');
require('../../../_chunks/dep-82fe2026.js');
require('../../../_chunks/dep-33b46a52.js');
require('../../../_chunks/dep-930a2de5.js');
require('../../../_chunks/dep-22dc294c.js');
require('../../../_chunks/dep-bf76dead.js');
require('../../../_chunks/dep-c3bbd06c.js');
require('../../../_chunks/dep-6d87f74d.js');
require('../../../_chunks/dep-281f7eb2.js');
require('../../../_chunks/dep-f8f7b1af.js');
require('../../../_chunks/dep-66d3f30f.js');
require('../../../_chunks/dep-e1d22111.js');
require('../../../_chunks/dep-754c0523.js');
require('../../../_chunks/dep-7076a08a.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../base/Header.js');
require('../../../pagination/index.js');
require('../../../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-1862d466.js');
require('../../../_chunks/dep-8d990e66.js');
require('../../../_chunks/dep-631750cd.js');
require('../../../_chunks/dep-a6551cc4.js');
require('../../../_chunks/dep-7d9abefd.js');
require('../../../_chunks/dep-2fb8d44f.js');
require('../../../_chunks/dep-80a41429.js');
require('../../../_chunks/dep-91305690.js');
require('../../../_chunks/dep-0b92e695.js');
require('../../../_chunks/dep-b2024592.js');
require('../../../_chunks/dep-9f98bf9b.js');
require('../../../_chunks/dep-0afce170.js');
require('../../../_chunks/dep-c26a7d50.js');
require('../../../_chunks/dep-ce9c15f5.js');
require('../../../input-number/index.js');
require('../../../input-number/input-number.js');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-34cd76d4.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-d632b788.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-4ba41627.js');
require('../../../_chunks/dep-2aa3d3cf.js');
require('../../../_chunks/dep-3d29e358.js');
require('../../../_chunks/dep-6212811d.js');
require('../../../_chunks/dep-44195fb2.js');
require('../../../_chunks/dep-38acad3d.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-2fc884aa.js');
require('../../../_chunks/dep-75d6da05.js');
require('../../../_chunks/dep-274369eb.js');
require('../../../_chunks/dep-df6e14a0.js');
require('../../../_chunks/dep-43dcd2db.js');
require('../../../_chunks/dep-392dde71.js');
require('../../../_chunks/dep-02ea4f3c.js');
require('../../../_chunks/dep-5103206f.js');
require('../../../_chunks/dep-7707ffe2.js');
require('../../../_chunks/dep-74a5bebe.js');
require('../../../_chunks/dep-4eca9add.js');
require('../../../_chunks/dep-42740075.js');
require('../../../_chunks/dep-9ac5c6ca.js');
require('../../../_chunks/dep-e3edff33.js');
require('../../../button/props.js');
require('../../../_chunks/dep-362c5212.js');
require('../../../_chunks/dep-74b143e0.js');
require('../../../_chunks/dep-719a33c8.js');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-9dbade68.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-e2122882.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../input-number/props.js');
require('../../../_chunks/dep-a13f0135.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-6bac596c.js');
require('../../../_chunks/dep-73734701.js');
require('../../../select/index.js');
require('../../../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../../common-components/fake-arrow.js');
require('../../../select-input/index.js');
require('../../../select-input/select-input.js');
require('../../../popup/index.js');
require('../../../popup/popup.js');
require('@popperjs/core');
require('../../../popup/container.js');
require('../../../popup/props.js');
require('../../../_chunks/dep-9bee89d3.js');
require('../../../_chunks/dep-027cd76f.js');
require('../../../_chunks/dep-ae4040c5.js');
require('../../../_chunks/dep-79476fcb.js');
require('../../../select-input/props.js');
require('../../../select-input/hooks/useMultiple.js');
require('../../../tag-input/index.js');
require('../../../tag-input/tag-input.js');
require('../../../tag-input/props.js');
require('../../../tag-input/hooks/useDragSorter.js');
require('../../../tag-input/hooks/useHover.js');
require('../../../tag-input/hooks/useTagScroll.js');
require('../../../tag-input/hooks/useTagList.js');
require('../../../tag/index.js');
require('../../../tag/tag.js');
require('tinycolor2');
require('../../../tag/props.js');
require('../../../tag/check-tag.js');
require('../../../tag/check-tag-props.js');
require('../../../_chunks/dep-a4be1915.js');
require('../../../tag/check-tag-group.js');
require('../../../tag/check-tag-group-props.js');
require('../../../select-input/hooks/useOverlayInnerStyle.js');
require('../../../select-input/hooks/useSingle.js');
require('../../../_chunks/dep-eb3114d9.js');
require('../../../_chunks/dep-690970a4.js');
require('../../../_chunks/dep-7870f4cb.js');
require('../../../select/components/select-panel.js');
require('../../../select/option.js');
require('../../../select/option-props.js');
require('../../../checkbox/index.js');
require('../../../checkbox/checkbox.js');
require('../../../checkbox/props.js');
require('../../../checkbox/constants/index.js');
require('../../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../../_chunks/dep-b9a88912.js');
require('../../../checkbox/hooks/useKeyboardEvent.js');
require('../../../checkbox/group.js');
require('../../../checkbox/checkbox-group-props.js');
require('../../../_chunks/dep-0cda9fe9.js');
require('../../../_chunks/dep-b0334c78.js');
require('../../../_chunks/dep-a214821d.js');
require('../../../select/utils/index.js');
require('../../../select/constants/index.js');
require('../../../select/option-group.js');
require('../../../select/option-group-props.js');
require('../../../select/props.js');
require('../../../select/hooks/usePanelVirtualScroll.js');
require('../../../_chunks/dep-5c79e9ef.js');
require('../../../_chunks/dep-9a54a293.js');
require('../../../select/hooks/useKeyboardControl.js');
require('../../../select/hooks/useSelectOptions.js');
require('../../../_chunks/dep-1c1b8615.js');
require('../../../input-adornment/index.js');
require('../../../input-adornment/input-adornment.js');
require('../../../input-adornment/props.js');
require('../../../pagination/props.js');
require('../../../pagination/hooks/useMoreAction.js');
require('../../../pagination/hooks/usePaginationClasses.js');
require('../../../pagination/pagination-mini.js');
require('../../../pagination/pagination-mini-props.js');
require('../../../tooltip/index.js');
require('../../../tooltip/tooltip.js');
require('../../../tooltip/props.js');
require('../../../tooltip/utils/index.js');
require('../../hooks/useSelectRange.js');
require('../base/Table.js');
require('../base/Cell.js');
require('../../../time-picker/index.js');
require('../../../time-picker/time-picker.js');
require('../../../_chunks/dep-484d5d62.js');
require('../../../time-picker/panel/time-picker-panel.js');
require('../../../_chunks/dep-5241cd36.js');
require('../../../time-picker/props.js');
require('../../../_chunks/dep-f198a3a2.js');
require('../../../_chunks/dep-58022b88.js');
require('../../../time-picker/time-range-picker.js');
require('../../../range-input/index.js');
require('../../../range-input/range-input.js');
require('../../../range-input/props.js');
require('../../../range-input/range-input-popup.js');
require('../../../range-input/range-input-popup-props.js');
require('../../../time-picker/time-range-picker-props.js');
require('../../utils/index.js');
require('../../../_chunks/dep-95977a93.js');
require('../base/Footer.js');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TRangePanel = Vue.defineComponent({
  name: "TRangePanel",
  props: {
    hoverValue: Array,
    range: [Array, Function],
    activeIndex: Number,
    isFirstValueSelected: Boolean,
    disabled: {
      type: [Boolean, Array]
    },
    disableDate: [Object, Array, Function],
    disableTime: Function,
    mode: {
      type: String,
      "default": "date"
    },
    format: String,
    presetsPlacement: {
      type: String,
      "default": "bottom"
    },
    value: Array,
    timePickerProps: Object,
    presets: Object,
    popupVisible: Boolean,
    enableTimePicker: Boolean,
    panelPreselection: Boolean,
    firstDayOfWeek: Number,
    year: Array,
    month: Array,
    time: Array,
    cancelRangeSelectLimit: Boolean,
    defaultTime: Array,
    onClick: Function,
    onCellClick: Function,
    onCellMouseEnter: Function,
    onCellMouseLeave: Function,
    onJumperClick: Function,
    onConfirmClick: Function,
    onPresetClick: Function,
    onYearChange: Function,
    onMonthChange: Function,
    onTimePickerChange: Function,
    needConfirm: Boolean,
    cell: {
      type: Function
    }
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var COMPONENT_NAME = index.usePrefixClass("date-range-picker__panel");
    var _useConfig = configProvider_hooks_useConfig.useConfig("datePicker"),
      globalConfig = _useConfig.globalConfig;
    var format$1 = Vue.computed(function () {
      var _getDefaultFormat;
      return (_getDefaultFormat = format.getDefaultFormat({
        mode: props.mode,
        format: props.format,
        enableTimePicker: props.enableTimePicker
      })) === null || _getDefaultFormat === void 0 ? void 0 : _getDefaultFormat.format;
    });
    var hidePreselection = !props.panelPreselection && props.value.length === 2;
    var disableDateOptions = Vue.computed(function () {
      var startDateValue = new Date(format.parseToDayjs(props.value[0], format$1.value, "start").toDate().setHours(0, 0, 0));
      var endDateValue = new Date(format.parseToDayjs(props.value[1], format$1.value, "end").toDate().setHours(23, 59, 59));
      var start = props.isFirstValueSelected && props.activeIndex === 1 ? startDateValue : void 0;
      var end = props.isFirstValueSelected && props.activeIndex === 0 ? endDateValue : void 0;
      if (props.disabled && isArray.isArray(props.disabled)) {
        if (props.disabled[0]) start = startDateValue;else if (props.disabled[1]) end = endDateValue;
      }
      return datePicker_hooks_useDisableDate.useDisableDate({
        format: format$1.value,
        mode: props.mode,
        disableDate: props.disableDate,
        start: start,
        end: end
      });
    });
    var rangeValue = Vue.computed(function () {
      var startRange = props.range;
      var endRange = props.range;
      if (isArray.isArray(props.range)) {
        if (props.range.length !== 2) {
          log.log.warn("DateRangePicker", "`range` length must be 2 when `range` is an array.");
        }
        var first = props.range[0];
        var second = props.range[1];
        if ((isArray.isArray(first) || isFunction.isFunction(first)) && (isArray.isArray(second) || isFunction.isFunction(second))) {
          startRange = first;
          endRange = second;
        }
      }
      return {
        start: startRange,
        end: endRange
      };
    });
    var startTableData = Vue.computed(function () {
      var disableDate = isFunction.isFunction(props.disableDate) ? props.disableDate({
        partial: "start",
        value: props.value[0]
      }) : disableDateOptions.value.disableDate;
      return datePicker_hooks_useTableData.useTableData(_objectSpread(_objectSpread({
        isRange: true,
        start: props.value[0] ? format.parseToDayjs(props.value[0], format$1.value).toDate() : void 0,
        end: props.value[1] ? format.parseToDayjs(props.value[1], format$1.value).toDate() : void 0,
        hoverStart: !hidePreselection && props.hoverValue[0] ? format.parseToDayjs(props.hoverValue[0], format$1.value).toDate() : void 0,
        hoverEnd: !hidePreselection && props.hoverValue[1] ? format.parseToDayjs(props.hoverValue[1], format$1.value).toDate() : void 0,
        year: props.year[0],
        month: props.month[0],
        mode: props.mode,
        firstDayOfWeek: props.firstDayOfWeek || globalConfig.value.firstDayOfWeek,
        range: rangeValue.value.start
      }, disableDateOptions.value), {}, {
        disableDate: disableDate,
        cancelRangeSelectLimit: props.cancelRangeSelectLimit
      }));
    });
    var endTableData = Vue.computed(function () {
      var disableDate = isFunction.isFunction(props.disableDate) ? props.disableDate({
        partial: "end",
        value: props.value
      }) : disableDateOptions.value.disableDate;
      return datePicker_hooks_useTableData.useTableData(_objectSpread(_objectSpread({
        isRange: true,
        start: props.value[0] ? format.parseToDayjs(props.value[0], format$1.value).toDate() : void 0,
        end: props.value[1] ? format.parseToDayjs(props.value[1], format$1.value).toDate() : void 0,
        hoverStart: !hidePreselection && props.hoverValue[0] ? format.parseToDayjs(props.hoverValue[0], format$1.value).toDate() : void 0,
        hoverEnd: !hidePreselection && props.hoverValue[1] ? format.parseToDayjs(props.hoverValue[1], format$1.value).toDate() : void 0,
        year: props.mode === "year" && props.year[1] - props.year[0] <= 9 ? props.year[1] + 9 : props.year[1],
        month: props.month[1],
        mode: props.mode,
        firstDayOfWeek: props.firstDayOfWeek || globalConfig.value.firstDayOfWeek,
        range: rangeValue.value.end
      }, disableDateOptions.value), {}, {
        disableDate: disableDate,
        cancelRangeSelectLimit: props.cancelRangeSelectLimit
      }));
    });
    var panelContentProps = Vue.computed(function () {
      return {
        format: format$1.value,
        mode: props.mode,
        firstDayOfWeek: props.firstDayOfWeek || globalConfig.value.firstDayOfWeek,
        internalYear: props.year,
        popupVisible: props.popupVisible,
        enableTimePicker: props.enableTimePicker,
        timePickerProps: props.timePickerProps,
        onMonthChange: props.onMonthChange,
        onYearChange: props.onYearChange,
        onJumperClick: props.onJumperClick,
        onCellClick: props.onCellClick,
        onCellMouseEnter: props.onCellMouseEnter,
        onCellMouseLeave: props.onCellMouseLeave,
        onTimePickerChange: props.onTimePickerChange,
        disableTime: props.disableTime,
        defaultTime: props.defaultTime,
        cell: props.cell
      };
    });
    return function () {
      return Vue.createVNode("div", {
        "class": [COMPONENT_NAME.value, _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--direction-row"), ["left", "right"].includes(props.presetsPlacement))],
        "onClick": function onClick(e) {
          var _props$onClick;
          return (_props$onClick = props.onClick) === null || _props$onClick === void 0 ? void 0 : _props$onClick.call(props, {
            e: e
          });
        }
      }, [["top", "left"].includes(props.presetsPlacement) ? Vue.createVNode(datePicker_components_panel_ExtraContent["default"], {
        "presets": props.presets,
        "selectedValue": props.value[props.activeIndex],
        "enableTimePicker": props.enableTimePicker,
        "onPresetClick": props.onPresetClick,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null, Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-content-wrapper")
      }, [!props.enableTimePicker ? [Vue.createVNode(datePicker_components_panel_PanelContent["default"], Vue.mergeProps({
        "key": "startPanel",
        "partial": "start",
        "year": props.year[0],
        "month": props.month[0],
        "time": props.time[props.activeIndex],
        "value": props.value,
        "tableData": startTableData.value,
        "range": rangeValue.value.start
      }, panelContentProps.value), slots), Vue.createVNode(datePicker_components_panel_PanelContent["default"], Vue.mergeProps({
        "key": "endPanel",
        "partial": "end",
        "year": props.year[1],
        "month": props.month[1],
        "time": props.time[props.activeIndex],
        "value": props.value,
        "tableData": endTableData.value,
        "range": rangeValue.value.end
      }, panelContentProps.value), slots)] : Vue.createVNode(datePicker_components_panel_PanelContent["default"], Vue.mergeProps({
        "key": "start",
        "partial": props.activeIndex ? "end" : "start",
        "year": props.activeIndex ? props.year[1] : props.year[0],
        "month": props.activeIndex ? props.month[1] : props.month[0],
        "time": props.activeIndex ? props.time[1] : props.time[0],
        "value": props.value,
        "tableData": props.activeIndex ? endTableData.value : startTableData.value,
        "range": props.activeIndex ? rangeValue.value.end : rangeValue.value.start
      }, panelContentProps.value), null)]), ["bottom", "right"].includes(props.presetsPlacement) ? Vue.createVNode(datePicker_components_panel_ExtraContent["default"], {
        "presets": props.presets,
        "selectedValue": props.value[props.activeIndex],
        "enableTimePicker": props.enableTimePicker,
        "onPresetClick": props.onPresetClick,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null]);
    };
  }
});

exports["default"] = TRangePanel;
