// tdesign-vue-next/es/badge/badge.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode, mergeProps } from 'vue';
import { _ as _defineProperty } from '../_chunks/dep-ba214d75.mjs';
import { _ as _slicedToArray } from '../_chunks/dep-f558ebb8.mjs';
import props from './props.mjs';
import { b as useContent, a as useTNodeJSX, u as usePrefixClass } from '../_chunks/dep-fbeb2963.mjs';
import '../_chunks/dep-e1c62679.mjs';
import '../_chunks/dep-b0b422fe.mjs';
import 'lodash-es';
import '../_chunks/dep-47c3cae9.mjs';
import '../_chunks/dep-feed89df.mjs';
import '../_chunks/dep-5b510fcd.mjs';
import '../config-provider/hooks/useConfig.mjs';
import '../config-provider/utils/context.mjs';
import '../_chunks/dep-e0ab6d55.mjs';
import '../_chunks/dep-75f265c4.mjs';
import '../_chunks/dep-67ffcfeb.mjs';
import '../_chunks/dep-edab0e6a.mjs';
import '../_chunks/dep-76dddedf.mjs';
import '../_chunks/dep-d92220c4.mjs';
import '../_chunks/dep-466a8f31.mjs';
import '../_chunks/dep-e66c0337.mjs';

var _Badge = defineComponent({
  name: "TBadge",
  inheritAttrs: false,
  props: props,
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var renderContent = useContent();
    var renderTNodeJSX = useTNodeJSX();
    var displayCount = computed(function () {
      var count = renderTNodeJSX("count");
      if (Number.isNaN(Number(count))) {
        return count;
      }
      count = Number(props2.count);
      return count > props2.maxCount ? "".concat(props2.maxCount, "+") : count;
    });
    var getOffset = function getOffset() {
      if (!props2.offset) return {};
      var _props2$offset = _slicedToArray(props2.offset, 2),
        xOffset = _props2$offset[0],
        yOffset = _props2$offset[1];
      xOffset = Number.isNaN(Number(xOffset)) ? xOffset : "".concat(xOffset, "px");
      yOffset = Number.isNaN(Number(yOffset)) ? yOffset : "".concat(yOffset, "px");
      return {
        xOffset: xOffset,
        yOffset: yOffset
      };
    };
    var classPrefix = usePrefixClass();
    var COMPONENT_NAME = usePrefixClass("badge");
    var isHidden = computed(function () {
      return !props2.showZero && (displayCount.value === 0 || displayCount.value === "0");
    });
    var badgeClassNames = computed(function () {
      return [_defineProperty(_defineProperty(_defineProperty(_defineProperty({}, "".concat(COMPONENT_NAME.value, "--dot"), !!props2.dot), "".concat(COMPONENT_NAME.value, "--circle"), !props2.dot && props2.shape === "circle"), "".concat(COMPONENT_NAME.value, "--round"), !props2.dot && props2.shape === "round"), "".concat(classPrefix.value, "-size-s"), props2.size === "small")];
    });
    var inlineStyle = computed(function () {
      var _getOffset = getOffset(),
        xOffset = _getOffset.xOffset,
        yOffset = _getOffset.yOffset;
      return {
        background: props2.color,
        right: xOffset,
        top: yOffset
      };
    });
    return function () {
      return createVNode("div", mergeProps({
        "class": COMPONENT_NAME.value
      }, attrs), [renderContent("default", "content"), isHidden.value ? null : createVNode("sup", {
        "class": badgeClassNames.value,
        "style": inlineStyle.value
      }, [props2.dot ? null : displayCount.value])]);
    };
  }
});

export { _Badge as default };
