// tdesign-vue-next/cjs/menu/menu-item.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var menu_menuItemProps = require('./menu-item-props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$1 = require('../_chunks/dep-1862d466.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
var index$2 = require('../_chunks/dep-719a33c8.js');
var tooltip_index = require('../tooltip/index.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-4ba41627.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-38acad3d.js');
require('../_chunks/dep-ce9c15f5.js');
require('../popup/container.js');
require('../_chunks/dep-e2122882.js');
require('../_chunks/dep-9bee89d3.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-79476fcb.js');
require('../_chunks/dep-e3edff33.js');
require('../tooltip/utils/index.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var _MenuItem = Vue.defineComponent({
  name: "TMenuItem",
  props: menu_menuItemProps["default"],
  emits: ["click"],
  setup: function setup(props2, ctx) {
    var _toRefs = Vue.toRefs(props2),
      href = _toRefs.href,
      target = _toRefs.target,
      to = _toRefs.to,
      disabled = _toRefs.disabled,
      value = _toRefs.value,
      onClick = _toRefs.onClick,
      routerLink = _toRefs.routerLink,
      replace = _toRefs.replace;
    var classPrefix = index.usePrefixClass();
    var renderTNodeJSX = index$1.useTNodeJSX();
    var renderContent = index$1.useContent();
    var menu = Vue.inject("TdMenu");
    var itemRef = Vue.ref();
    var instance = Vue.getCurrentInstance();
    index$2.useRipple(itemRef);
    var submenu = Vue.inject("TdSubmenu", null);
    var active = Vue.computed(function () {
      return menu.activeValue.value === props2.value;
    });
    var collapsed = Vue.computed(function () {
      var _menu$collapsed;
      return (_menu$collapsed = menu.collapsed) === null || _menu$collapsed === void 0 ? void 0 : _menu$collapsed.value;
    });
    var classes = Vue.computed(function () {
      return ["".concat(classPrefix.value, "-menu__item"), _defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(classPrefix.value, "-is-active"), active.value), "".concat(classPrefix.value, "-is-disabled"), props2.disabled), "".concat(classPrefix.value, "-menu__item--plain"), !ctx.slots.icon && !props2.icon), "".concat(classPrefix.value, "-submenu__item"), !!submenu && !menu.isHead)];
    });
    var router = Vue.computed(function () {
      return props2.router || (instance === null || instance === void 0 ? void 0 : instance.proxy.$router);
    });
    var handleClick = function handleClick(e) {
      var _onClick$value, _submenu$closeParentP;
      e.stopPropagation();
      if (disabled.value) return;
      menu.select(value.value);
      (_onClick$value = onClick.value) === null || _onClick$value === void 0 || _onClick$value.call(onClick, {
        e: e,
        value: value.value
      });
      if (to.value || routerLink.value && href.value) {
        var methods = replace.value ? "replace" : "push";
        router.value[methods](to.value || href.value)["catch"](function (err) {
          if (err.name !== "NavigationDuplicated" && !err.message.includes("Avoided redundant navigation to current location")) {
            throw err;
          }
        });
      }
      submenu === null || submenu === void 0 || (_submenu$closeParentP = submenu.closeParentPopup) === null || _submenu$closeParentP === void 0 || _submenu$closeParentP.call(submenu, e);
    };
    Vue.onMounted(function () {
      var _menu$vMenu;
      menu === null || menu === void 0 || (_menu$vMenu = menu.vMenu) === null || _menu$vMenu === void 0 || _menu$vMenu.add(_objectSpread({
        value: props2.value,
        parent: submenu === null || submenu === void 0 ? void 0 : submenu.value,
        vnode: ctx.slots["default"]
      }, props2));
      var activeValues = menu.activeValues.value;
      if (activeValues.includes(props2.value) && !activeValues.includes(submenu === null || submenu === void 0 ? void 0 : submenu.value)) {
        activeValues.push(submenu === null || submenu === void 0 ? void 0 : submenu.value);
      }
    });
    return function () {
      var _router$value;
      var liContent = Vue.createVNode("li", {
        "ref": itemRef,
        "class": classes.value,
        "onClick": handleClick
      }, [renderTNodeJSX("icon"), routerLink.value ? Vue.createVNode("a", {
        "href": href.value ? href.value : to.value ? (_router$value = router.value) === null || _router$value === void 0 ? void 0 : _router$value.resolve(to.value).href : "",
        "target": target.value,
        "class": "".concat(classPrefix.value, "-menu__item-link"),
        "onClick": function onClick(e) {
          return e.preventDefault();
        }
      }, [Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-menu__content")
      }, [renderContent("default", "content")])]) : href.value ? Vue.createVNode("a", {
        "href": href.value,
        "target": target.value,
        "class": "".concat(classPrefix.value, "-menu__item-link"),
        "onClick": function onClick(e) {
          return disabled.value && e.preventDefault();
        }
      }, [Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-menu__content")
      }, [renderContent("default", "content")])]) : Vue.createVNode("span", {
        "class": "".concat(classPrefix.value, "-menu__content")
      }, [renderContent("default", "content")])]);
      if (collapsed.value && !submenu) {
        return Vue.createVNode(tooltip_index.Tooltip, {
          "content": function content() {
            return renderContent("default", "content");
          },
          "placement": "right"
        }, _isSlot(liContent) ? liContent : {
          "default": function _default() {
            return [liContent];
          }
        });
      }
      return liContent;
    };
  }
});

exports["default"] = _MenuItem;
