// tdesign-vue-next/esm/cascader/cascader-panel.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, createVNode } from 'vue';
import TCascaderSubPanel from './components/Panel.js';
import props from './props.js';
import { useCascaderContext } from './hooks/index.js';
import '@babel/runtime/helpers/defineProperty';
import './components/Item.js';
import 'tdesign-icons-vue-next';
import './utils/className.js';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/slicedToArray';
import 'lodash-es';
import './utils/helper.js';
import '../checkbox/index.js';
import '../checkbox/checkbox.js';
import '../checkbox/props.js';
import '@babel/runtime/helpers/typeof';
import '../_chunks/dep-c68ea098.js';
import '../_chunks/dep-f0f392fb.js';
import '../_chunks/dep-7bdccf65.js';
import '../_chunks/dep-d518fdfb.js';
import '../_chunks/dep-8d4d971b.js';
import '../_chunks/dep-98d89c71.js';
import '../config-provider/hooks/useConfig.js';
import '../config-provider/utils/context.js';
import '../_chunks/dep-509ddbe3.js';
import 'dayjs';
import '../_chunks/dep-e8dd47a9.js';
import '../_chunks/dep-2ba9b7d0.js';
import '../_chunks/dep-91fc762d.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../_chunks/dep-8be9c790.js';
import '../_chunks/dep-d67d6f8a.js';
import '../_chunks/dep-ef7a41ce.js';
import '../_chunks/dep-e332908e.js';
import '../checkbox/constants/index.js';
import '../checkbox/hooks/useCheckboxLazyLoad.js';
import '../_chunks/dep-6f148007.js';
import '../checkbox/hooks/useKeyboardEvent.js';
import '../_chunks/dep-e9e05226.js';
import '../checkbox/group.js';
import '../checkbox/checkbox-group-props.js';
import '../_chunks/dep-b36a4e94.js';
import '../_chunks/dep-ae3e94b6.js';
import 'tdesign-vue-next/esm/common/style/web/components/checkbox/_index.less';
import '../loading/index.js';
import '../loading/directive.js';
import '../loading/plugin.js';
import '../_chunks/dep-2a7145b5.js';
import '../loading/icon/gradient.js';
import '../_chunks/dep-fbf70ecb.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../loading/props.js';
import 'tdesign-vue-next/esm/common/style/web/components/loading/_index.less';
import '../_chunks/dep-52eae58a.js';
import './utils/effect.js';
import '../_chunks/dep-1e8be7b4.js';
import 'mitt';
import '../_chunks/dep-58eae151.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../_chunks/dep-06660d60.js';
import '../_chunks/dep-c3757953.js';

var _CascaderPanel = defineComponent({
  name: "TCascaderPanel",
  props: props,
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var _useCascaderContext = useCascaderContext(props2),
      cascaderContext = _useCascaderContext.cascaderContext;
    return function () {
      return createVNode(TCascaderSubPanel, {
        "trigger": props2.trigger,
        "cascaderContext": cascaderContext.value,
        "empty": props2.empty
      }, {
        empty: slots.empty,
        option: slots.option,
        loadingText: slots.loadingText
      });
    };
  }
});

export { _CascaderPanel as default };
