// tdesign-vue-next/cjs/tabs/tab-nav-item.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var tabs_props = require('./props.js');
var tabs_tabPanelProps = require('./tab-panel-props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$3 = require('../_chunks/dep-38acad3d.js');
var index$1 = require('../_chunks/dep-0238ed9a.js');
var index$2 = require('../_chunks/dep-c26a7d50.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
var index = require('../_chunks/dep-719a33c8.js');
require('../_chunks/dep-4a7162e7.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-4ba41627.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var TTabNavItem = Vue.defineComponent({
  name: "TTabNavItem",
  props: {
    index: Number,
    active: {
      type: Boolean
    },
    theme: tabs_props["default"].theme,
    size: tabs_props["default"].size,
    placement: tabs_props["default"].placement,
    label: {
      type: null
    },
    disabled: tabs_tabPanelProps["default"].disabled,
    removable: tabs_tabPanelProps["default"].removable,
    value: tabs_tabPanelProps["default"].value,
    draggable: tabs_tabPanelProps["default"].draggable,
    onClick: Function,
    onTabRemove: Function,
    onTabPanelRemove: Function
  },
  setup: function setup(props) {
    var itemRef = Vue.ref();
    index.useRipple(itemRef);
    var COMPONENT_NAME = index$1.usePrefixClass("tabs__nav-item");
    var _useGlobalIcon = index$2.useGlobalIcon({
        CloseIcon: tdesignIconsVueNext.CloseIcon
      }),
      CloseIcon = _useGlobalIcon.CloseIcon;
    var classPrefix = index$1.usePrefixClass();
    var _useCommonClassName = index$3.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS,
      SIZE = _useCommonClassName.SIZE;
    var removeBtnClick = function removeBtnClick(_ref) {
      var _props$onTabPanelRemo;
      var e = _ref.e;
      if (e) e.stopPropagation();
      props.onTabRemove({
        e: e,
        value: props.value,
        index: props.index
      });
      (_props$onTabPanelRemo = props.onTabPanelRemove) === null || _props$onTabPanelRemo === void 0 || _props$onTabPanelRemo.call(props, {
        e: e,
        value: props.value
      });
    };
    var onClickNav = function onClickNav(e) {
      if (props.disabled) return;
      props.onClick(e);
    };
    var navItemClass = Vue.computed(function () {
      return _defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, COMPONENT_NAME.value, true), "".concat(classPrefix.value, "-tabs__nav--card"), props.theme === "card"), STATUS.value.disabled, props.disabled), STATUS.value.active, props.active), "".concat(classPrefix.value, "-is-left"), props.placement === "left"), "".concat(classPrefix.value, "-is-right"), props.placement === "right"), SIZE.value.medium, props.size === "medium"), SIZE.value.large, props.size === "large");
    });
    var removeBtn = function removeBtn() {
      return props.removable && !props.disabled ? Vue.createVNode("span", {
        "onClick": function onClick(e) {
          return removeBtnClick({
            e: e
          });
        },
        "class": "remove-btn"
      }, [Vue.createVNode(CloseIcon, null, null)]) : null;
    };
    var renderCardItem = function renderCardItem() {
      return Vue.createVNode("div", {
        "class": navItemClass.value,
        "onClick": onClickNav,
        "ref": itemRef,
        "draggable": props.draggable
      }, [Vue.createVNode("span", {
        "class": "".concat(COMPONENT_NAME.value, "-text-wrapper")
      }, [props.label]), removeBtn()]);
    };
    var renderNormalItem = function renderNormalItem() {
      return Vue.createVNode("div", {
        "class": navItemClass.value,
        "onClick": onClickNav,
        "draggable": props.draggable
      }, [Vue.createVNode("div", {
        "class": ["".concat(COMPONENT_NAME.value, "-wrapper"), _defineProperty__default["default"](_defineProperty__default["default"]({}, STATUS.value.disabled, props.disabled), STATUS.value.active, props.active)],
        "ref": itemRef
      }, [Vue.createVNode("span", {
        "class": "".concat(COMPONENT_NAME.value, "-text-wrapper")
      }, [props.label])]), removeBtn()]);
    };
    return function () {
      return props.theme === "card" ? renderCardItem() : renderNormalItem();
    };
  }
});

exports["default"] = TTabNavItem;
