// tdesign-vue-next/cjs/qrcode/qrcode.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var qrcode_props = require('./props.js');
var qrcode_hooks_useQRCode = require('../_chunks/dep-a74d426d.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$1 = require('../_chunks/dep-1862d466.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
var index$2 = require('../_chunks/dep-cbefbfaf.js');
var qrcode_components_qrcodeCanvas = require('./components/qrcode-canvas.js');
var qrcode_components_qrcodeSvg = require('./components/qrcode-svg.js');
var qrcode_components_qrcodeStatus = require('./components/qrcode-status.js');
var isNumber = require('../_chunks/dep-44195fb2.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-a4be1915.js');
require('../_chunks/dep-43dcd2db.js');
require('../_chunks/dep-392dde71.js');
require('../_chunks/dep-02ea4f3c.js');
require('../_chunks/dep-5103206f.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-79476fcb.js');
require('../_chunks/dep-027cd76f.js');
require('./components/props.js');
require('tdesign-icons-vue-next');
require('../loading/index.js');
require('../_chunks/dep-34cd76d4.js');
require('../loading/plugin.js');
require('../_chunks/dep-d632b788.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-38acad3d.js');
require('../loading/props.js');
require('../_chunks/dep-7707ffe2.js');
require('../_chunks/dep-74a5bebe.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-42740075.js');
require('../_chunks/dep-9ac5c6ca.js');
require('../_chunks/dep-e3edff33.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var _QRCode = Vue.defineComponent({
  name: "TQRCode",
  props: qrcode_props["default"],
  setup: function setup(props2) {
    var classPrefix = index.usePrefixClass();
    var renderTNodeJSX = index$1.useTNodeJSX();
    var _useConfig = configProvider_hooks_useConfig.useConfig("qrcode"),
      globalConfig = _useConfig.globalConfig;
    var _useVariables = index$2.useVariables({
        themeFgColor: "--td-text-color-primary",
        themeBgColor: "--td-bg-color-specialcomponent"
      }),
      themeFgColor = _useVariables.themeFgColor,
      themeBgColor = _useVariables.themeBgColor;
    var finalBgColor = Vue.computed(function () {
      return props2.bgColor || themeBgColor.value || "transparent";
    });
    var finalFgColor = Vue.computed(function () {
      return props2.color || themeFgColor.value || qrcode_hooks_useQRCode.DEFAULT_FRONT_COLOR;
    });
    var imageSettings = Vue.computed(function () {
      var _props2$iconSize$heig, _props2$iconSize, _props2$iconSize$widt, _props2$iconSize2;
      return {
        src: props2.icon,
        x: void 0,
        y: void 0,
        height: isNumber.isNumber(props2.iconSize) ? props2.iconSize : (_props2$iconSize$heig = (_props2$iconSize = props2.iconSize) === null || _props2$iconSize === void 0 ? void 0 : _props2$iconSize.height) !== null && _props2$iconSize$heig !== void 0 ? _props2$iconSize$heig : 40,
        width: isNumber.isNumber(props2.iconSize) ? props2.iconSize : (_props2$iconSize$widt = (_props2$iconSize2 = props2.iconSize) === null || _props2$iconSize2 === void 0 ? void 0 : _props2$iconSize2.width) !== null && _props2$iconSize$widt !== void 0 ? _props2$iconSize$widt : 40,
        excavate: true,
        crossOrigin: "anonymous"
      };
    });
    var classes = Vue.computed(function () {
      return ["".concat(classPrefix.value, "-qrcode"), _defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(classPrefix.value, "-borderless"), props2.borderless), "".concat(classPrefix.value, "-qrcode-svg"), props2.type === "svg")];
    });
    var mergedStyle = Vue.computed(function () {
      return {
        backgroundColor: finalBgColor.value,
        width: "".concat(props2.size, "px"),
        height: "".concat(props2.size, "px")
      };
    });
    return function () {
      var QRCodeProps = {
        value: props2.value,
        size: props2.size,
        bgColor: finalBgColor.value,
        fgColor: finalFgColor.value,
        imageSettings: props2.icon ? imageSettings.value : void 0,
        level: props2.level
      };
      return Vue.createVNode("div", Vue.mergeProps({
        "class": classes.value,
        "style": mergedStyle.value
      }, {
        level: props2.level
      }), [props2.status !== "active" && Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-mask"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-").concat(props2.status), props2.status !== "loading")]
      }, [Vue.createVNode(qrcode_components_qrcodeStatus["default"], {
        "locale": globalConfig.value,
        "status": props2.status,
        "onRefresh": props2.onRefresh,
        "statusRender": renderTNodeJSX("statusRender", {
          params: {
            status: props2.status,
            onRefresh: props2.onRefresh
          }
        })
      }, null)]), props2.type === "canvas" ? Vue.createVNode(qrcode_components_qrcodeCanvas["default"], Vue.mergeProps(QRCodeProps, {
        "size": props2.size
      }), null) : Vue.createVNode(qrcode_components_qrcodeSvg["default"], Vue.mergeProps(QRCodeProps, {
        "size": props2.size
      }), null)]);
    };
  }
});

exports["default"] = _QRCode;
