// tdesign-vue-next/cjs/loading/plugin.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var loading_loading = require('../_chunks/dep-d632b788.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
var dom = require('../_chunks/dep-8d990e66.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/defineProperty');
var configProvider_utils_context = require('../_chunks/dep-be1af85d.js');
require('./icon/gradient.js');
require('../_chunks/dep-35e23c2b.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-44195fb2.js');
require('../_chunks/dep-1862d466.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-38acad3d.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('./props.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');

var fullScreenLoadingInstance = null;
function mergeDefaultProps(props) {
  var options = configProvider_utils_context.merge({
    fullscreen: false,
    attach: "body",
    loading: true,
    preventScrollThrough: true
  }, props);
  return options;
}
function createLoading(props, context) {
  var mergedProps = mergeDefaultProps(props);
  if (mergedProps.fullscreen && fullScreenLoadingInstance) {
    return fullScreenLoadingInstance;
  }
  var parentRelativeClass = Vue.ref("");
  var lockClass = Vue.ref("");
  var component = Vue.defineComponent({
    setup: function setup() {
      var loadingOptions = Vue.reactive(mergedProps);
      parentRelativeClass.value = index.usePrefixClass("loading__parent--relative").value;
      lockClass.value = index.usePrefixClass("loading--lock").value;
      return function () {
        return Vue.h(loading_loading._Loading, loadingOptions);
      };
    }
  });
  var attach = dom.getAttach(mergedProps.fullscreen ? "body" : mergedProps.attach);
  var instance = Vue.createVNode(component);
  if (context !== null && context !== void 0 ? context : LoadingPlugin._context) {
    instance.appContext = context !== null && context !== void 0 ? context : LoadingPlugin._context;
  }
  var wrapper = document.createElement("div");
  Vue.render(instance, wrapper);
  var lockFullscreen = mergedProps.preventScrollThrough && mergedProps.fullscreen;
  if (lockFullscreen) {
    dom.addClass(document.body, lockClass.value);
  }
  if (attach) {
    dom.addClass(attach, parentRelativeClass.value);
  } else {
    console.error("attach is not exist");
  }
  var loadingInstance = {
    hide: function hide() {
      dom.removeClass(attach, parentRelativeClass.value);
      dom.removeClass(document.body, lockClass.value);
      Vue.render(null, wrapper);
      wrapper.remove();
    }
  };
  return loadingInstance;
}
function produceLoading(props, context) {
  if (props === true) {
    fullScreenLoadingInstance = createLoading({
      fullscreen: true,
      loading: true,
      attach: "body",
      preventScrollThrough: true
    }, context);
    return fullScreenLoadingInstance;
  }
  if (props === false) {
    var _fullScreenLoadingIns;
    (_fullScreenLoadingIns = fullScreenLoadingInstance) === null || _fullScreenLoadingIns === void 0 || _fullScreenLoadingIns.hide();
    fullScreenLoadingInstance = null;
    return;
  }
  return createLoading(props);
}
var LoadingPlugin = produceLoading;
LoadingPlugin.install = function (app) {
  app.config.globalProperties.$loading = produceLoading;
  LoadingPlugin._context = app._context;
};

exports.LoadingPlugin = LoadingPlugin;
exports["default"] = LoadingPlugin;
