// tdesign-vue-next/es/menu/components/popup-overflow-content.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, ref, onMounted, onUpdated, createVNode } from 'vue';
import { u as usePrefixClass } from '../../_chunks/dep-fbeb2963.mjs';
import 'lodash-es';
import '../../_chunks/dep-47c3cae9.mjs';
import '../../_chunks/dep-b0b422fe.mjs';
import '../../_chunks/dep-e1c62679.mjs';
import '../../_chunks/dep-feed89df.mjs';
import '../../_chunks/dep-5b510fcd.mjs';
import '../../config-provider/hooks/useConfig.mjs';
import '../../config-provider/utils/context.mjs';
import '../../_chunks/dep-e0ab6d55.mjs';
import '../../_chunks/dep-75f265c4.mjs';
import '../../_chunks/dep-67ffcfeb.mjs';
import '../../_chunks/dep-f558ebb8.mjs';
import '../../_chunks/dep-edab0e6a.mjs';
import '../../_chunks/dep-76dddedf.mjs';
import '../../_chunks/dep-d92220c4.mjs';
import '../../_chunks/dep-466a8f31.mjs';
import '../../_chunks/dep-ba214d75.mjs';
import '../../_chunks/dep-e66c0337.mjs';

var PopupOverflowContent = defineComponent({
  name: "PopupOverflowContent",
  props: {
    foldIndex: {
      type: Number,
      required: true
    }
  },
  setup: function setup(props, ctx) {
    var classPrefix = usePrefixClass();
    var wrapperRef = ref();
    var hideBeforeFoldIndex = function hideBeforeFoldIndex() {
      if (!wrapperRef.value) return;
      var menuItemClass = "".concat(classPrefix.value, "-menu__item");
      var submenuClass = "".concat(classPrefix.value, "-submenu");
      var items = [];
      var _collect = function collect(parent, depth) {
        if (depth > 3) return;
        Array.from(parent.children).forEach(function (el) {
          if (!(el instanceof HTMLElement)) return;
          if (el.classList.contains(menuItemClass) || el.classList.contains(submenuClass)) {
            items.push(el);
          } else {
            _collect(el, depth + 1);
          }
        });
      };
      _collect(wrapperRef.value, 0);
      items.forEach(function (el, index) {
        el.style.display = index < props.foldIndex ? "none" : "";
      });
    };
    onMounted(hideBeforeFoldIndex);
    onUpdated(hideBeforeFoldIndex);
    return function () {
      var _ctx$slots$default, _ctx$slots;
      return createVNode("div", {
        "ref": wrapperRef
      }, [(_ctx$slots$default = (_ctx$slots = ctx.slots)["default"]) === null || _ctx$slots$default === void 0 ? void 0 : _ctx$slots$default.call(_ctx$slots)]);
    };
  }
});

export { PopupOverflowContent as default };
