// tdesign-vue-next/cjs/empty/empty.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$2 = require('../_chunks/dep-1862d466.js');
var index$1 = require('../_chunks/dep-38acad3d.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/defineProperty');
var empty_props = require('./props.js');
var image_index = require('../image/index.js');
var empty_components_MaintenanceSvg = require('./components/MaintenanceSvg.js');
var empty_components_NetworkErrorSvg = require('./components/NetworkErrorSvg.js');
var empty_components_EmptySvg = require('./components/EmptySvg.js');
var empty_components_FailSvg = require('./components/FailSvg.js');
var empty_components_SuccessSvg = require('./components/SuccessSvg.js');
var isString = require('../_chunks/dep-631750cd.js');
var configProvider_utils_context = require('../_chunks/dep-be1af85d.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../image/image.js');
require('tdesign-icons-vue-next');
require('../_chunks/dep-b9a88912.js');
require('../_chunks/dep-03ea53c4.js');
require('../_chunks/dep-50d6b5ee.js');
require('../_chunks/dep-692bdebe.js');
require('../image/props.js');
require('../space/index.js');
require('../space/space.js');
require('../space/props.js');
require('../_chunks/dep-0cda9fe9.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-e3edff33.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');

var _Empty = Vue.defineComponent({
  name: "TEmpty",
  components: {
    TImage: image_index.Image
  },
  props: empty_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var _toRefs = Vue.toRefs(props2),
      size = _toRefs.size,
      propsImage = _toRefs.image,
      propsDescription = _toRefs.description,
      propsTitle = _toRefs.title,
      type = _toRefs.type;
    var _useConfig = configProvider_hooks_useConfig.useConfig("empty"),
      globalConfig = _useConfig.globalConfig;
    var classPrefix = index.usePrefixClass("empty");
    var showAction = Vue.computed(function () {
      return props2.action || slots.action;
    });
    var _useCommonClassName = index$1.useCommonClassName(),
      SIZE = _useCommonClassName.SIZE;
    var renderTNodeJSX = index$2.useTNodeJSX();
    var defaultMaps = Vue.computed(function () {
      return {
        maintenance: {
          image: globalConfig.value.image.maintenance || empty_components_MaintenanceSvg["default"],
          title: globalConfig.value.titleText.maintenance
        },
        success: {
          image: globalConfig.value.image.success || empty_components_SuccessSvg["default"],
          title: globalConfig.value.titleText.success
        },
        fail: {
          image: globalConfig.value.image.fail || empty_components_FailSvg["default"],
          title: globalConfig.value.titleText.fail
        },
        "network-error": {
          image: globalConfig.value.image.networkError || empty_components_NetworkErrorSvg["default"],
          title: globalConfig.value.titleText.networkError
        },
        empty: {
          image: globalConfig.value.image.empty || empty_components_EmptySvg["default"],
          title: globalConfig.value.titleText.empty
        }
      };
    });
    var emptyClasses = Vue.computed(function () {
      return [classPrefix.value, SIZE.value[size.value]];
    });
    var titleClasses = ["".concat(classPrefix.value, "__title")];
    var imageClasses = ["".concat(classPrefix.value, "__image")];
    var descriptionClasses = ["".concat(classPrefix.value, "__description")];
    var actionClass = ["".concat(classPrefix.value, "__action")];
    var typeImageProps = Vue.computed(function () {
      var _defaultMaps$value$ty;
      return (_defaultMaps$value$ty = defaultMaps.value[type.value]) !== null && _defaultMaps$value$ty !== void 0 ? _defaultMaps$value$ty : null;
    });
    var showImage = Vue.computed(function () {
      var _slots$image, _typeImageProps$value;
      return propsImage.value || (slots === null || slots === void 0 || (_slots$image = slots.image) === null || _slots$image === void 0 ? void 0 : _slots$image.call(slots)) || ((_typeImageProps$value = typeImageProps.value) === null || _typeImageProps$value === void 0 ? void 0 : _typeImageProps$value.image);
    });
    var showTitle = Vue.computed(function () {
      var _slots$title, _typeImageProps$value2;
      return propsTitle.value || (slots === null || slots === void 0 || (_slots$title = slots.title) === null || _slots$title === void 0 ? void 0 : _slots$title.call(slots)) || ((_typeImageProps$value2 = typeImageProps.value) === null || _typeImageProps$value2 === void 0 ? void 0 : _typeImageProps$value2.title);
    });
    var showDescription = Vue.computed(function () {
      var _slots$description;
      return propsDescription.value || (slots === null || slots === void 0 || (_slots$description = slots.description) === null || _slots$description === void 0 ? void 0 : _slots$description.call(slots));
    });
    var renderTitle = function renderTitle() {
      if (!showTitle.value) {
        return null;
      }
      return Vue.createVNode("div", {
        "class": titleClasses
      }, [showTitle.value]);
    };
    var renderDescription = function renderDescription() {
      if (!showDescription.value) {
        return null;
      }
      return Vue.createVNode("div", {
        "class": descriptionClasses
      }, [showDescription.value]);
    };
    var getImageIns = function getImageIns() {
      var data = showImage.value;
      var result = null;
      if (isString.isString(data)) {
        result = Vue.createVNode(image_index.Image, {
          "src": data
        }, null);
      } else if (data && Reflect.has(data, "setup")) {
        result = Vue.h(data);
      } else if (configProvider_utils_context.isPlainObject(data)) {
        result = Vue.createVNode(image_index.Image, data, null);
      }
      return result;
    };
    return function () {
      return Vue.createVNode("div", {
        "class": emptyClasses.value
      }, [showImage.value ? Vue.createVNode("div", {
        "class": imageClasses,
        "style": props2.imageStyle
      }, [slots !== null && slots !== void 0 && slots.image ? renderTNodeJSX("image") : getImageIns()]) : null, renderTitle(), renderDescription(), showAction.value ? Vue.createVNode("div", {
        "class": actionClass
      }, [renderTNodeJSX("action")]) : null]);
    };
  }
});

exports["default"] = _Empty;
