// tdesign-vue-next/cjs/menu/components/popup-overflow-content.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var index = require('../../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/defineProperty');
require('../../_chunks/dep-4a7162e7.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

var PopupOverflowContent = Vue.defineComponent({
  name: "PopupOverflowContent",
  props: {
    foldIndex: {
      type: Number,
      required: true
    }
  },
  setup: function setup(props, ctx) {
    var classPrefix = index.usePrefixClass();
    var wrapperRef = Vue.ref();
    var hideBeforeFoldIndex = function hideBeforeFoldIndex() {
      if (!wrapperRef.value) return;
      var menuItemClass = "".concat(classPrefix.value, "-menu__item");
      var submenuClass = "".concat(classPrefix.value, "-submenu");
      var items = [];
      var _collect = function collect(parent, depth) {
        if (depth > 3) return;
        Array.from(parent.children).forEach(function (el) {
          if (!(el instanceof HTMLElement)) return;
          if (el.classList.contains(menuItemClass) || el.classList.contains(submenuClass)) {
            items.push(el);
          } else {
            _collect(el, depth + 1);
          }
        });
      };
      _collect(wrapperRef.value, 0);
      items.forEach(function (el, index) {
        el.style.display = index < props.foldIndex ? "none" : "";
      });
    };
    Vue.onMounted(hideBeforeFoldIndex);
    Vue.onUpdated(hideBeforeFoldIndex);
    return function () {
      var _ctx$slots$default, _ctx$slots;
      return Vue.createVNode("div", {
        "ref": wrapperRef
      }, [(_ctx$slots$default = (_ctx$slots = ctx.slots)["default"]) === null || _ctx$slots$default === void 0 ? void 0 : _ctx$slots$default.call(_ctx$slots)]);
    };
  }
});

exports["default"] = PopupOverflowContent;
