// tdesign-vue-next/es/cascader/cascader-panel.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, createVNode } from 'vue';
import TCascaderSubPanel from './components/Panel.mjs';
import props from './props.mjs';
import { useCascaderContext } from './hooks/index.mjs';
import '../_chunks/dep-ba214d75.mjs';
import '../_chunks/dep-e1c62679.mjs';
import './components/Item.mjs';
import 'tdesign-icons-vue-next';
import './utils/index.mjs';
import './utils/className.mjs';
import '../_chunks/dep-47c3cae9.mjs';
import '../_chunks/dep-b0b422fe.mjs';
import './utils/effect.mjs';
import '../_chunks/dep-f558ebb8.mjs';
import 'lodash-es';
import './utils/helper.mjs';
import '../checkbox/index.mjs';
import '../checkbox/checkbox.mjs';
import '../checkbox/props.mjs';
import '../_chunks/dep-fbeb2963.mjs';
import '../_chunks/dep-feed89df.mjs';
import '../_chunks/dep-5b510fcd.mjs';
import '../config-provider/hooks/useConfig.mjs';
import '../config-provider/utils/context.mjs';
import '../_chunks/dep-e0ab6d55.mjs';
import '../_chunks/dep-75f265c4.mjs';
import '../_chunks/dep-67ffcfeb.mjs';
import '../_chunks/dep-edab0e6a.mjs';
import '../_chunks/dep-76dddedf.mjs';
import '../_chunks/dep-d92220c4.mjs';
import '../_chunks/dep-466a8f31.mjs';
import '../_chunks/dep-e66c0337.mjs';
import '../checkbox/constants/index.mjs';
import '../checkbox/hooks/useCheckboxLazyLoad.mjs';
import '../checkbox/hooks/useKeyboardEvent.mjs';
import '../checkbox/group.mjs';
import '../checkbox/checkbox-group-props.mjs';
import './style/css.mjs';
import '../loading/index.mjs';
import '../loading/directive.mjs';
import '../loading/plugin.mjs';
import '../loading/loading.mjs';
import '../loading/icon/gradient.mjs';
import '../_chunks/dep-58979704.mjs';
import '../_chunks/dep-c231a5b4.mjs';
import '../loading/props.mjs';
import '../_chunks/dep-69b56438.mjs';
import 'mitt';
import '../_chunks/dep-9e0400d9.mjs';
import '../_chunks/dep-3114e4de.mjs';
import '../_chunks/dep-739ea4cd.mjs';

var _CascaderPanel = defineComponent({
  name: "TCascaderPanel",
  props: props,
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var _useCascaderContext = useCascaderContext(props2),
      cascaderContext = _useCascaderContext.cascaderContext;
    return function () {
      return createVNode(TCascaderSubPanel, {
        "trigger": props2.trigger,
        "cascaderContext": cascaderContext.value,
        "empty": props2.empty
      }, {
        empty: slots.empty,
        option: slots.option,
        loadingText: slots.loadingText
      });
    };
  }
});

export { _CascaderPanel as default };
