// tdesign-vue-next/es/date-picker/components/panel/ExtraContent.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import TDateFooter from '../base/Footer.mjs';
import '../../../_chunks/dep-ba214d75.mjs';
import '../../../_chunks/dep-e1c62679.mjs';
import 'lodash-es';
import '../../../_chunks/dep-fbeb2963.mjs';
import '../../../_chunks/dep-47c3cae9.mjs';
import '../../../_chunks/dep-b0b422fe.mjs';
import '../../../_chunks/dep-feed89df.mjs';
import '../../../_chunks/dep-5b510fcd.mjs';
import '../../../config-provider/hooks/useConfig.mjs';
import '../../../config-provider/utils/context.mjs';
import '../../../_chunks/dep-e0ab6d55.mjs';
import '../../../_chunks/dep-75f265c4.mjs';
import '../../../_chunks/dep-67ffcfeb.mjs';
import '../../../_chunks/dep-f558ebb8.mjs';
import '../../../_chunks/dep-edab0e6a.mjs';
import '../../../_chunks/dep-76dddedf.mjs';
import '../../../_chunks/dep-d92220c4.mjs';
import '../../../_chunks/dep-466a8f31.mjs';
import '../../../_chunks/dep-e66c0337.mjs';
import '../../../button/index.mjs';
import '../../../button/button.mjs';
import '../../../loading/index.mjs';
import '../../../loading/directive.mjs';
import '../../../loading/plugin.mjs';
import '../../../loading/loading.mjs';
import '../../../loading/icon/gradient.mjs';
import '../../../_chunks/dep-58979704.mjs';
import '../../../_chunks/dep-c231a5b4.mjs';
import '../../../loading/props.mjs';
import './style/css.mjs';
import '../../../button/props.mjs';

var TExtraContent = defineComponent({
  name: "TExtraContent",
  props: {
    presets: Object,
    enableTimePicker: Boolean,
    presetsPlacement: String,
    needConfirm: Boolean,
    onPresetClick: Function,
    onConfirmClick: Function,
    selectedValue: [String, Number, Array, Date]
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var showPanelFooter = computed(function () {
      return props.enableTimePicker && props.needConfirm || props.presets || !!slots.presets;
    });
    return function () {
      return showPanelFooter.value ? createVNode(TDateFooter, {
        "presets": props.presets,
        "onPresetClick": props.onPresetClick,
        "enableTimePicker": props.enableTimePicker,
        "onConfirmClick": props.onConfirmClick,
        "presetsPlacement": props.presetsPlacement,
        "selectedValue": props.selectedValue,
        "needConfirm": props.needConfirm
      }, {
        presets: slots.presets
      }) : null;
    };
  }
});

export { TExtraContent as default };
