// tdesign-vue-next/cjs/time-picker/panel/time-picker-panel.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var dayjs = require('dayjs');
var customParseFormat = require('../../_chunks/dep-484d5d62.js');
var timePicker_panel_props = require('../../_chunks/dep-5241cd36.js');
var timePicker_panel_singlePanel = require('../../_chunks/dep-f198a3a2.js');
var button_button = require('../../button/button.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var index = require('../../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
var log = require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-0989c3be.js');
var isUndefined = require('../../_chunks/dep-3d29e358.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-4a7162e7.js');
require('../props.js');
require('../../_chunks/dep-74b143e0.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-58022b88.js');
require('../../_chunks/dep-9bee89d3.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-38acad3d.js');
require('../../_chunks/dep-79476fcb.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-9ac5c6ca.js');
require('../../loading/index.js');
require('../../_chunks/dep-34cd76d4.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-d632b788.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-4ba41627.js');
require('../../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6212811d.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-44195fb2.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-1862d466.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../loading/props.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-43dcd2db.js');
require('../../_chunks/dep-392dde71.js');
require('../../_chunks/dep-02ea4f3c.js');
require('../../_chunks/dep-5103206f.js');
require('../../_chunks/dep-7707ffe2.js');
require('../../_chunks/dep-74a5bebe.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-42740075.js');
require('../../_chunks/dep-e3edff33.js');
require('../../button/props.js');
require('../../_chunks/dep-362c5212.js');
require('../../_chunks/dep-719a33c8.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
dayjs__default["default"].extend(customParseFormat.customParseFormat);
var _TimePickerPanel = Vue.defineComponent({
  name: "TTimePickerPanel",
  props: _objectSpread(_objectSpread({}, timePicker_panel_props.panelProps()), {}, {
    isFooterDisplay: Boolean,
    handleConfirmClick: Function,
    onChange: Function,
    position: String
  }),
  setup: function setup(props) {
    var _useConfig = configProvider_hooks_useConfig.useConfig("timePicker"),
      globalConfig = _useConfig.globalConfig;
    var COMPONENT_NAME = index.usePrefixClass("time-picker__panel");
    var _toRefs = Vue.toRefs(props),
      steps = _toRefs.steps,
      isFooterDisplay = _toRefs.isFooterDisplay,
      isShowPanel = _toRefs.isShowPanel;
    var triggerScroll = Vue.ref(false);
    var panelRef = Vue.ref();
    var showNowTimeBtn = Vue.computed(function () {
      return !!steps.value.filter(function (step) {
        return Number(step) > 1;
      }).length;
    });
    var defaultValue = Vue.computed(function () {
      var isStepsSet = showNowTimeBtn.value;
      var formattedValue = dayjs__default["default"](props.value, props.format);
      if (props.value && formattedValue.isValid()) {
        return formattedValue.format(props.format);
      }
      if (isStepsSet) {
        return dayjs__default["default"]().hour(0).minute(0).second(0).format(props.format);
      }
      return dayjs__default["default"]().hour(0).minute(0).second(0).format(props.format);
    });
    var panelColUpdate = function panelColUpdate() {
      Vue.nextTick(function () {
        triggerScroll.value = true;
      });
    };
    var resetTriggerScroll = function resetTriggerScroll() {
      triggerScroll.value = false;
    };
    var handlePresetClick = function handlePresetClick(presetValue) {
      var presetVal = typeof presetValue === "function" ? presetValue() : presetValue;
      if (typeof props.activeIndex === "number") {
        if (Array.isArray(presetVal)) {
          var _props$onChange;
          (_props$onChange = props.onChange) === null || _props$onChange === void 0 || _props$onChange.call(props, presetVal[props.activeIndex]);
        } else {
          log.log.error("TimePicker", "preset: ".concat(props.presets, " \u9884\u8BBE\u503C\u5FC5\u987B\u662F\u6570\u7EC4!"));
        }
      } else {
        var _props$onChange2;
        (_props$onChange2 = props.onChange) === null || _props$onChange2 === void 0 || _props$onChange2.call(props, presetVal);
      }
    };
    var renderFooter = function renderFooter() {
      if (!isUndefined.isUndefined(props.presets)) return Object.keys(props.presets || []).map(function (key) {
        return Vue.createVNode(button_button["default"], {
          "key": key,
          "theme": "primary",
          "size": "small",
          "variant": "text",
          "onClick": function onClick() {
            return handlePresetClick(props.presets[key]);
          }
        }, _isSlot(key) ? key : {
          "default": function _default() {
            return [key];
          }
        });
      });
      return !showNowTimeBtn.value ? Vue.createVNode(button_button["default"], {
        "theme": "primary",
        "variant": "text",
        "size": "small",
        "onClick": function onClick() {
          var _props$onChange3;
          return (_props$onChange3 = props.onChange) === null || _props$onChange3 === void 0 ? void 0 : _props$onChange3.call(props, dayjs__default["default"]().format(props.format));
        }
      }, {
        "default": function _default() {
          return [globalConfig.value.now];
        }
      }) : null;
    };
    Vue.onMounted(function () {
      panelColUpdate();
    });
    Vue.watch(function () {
      return isShowPanel.value;
    }, function () {
      panelColUpdate();
    });
    return function () {
      return Vue.createVNode("div", {
        "class": COMPONENT_NAME.value
      }, [Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-section-body")
      }, [Vue.createVNode(timePicker_panel_singlePanel.SinglePanel, Vue.mergeProps(props, {
        "ref": panelRef,
        "format": props.format || timePicker_panel_props.DEFAULT_FORMAT,
        "steps": props.steps || timePicker_panel_props.DEFAULT_STEPS,
        "value": dayjs__default["default"](props.value, props.format).isValid() ? props.value : defaultValue.value,
        "triggerScroll": triggerScroll.value,
        "onChange": props.onChange,
        "resetTriggerScroll": resetTriggerScroll,
        "isShowPanel": props.isShowPanel
      }), null)]), isFooterDisplay.value ? Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-section-footer")
      }, [Vue.createVNode(button_button["default"], {
        "theme": "primary",
        "variant": "base",
        "disabled": !props.value,
        "onClick": function onClick(e) {
          var _props$handleConfirmC;
          return (_props$handleConfirmC = props.handleConfirmClick) === null || _props$handleConfirmC === void 0 ? void 0 : _props$handleConfirmC.call(props, e, defaultValue.value);
        },
        "size": "small"
      }, {
        "default": function _default() {
          return [globalConfig.value.confirm];
        }
      }), renderFooter()]) : null]);
    };
  }
});

exports["default"] = _TimePickerPanel;
