// tdesign-vue-next/esm/color-picker/components/panel/alpha.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, computed, createVNode } from 'vue';
import ColorSlider from './slider.js';
import { useBaseClassName } from '../../hooks/index.js';
import '@babel/runtime/helpers/defineProperty';
import '../../../_chunks/dep-fd7e9119.js';
import '../../../_chunks/dep-233529f8.js';
import '@babel/runtime/helpers/slicedToArray';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/createClass';
import 'tinycolor2';
import 'lodash-es';
import '../../../_chunks/dep-0ce1ea95.js';
import '@babel/runtime/helpers/toConsumableArray';
import './base-props.js';
import '@babel/runtime/helpers/typeof';
import '../../../_chunks/dep-c68ea098.js';
import '../../../_chunks/dep-f0f392fb.js';
import '../../../_chunks/dep-e8dd47a9.js';
import '../../../config-provider/hooks/useConfig.js';
import '../../../config-provider/utils/context.js';
import '../../../_chunks/dep-509ddbe3.js';
import 'dayjs';
import '../../../_chunks/dep-91fc762d.js';

var AlphaSlider = defineComponent({
  name: "AlphaSlider",
  inheritAttrs: false,
  props: {
    color: {
      type: Object
    },
    disabled: {
      type: Boolean,
      "default": false
    },
    onChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    }
  },
  setup: function setup(props) {
    var baseClassName = useBaseClassName();
    var handleChange = function handleChange(v, isDragEnd) {
      props.onChange(v / 100, isDragEnd);
    };
    var railStyle = computed(function () {
      return {
        background: "linear-gradient(to right, rgba(0, 0, 0, 0), ".concat(props.color.rgb, ")")
      };
    });
    return function () {
      return createVNode(ColorSlider, {
        "class": ["".concat(baseClassName.value, "__alpha"), "".concat(baseClassName.value, "--bg-alpha")],
        "color": props.color,
        "value": props.color.alpha * 100,
        "onChange": handleChange,
        "rail-style": railStyle.value,
        "max-value": 100,
        "disabled": props.disabled,
        "type": "alpha"
      }, null);
    };
  }
});

export { AlphaSlider as default };
