// tdesign-vue-next/esm/color-picker/components/panel/header.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, ref, watch, createVNode, isVNode } from 'vue';
import _defineProperty from '@babel/runtime/helpers/defineProperty';
import { c as COLOR_MODES } from '../../../_chunks/dep-fd7e9119.js';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../../_chunks/dep-c68ea098.js';
import { useConfig } from '../../../config-provider/hooks/useConfig.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../../_chunks/dep-91fc762d.js';
import { RadioGroup, RadioButton } from '../../../radio/index.js';
import props from '../../color-picker-panel-props.js';
import { useBaseClassName } from '../../hooks/index.js';
import '../../../_chunks/dep-f0f392fb.js';
import '../../../config-provider/utils/context.js';
import '../../../_chunks/dep-509ddbe3.js';
import 'dayjs';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../../../radio/radio.js';
import '../../../_chunks/dep-7bdccf65.js';
import '../../../_chunks/dep-d518fdfb.js';
import '../../../_chunks/dep-8d4d971b.js';
import '../../../_chunks/dep-98d89c71.js';
import '../../../_chunks/dep-e8dd47a9.js';
import '../../../_chunks/dep-2ba9b7d0.js';
import '../../../_chunks/dep-8be9c790.js';
import '../../../_chunks/dep-e332908e.js';
import '../../../radio/props.js';
import '../../../radio/constants/index.js';
import '../../../_chunks/dep-fbf70ecb.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../../radio/group.js';
import '@babel/runtime/helpers/asyncToGenerator';
import '@babel/runtime/regenerator';
import '../../../radio/radio-group-props.js';
import '../../../radio/radio-button.js';
import '../../../_chunks/dep-d5fc4caa.js';
import '../../../radio/hooks/useKeyboard.js';
import '../../../_chunks/dep-e9e05226.js';
import '../../../watermark/hooks/index.js';
import '../../../_chunks/dep-ae3e94b6.js';
import 'tdesign-vue-next/esm/common/style/web/components/radio/_index.less';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !isVNode(s);
}
var PanelHeader = defineComponent({
  name: "PanelHeader",
  props: _objectSpread(_objectSpread({}, props), {}, {
    mode: {
      type: String,
      "default": "color"
    },
    onModeChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    }
  }),
  setup: function setup(props2) {
    var _useConfig = useConfig("colorPicker"),
      globalConfig = _useConfig.globalConfig;
    var baseClassName = useBaseClassName();
    var modeValue = ref(props2.mode);
    var handleModeChange = function handleModeChange(v) {
      return props2.onModeChange(v);
    };
    watch(function () {
      return props2.mode;
    }, function (v) {
      return modeValue.value = v;
    });
    return function () {
      var _slot2;
      var _props2$colorModes;
      if (((_props2$colorModes = props2.colorModes) === null || _props2$colorModes === void 0 ? void 0 : _props2$colorModes.length) === 1) {
        return null;
      }
      return createVNode("div", {
        "class": "".concat(baseClassName.value, "__head")
      }, [createVNode("div", {
        "class": "".concat(baseClassName.value, "__mode")
      }, [createVNode(RadioGroup, {
        "variant": "default-filled",
        "size": "small",
        "modelValue": modeValue.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return modeValue.value = $event;
        },
        "onChange": handleModeChange
      }, _isSlot(_slot2 = Object.keys(COLOR_MODES).map(function (key) {
        var _slot;
        return createVNode(RadioButton, {
          "key": key,
          "value": key
        }, _isSlot(_slot = Reflect.get(globalConfig.value, COLOR_MODES[key])) ? _slot : {
          "default": function _default() {
            return [_slot];
          }
        });
      })) ? _slot2 : {
        "default": function _default() {
          return [_slot2];
        }
      })])]);
    };
  }
});

export { PanelHeader as default };
