// tdesign-vue-next/esm/color-picker/components/trigger/index.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, ref, watch, createVNode, mergeProps } from 'vue';
import _defineProperty from '@babel/runtime/helpers/defineProperty';
import { Input } from '../../../input/index.js';
import { g as getColorObject, C as Color } from '../../../_chunks/dep-233529f8.js';
import '@babel/runtime/helpers/classCallCheck';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/toConsumableArray';
import { useBaseClassName } from '../../hooks/index.js';
import 'lodash-es';
import '@babel/runtime/helpers/typeof';
import '../../../_chunks/dep-c68ea098.js';
import { u as useCommonClassName } from '../../../_chunks/dep-98d89c71.js';
import '../../../config-provider/hooks/useConfig.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../../_chunks/dep-91fc762d.js';
import { u as useEventForward } from '../../../_chunks/dep-cd9622b1.js';
import '../../../input/input.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import 'tdesign-icons-vue-next';
import '../../../input/props.js';
import '../../../_chunks/dep-7bdccf65.js';
import '../../../_chunks/dep-d518fdfb.js';
import '../../../_chunks/dep-8d4d971b.js';
import '../../../_chunks/dep-e8dd47a9.js';
import '../../../_chunks/dep-2ba9b7d0.js';
import '../../../_chunks/dep-52eae58a.js';
import '../../../_chunks/dep-8be9c790.js';
import '../../../input/hooks/useInput.js';
import '../../../form/constants/index.js';
import '../../../_chunks/dep-f0f392fb.js';
import '../../../config-provider/utils/context.js';
import '../../../_chunks/dep-509ddbe3.js';
import 'dayjs';
import '../../../_chunks/dep-e332908e.js';
import '../../../input/hooks/useLengthLimit.js';
import '../../../_chunks/dep-06660d60.js';
import '../../../_chunks/dep-fbf70ecb.js';
import '../../../input/hooks/useInputEventHandler.js';
import '../../../input/hooks/useInputWidth.js';
import '../../../_chunks/dep-d5fc4caa.js';
import '../../../input/input-group.js';
import '../../../input/input-group-props.js';
import '../../../_chunks/dep-ae3e94b6.js';
import 'tdesign-vue-next/esm/common/style/web/components/input/_index.less';
import 'tinycolor2';
import '../../../_chunks/dep-fd7e9119.js';

var DefaultTrigger = defineComponent({
  name: "DefaultTrigger",
  inheritAttrs: false,
  props: {
    color: {
      type: String,
      "default": ""
    },
    disabled: {
      type: Boolean,
      "default": false
    },
    borderless: {
      type: Boolean,
      "default": false
    },
    clearable: {
      type: Boolean,
      "default": false
    },
    inputProps: {
      type: Object,
      "default": function _default() {
        return {
          autoWidth: true
        };
      }
    },
    onTriggerChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    },
    onTriggerClear: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    },
    size: {
      type: String,
      "default": "medium"
    }
  },
  setup: function setup(props) {
    var baseClassName = useBaseClassName();
    var value = ref(props.color);
    var _useCommonClassName = useCommonClassName(),
      sizeClassNames = _useCommonClassName.SIZE;
    watch(function () {
      return [props.color];
    }, function () {
      return value.value = props.color;
    });
    var handleChange = function handleChange(input) {
      if (input !== props.color) {
        props.onTriggerChange(value.value, {
          color: getColorObject(new Color(input)),
          trigger: "input"
        });
      }
    };
    var handleClear = function handleClear(context) {
      var _props$onTriggerClear;
      return (_props$onTriggerClear = props.onTriggerClear) === null || _props$onTriggerClear === void 0 ? void 0 : _props$onTriggerClear.call(props, context);
    };
    var inputEvents = useEventForward(props.inputProps, {
      onBlur: handleChange,
      onChange: handleChange,
      onClear: handleClear
    });
    return function () {
      var inputSlots = {
        label: function label() {
          return createVNode("div", {
            "class": ["".concat(baseClassName.value, "__trigger--default__color"), "".concat(baseClassName.value, "--bg-alpha")]
          }, [createVNode("span", {
            "class": ["color-inner", _defineProperty({}, sizeClassNames.value[props.size], props.size !== "medium")],
            "style": {
              background: value.value
            }
          }, null)]);
        }
      };
      return createVNode(Input, mergeProps({
        "borderless": props.borderless,
        "clearable": props.clearable,
        "size": props.size,
        "modelValue": value.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return value.value = $event;
        },
        "disabled": props.disabled
      }, inputEvents.value), inputSlots);
    };
  }
});

export { DefaultTrigger as default };
