// tdesign-vue-next/cjs/descriptions/descriptions.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var index$2 = require('../_chunks/dep-0cda9fe9.js');
var index$3 = require('../_chunks/dep-1862d466.js');
var index$1 = require('../_chunks/dep-38acad3d.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
require('../_chunks/dep-35e23c2b.js');
require('@babel/runtime/helpers/toConsumableArray');
var descriptions_props = require('./props.js');
var descriptions_constants_index = require('./constants/index.js');
var descriptions_descriptionsRow = require('./descriptions-row.js');
var descriptions_utils_index = require('../_chunks/dep-c4ebb122.js');
var isArray = require('../_chunks/dep-4cb26289.js');
var isNil = require('../_chunks/dep-95977a93.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('../_chunks/dep-4a7162e7.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var _Descriptions = Vue.defineComponent({
  name: "TDescriptions",
  props: descriptions_props["default"],
  setup: function setup(props2) {
    var COMPONENT_NAME = index.usePrefixClass("descriptions");
    var _useCommonClassName = index$1.useCommonClassName(),
      SIZE = _useCommonClassName.SIZE;
    var getChildByName = index$2.useChildComponentSlots();
    var renderTNodeJSX = index$3.useTNodeJSX();
    var itemsType = Vue.ref(descriptions_utils_index.ItemsType.props);
    var getRows = function getRows() {
      var column = props2.column,
        layout = props2.layout;
      var items = [];
      if (isArray.isArray(props2.items)) {
        items = props2.items.map(function (item) {
          return {
            label: descriptions_utils_index.renderCustomNode(item.label),
            content: descriptions_utils_index.renderCustomNode(item.content),
            span: item.span || 1
          };
        });
        itemsType.value = descriptions_utils_index.ItemsType.props;
      } else {
        var slots = getChildByName("TDescriptionsItem");
        if (slots.length !== 0) {
          items = slots;
          itemsType.value = descriptions_utils_index.ItemsType.slots;
        }
      }
      if (layout === "vertical") {
        return [items];
      }
      var temp = [];
      var reset = column;
      var res = [];
      items.forEach(function (item, index) {
        var span = 1;
        if (descriptions_utils_index.itemTypeIsProps(itemsType.value)) {
          span = isNil.isNil(item.span) ? span : item.span;
          span = span > column ? column : span;
        } else {
          var _item$props;
          item.props = item.props || {};
          span = isNil.isNil((_item$props = item.props) === null || _item$props === void 0 ? void 0 : _item$props.span) ? span : item.props.span;
          span = span > column ? column : span;
          item.props.span = span;
        }
        if (reset >= span) {
          temp.push(item);
          reset -= span;
        } else {
          res.push(temp);
          temp = [item];
          reset = column - span;
        }
        if (index === items.length - 1) {
          if (descriptions_utils_index.itemTypeIsProps(itemsType.value)) {
            item.span += reset;
          } else {
            item.props.span += reset;
          }
          res.push(temp);
        }
      });
      return res;
    };
    Vue.provide(descriptions_constants_index.descriptionsKey, props2);
    var renderBody = function renderBody() {
      var tableClass = ["".concat(COMPONENT_NAME.value, "__body"), SIZE.value[props2.size], _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__body--fixed"), props2.tableLayout === "fixed"), _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__body--border"), props2.bordered)];
      return Vue.createVNode("table", {
        "class": tableClass
      }, [Vue.createVNode("tbody", null, [getRows().map(function (row) {
        return Vue.createVNode(descriptions_descriptionsRow["default"], {
          "item-type": itemsType.value,
          "row": row
        }, null);
      })])]);
    };
    var renderHeader = function renderHeader() {
      var title = renderTNodeJSX("title");
      return title ? Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__header")
      }, [title]) : "";
    };
    return function () {
      return Vue.createVNode("div", {
        "class": COMPONENT_NAME.value
      }, [renderHeader(), renderBody()]);
    };
  }
});

exports["default"] = _Descriptions;
