// tdesign-vue-next/cjs/rate/rate.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index = require('../_chunks/dep-1862d466.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-0989c3be.js');
var index$1 = require('../_chunks/dep-ce9c15f5.js');
require('@babel/runtime/helpers/defineProperty');
var rate_props = require('./props.js');
var tooltip_index = require('../tooltip/index.js');
var isArray = require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-38acad3d.js');
require('../_chunks/dep-0238ed9a.js');
require('../_chunks/dep-4ba41627.js');
require('../popup/container.js');
require('../_chunks/dep-e2122882.js');
require('../_chunks/dep-9bee89d3.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-79476fcb.js');
require('../_chunks/dep-e3edff33.js');
require('../tooltip/utils/index.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

var _Rate = Vue.defineComponent({
  name: "TRate",
  props: rate_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var renderTNodeJSX = index.useTNodeJSX();
    var _useConfig = configProvider_hooks_useConfig.useConfig("rate"),
      globalConfig = _useConfig.globalConfig;
    var activeColor = Vue.computed(function () {
      return isArray.isArray(props2.color) ? props2.color[0] : props2.color;
    });
    var defaultColor = Vue.computed(function () {
      return isArray.isArray(props2.color) ? props2.color[1] : "var(--td-bg-color-component)";
    });
    var _toRefs = Vue.toRefs(props2),
      inputValue = _toRefs.value,
      modelValue = _toRefs.modelValue;
    var _useVModel = index$1.useVModel(inputValue, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      starValue = _useVModel2[0],
      setStarValue = _useVModel2[1];
    var hoverValue = Vue.ref(void 0);
    var root = Vue.ref();
    var displayValue = Vue.computed(function () {
      return Number(hoverValue.value || starValue.value);
    });
    var displayTexts = Vue.computed(function () {
      return props2.texts.length === 0 ? globalConfig.value.rateText : props2.texts;
    });
    var displayText = Vue.computed(function () {
      return displayTexts.value[Math.ceil(displayValue.value - 1)];
    });
    var RateIcon = function RateIcon(iconProps) {
      if (slots.icon !== void 0) {
        return renderTNodeJSX("icon", {
          params: iconProps
        });
      }
      return Vue.createVNode(tdesignIconsVueNext.StarFilledIcon, iconProps, null);
    };
    var getStarValue = function getStarValue(event, index) {
      if (props2.allowHalf) {
        var _root$value$getBoundi = root.value.getBoundingClientRect(),
          left = _root$value$getBoundi.left;
        var firstStar = root.value.firstChild.nextSibling;
        var _firstStar$getBoundin = firstStar.getBoundingClientRect(),
          width = _firstStar$getBoundin.width;
        var clientX = event.clientX;
        var starMiddle = width * (index - 0.5) + props2.gap * (index - 1);
        if (clientX - left >= starMiddle) return index;
        if (clientX - left < starMiddle) return index - 0.5;
      }
      return index;
    };
    var mouseEnterHandler = function mouseEnterHandler(event, index) {
      if (props2.disabled) return;
      hoverValue.value = getStarValue(event, index);
    };
    var mouseLeaveHandler = function mouseLeaveHandler() {
      if (props2.disabled) return;
      hoverValue.value = void 0;
    };
    var clickHandler = function clickHandler(event, index) {
      if (props2.disabled) return;
      var value = getStarValue(event, index);
      if (props2.clearable && value === starValue.value) {
        hoverValue.value = void 0;
        setStarValue(0);
      } else {
        setStarValue(value);
      }
    };
    var getStarCls = function getStarCls(index) {
      if (props2.allowHalf && index + 0.5 === displayValue.value) return "".concat(classPrefix.value, "-rate__item--half");
      if (index >= displayValue.value) return "";
      if (index < displayValue.value) return "".concat(classPrefix.value, "-rate__item--full");
    };
    var _useConfig2 = configProvider_hooks_useConfig.useConfig("classPrefix"),
      classPrefix = _useConfig2.classPrefix;
    return function () {
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-rate"),
        "onMouseleave": mouseLeaveHandler
      }, [Vue.createVNode("ul", {
        "class": "".concat(classPrefix.value, "-rate__list"),
        "style": {
          gap: "".concat(props2.gap, "px")
        },
        "ref": root
      }, [_toConsumableArray__default["default"](Array(Number(props2.count))).map(function (_, index) {
        return Vue.createVNode("li", {
          "key": index,
          "class": ["".concat(classPrefix.value, "-rate__item"), getStarCls(index)],
          "onClick": function onClick(event) {
            return clickHandler(event, index + 1);
          },
          "onMousemove": function onMousemove(event) {
            return mouseEnterHandler(event, index + 1);
          }
        }, [props2.showText ? Vue.createVNode(tooltip_index.Tooltip, {
          "key": index,
          "content": displayText.value
        }, {
          "default": function _default() {
            return [Vue.createVNode("div", {
              "class": "".concat(classPrefix.value, "-rate__star-top")
            }, [Vue.createVNode(RateIcon, {
              "size": props2.size,
              "color": activeColor.value
            }, null)]), Vue.createVNode("div", {
              "class": "".concat(classPrefix.value, "-rate__star-bottom")
            }, [Vue.createVNode(RateIcon, {
              "size": props2.size,
              "color": defaultColor.value
            }, null)])];
          }
        }) : Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
          "class": "".concat(classPrefix.value, "-rate__star-top")
        }, [Vue.createVNode(RateIcon, {
          "size": props2.size,
          "color": activeColor.value
        }, null)]), Vue.createVNode("div", {
          "class": "".concat(classPrefix.value, "-rate__star-bottom")
        }, [Vue.createVNode(RateIcon, {
          "size": props2.size,
          "color": defaultColor.value
        }, null)])])]);
      })]), props2.showText && Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-rate__text")
      }, [displayText.value])]);
    };
  }
});

exports["default"] = _Rate;
