// tdesign-vue-next/esm/qrcode/components/qrcode-status.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, createVNode, computed, Fragment } from 'vue';
import { QRCodeStatusProps } from './props.js';
import { RefreshIcon, CheckCircleFilledIcon } from 'tdesign-icons-vue-next';
import { Loading } from '../../loading/index.js';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../../_chunks/dep-c68ea098.js';
import { u as usePrefixClass } from '../../_chunks/dep-e8dd47a9.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-91fc762d.js';
import '@babel/runtime/helpers/defineProperty';
import '../../loading/directive.js';
import '../../loading/plugin.js';
import '../../_chunks/dep-2a7145b5.js';
import '../../loading/icon/gradient.js';
import '../../_chunks/dep-ef7a41ce.js';
import '../../_chunks/dep-fbf70ecb.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../_chunks/dep-d518fdfb.js';
import '../../_chunks/dep-7bdccf65.js';
import '../../_chunks/dep-8d4d971b.js';
import '../../_chunks/dep-98d89c71.js';
import '../../config-provider/hooks/useConfig.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-509ddbe3.js';
import '../../_chunks/dep-f0f392fb.js';
import 'dayjs';
import '../../loading/props.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../../_chunks/dep-ae3e94b6.js';
import 'tdesign-vue-next/esm/common/style/web/components/loading/_index.less';

var QRcodeStatus = defineComponent({
  name: "QRCodeStatus",
  props: QRCodeStatusProps,
  setup: function setup(props) {
    var _props$locale3;
    var classPrefix = usePrefixClass();
    var defaultSpin = createVNode(Loading, {
      "size": "32px"
    }, null);
    var defaultExpiredNode = computed(function () {
      var _props$locale, _props$locale2;
      return createVNode(Fragment, null, [createVNode("p", {
        "class": "".concat(classPrefix.value, "-expired__text")
      }, [(_props$locale = props.locale) === null || _props$locale === void 0 ? void 0 : _props$locale.expiredText]), (props === null || props === void 0 ? void 0 : props.onRefresh) && createVNode("p", {
        "class": "".concat(classPrefix.value, "-expired__button"),
        "onClick": props === null || props === void 0 ? void 0 : props.onRefresh
      }, [createVNode(RefreshIcon, {
        "size": "16"
      }, null), (_props$locale2 = props.locale) === null || _props$locale2 === void 0 ? void 0 : _props$locale2.refreshText])]);
    });
    var defaultScannedNode = createVNode("p", {
      "class": "".concat(classPrefix.value, "-scanned")
    }, [createVNode(CheckCircleFilledIcon, {
      "size": "16",
      "class": "".concat(classPrefix.value, "-scanned__icon")
    }, null), (_props$locale3 = props.locale) === null || _props$locale3 === void 0 ? void 0 : _props$locale3.scannedText]);
    var defaultNodes = computed(function () {
      return {
        expired: defaultExpiredNode.value,
        loading: defaultSpin,
        scanned: defaultScannedNode,
        active: null
      };
    });
    var renderStatus = function renderStatus(info) {
      return defaultNodes.value[info.status];
    };
    return function () {
      return createVNode(Fragment, null, [props.statusRender || renderStatus({
        status: props.status,
        onRefresh: props.onRefresh
      })]);
    };
  }
});

export { QRcodeStatus as default };
