// tdesign-vue-next/cjs/anchor/anchor-item.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var anchor_utils_index = require('./utils/index.js');
var anchor_anchorItemProps = require('./anchor-item-props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$1 = require('../_chunks/dep-38acad3d.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
var anchor_constants_index = require('./constants/index.js');
var isString = require('../_chunks/dep-631750cd.js');
var isFunction = require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-4a7162e7.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var localProps = _objectSpread(_objectSpread({}, anchor_anchorItemProps["default"]), {}, {
  href: {
    type: String,
    required: true,
    validator: function validator(v) {
      return anchor_utils_index.ANCHOR_SHARP_REGEXP.test(v);
    }
  }
});
var _AnchorItem = Vue.defineComponent({
  name: "TAnchorItem",
  inject: {
    tAnchor: {
      "default": void 0
    }
  },
  props: localProps,
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var anchor = Vue.inject(anchor_constants_index.AnchorInjectionKey, void 0);
    var CLASSNAME_PREFIX = index.usePrefixClass("anchor__item");
    var _useCommonClassName = index$1.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var register = function register() {
      anchor.registerLink(props2.href);
    };
    var unregister = function unregister(hrefVal) {
      var val = hrefVal || props2.href;
      if (!val) return;
      anchor.unregisterLink(val);
    };
    var handleClick = function handleClick(e) {
      var href = props2.href,
        title = props2.title;
      if (!props2.customScroll) anchor.handleScrollTo(href);
      anchor.handleLinkClick({
        href: href,
        title: isString.isString(title) ? title : void 0,
        e: e
      });
    };
    var renderTitle = function renderTitle() {
      var title = props2.title;
      var titleSlot = slots.title;
      var titleVal;
      if (isString.isString(title)) {
        titleVal = title;
      } else if (isFunction.isFunction(title)) {
        titleVal = title(Vue.h);
      } else if (titleSlot) {
        titleVal = titleSlot(null);
      }
      return titleVal;
    };
    Vue.watch(function () {
      return props2.href;
    }, function (newVal, oldVal) {
      unregister(oldVal);
      register();
    }, {
      immediate: true
    });
    Vue.onMounted(function () {
      register();
    });
    Vue.onUnmounted(function () {
      unregister();
    });
    return function () {
      var href = props2.href,
        target = props2.target;
      var children = slots["default"],
        titleSlot = slots.title;
      var title = renderTitle();
      var titleAttr = isString.isString(title) ? title : null;
      var active = anchor.active === href;
      var wrapperClass = _defineProperty__default["default"](_defineProperty__default["default"]({}, CLASSNAME_PREFIX.value, true), STATUS.value.active, active);
      var titleClass = _defineProperty__default["default"]({}, "".concat(CLASSNAME_PREFIX.value, "-link"), true);
      return Vue.createVNode("div", {
        "class": wrapperClass
      }, [Vue.createVNode("a", {
        "href": href,
        "title": titleAttr,
        "class": titleClass,
        "target": target,
        "onClick": handleClick
      }, [titleSlot ? titleSlot(null) : title]), children && children(null)]);
    };
  }
});

exports["default"] = _AnchorItem;
