// tdesign-vue-next/cjs/image-viewer/image-viewer.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var renderTnode = require('../_chunks/dep-a6551cc4.js');
var index$1 = require('../_chunks/dep-1862d466.js');
var index = require('../_chunks/dep-0238ed9a.js');
var index$2 = require('../_chunks/dep-0afce170.js');
var index$4 = require('../_chunks/dep-0989c3be.js');
var loading_loading = require('../_chunks/dep-d632b788.js');
var index$3 = require('../_chunks/dep-ce9c15f5.js');
var isArray = require('../_chunks/dep-4cb26289.js');
var isString = require('../_chunks/dep-631750cd.js');
var image_index = require('../image/index.js');
var imageViewer_base_ImageItem = require('./base/ImageItem.js');
var imageViewer_base_ImageModalIcon = require('./base/ImageModalIcon.js');
var imageViewer_base_ImageViewerModal = require('./base/ImageViewerModal.js');
var imageViewer_base_ImageViewerUtils = require('./base/ImageViewerUtils.js');
var imageViewer_constants_index = require('./constants/index.js');
var imageViewer_hooks_index = require('./hooks/index.js');
var imageViewer_props = require('./props.js');
var imageViewer_utils_index = require('./utils/index.js');
var isNumber = require('../_chunks/dep-44195fb2.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../loading/icon/gradient.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-38acad3d.js');
require('../loading/props.js');
require('../image/image.js');
require('../_chunks/dep-b9a88912.js');
require('../_chunks/dep-03ea53c4.js');
require('../_chunks/dep-50d6b5ee.js');
require('../_chunks/dep-692bdebe.js');
require('../image/props.js');
require('../space/index.js');
require('../space/space.js');
require('../space/props.js');
require('../_chunks/dep-0cda9fe9.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-e3edff33.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../dialog/index.js');
require('../dialog/dialog.js');
require('../dialog/props.js');
require('../_chunks/dep-6a57aec8.js');
require('../button/index.js');
require('../button/button.js');
require('../loading/index.js');
require('../_chunks/dep-34cd76d4.js');
require('../loading/plugin.js');
require('../_chunks/dep-43dcd2db.js');
require('../_chunks/dep-392dde71.js');
require('../_chunks/dep-02ea4f3c.js');
require('../_chunks/dep-5103206f.js');
require('../_chunks/dep-7707ffe2.js');
require('../_chunks/dep-74a5bebe.js');
require('../_chunks/dep-42740075.js');
require('../_chunks/dep-9ac5c6ca.js');
require('../button/props.js');
require('../_chunks/dep-362c5212.js');
require('../_chunks/dep-74b143e0.js');
require('../_chunks/dep-719a33c8.js');
require('../dialog/hooks/useSameTarget.js');
require('../_chunks/dep-dc6f0c3f.js');
require('../dialog/utils/index.js');
require('../dialog/dialog-card.js');
require('../dialog/dialog-card-props.js');
require('../_chunks/dep-c26a7d50.js');
require('../dialog/hooks/useAction.js');
require('../dialog/plugin.js');
require('../tooltip/index.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../popup/container.js');
require('../_chunks/dep-e2122882.js');
require('../_chunks/dep-9bee89d3.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-79476fcb.js');
require('../tooltip/utils/index.js');
require('../_chunks/dep-73734701.js');
require('../_chunks/dep-6bac596c.js');
require('../_chunks/dep-27729d32.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var isSameOrigin = function isSameOrigin(url) {
  try {
    var imgUrl = new URL(url, window.location.href);
    return imgUrl.origin === window.location.origin;
  } catch (_unused) {
    return false;
  }
};
var directDownload = function directDownload(imgSrc, name) {
  var a = document.createElement("a");
  a.download = name;
  a.href = imgSrc;
  a.click();
  a.remove();
};
var fileDownload = function fileDownload(obj, name) {
  var url = URL.createObjectURL(obj);
  var a = document.createElement("a");
  a.href = url;
  a.download = name;
  a.click();
  a.remove();
  URL.revokeObjectURL(url);
};
var canvasDownload = function canvasDownload(imgSrc, name) {
  var image = new Image();
  image.setAttribute("crossOrigin", "anonymous");
  image.onload = function () {
    var _name$split$pop;
    var canvas = document.createElement("canvas");
    canvas.width = image.width;
    canvas.height = image.height;
    var context = canvas.getContext("2d");
    context.drawImage(image, 0, 0, image.width, image.height);
    var extension = ((_name$split$pop = name.split(".").pop()) === null || _name$split$pop === void 0 ? void 0 : _name$split$pop.toLowerCase()) || "png";
    var mimeType = "image/".concat(extension === "jpg" ? "jpeg" : extension);
    canvas.toBlob(function (blob) {
      fileDownload(blob, name);
    }, mimeType);
  };
  image.src = imgSrc;
};
var downloadImage = function downloadImage(imgSrc) {
  var _imgSrc$split, _imgSrc$split$split, _imgSrc$split$split$s;
  var randomName = Math.random().toString(32).slice(2);
  if (imgSrc instanceof File) {
    fileDownload(imgSrc, imgSrc.name);
    return;
  }
  var name = (imgSrc === null || imgSrc === void 0 || (_imgSrc$split = imgSrc.split) === null || _imgSrc$split === void 0 || (_imgSrc$split = _imgSrc$split.call(imgSrc, "?")) === null || _imgSrc$split === void 0 || (_imgSrc$split = _imgSrc$split[0]) === null || _imgSrc$split === void 0 || (_imgSrc$split$split = _imgSrc$split.split) === null || _imgSrc$split$split === void 0 || (_imgSrc$split$split = _imgSrc$split$split.call(_imgSrc$split, "#")) === null || _imgSrc$split$split === void 0 || (_imgSrc$split$split = _imgSrc$split$split[0]) === null || _imgSrc$split$split === void 0 || (_imgSrc$split$split$s = _imgSrc$split$split.split) === null || _imgSrc$split$split$s === void 0 ? void 0 : _imgSrc$split$split$s.call(_imgSrc$split$split, "/").pop()) || randomName;
  if (isSameOrigin(imgSrc)) {
    directDownload(imgSrc, name);
  } else {
    canvasDownload(imgSrc, name);
  }
};
var isImageInfo = function isImageInfo(image) {
  return !!image && !isString.isString(image) && !(image instanceof File);
};
var formatImages = function formatImages(images) {
  if (!isArray.isArray(images)) return [];
  return images.map(function (item) {
    if (isImageInfo(item)) {
      return _objectSpread({
        download: true,
        thumbnail: item.mainImage
      }, item);
    }
    return {
      mainImage: item,
      thumbnail: item,
      download: true
    };
  });
};

var _ImageViewer = Vue.defineComponent({
  name: "TImageViewer",
  props: imageViewer_props["default"],
  setup: function setup(props2) {
    var _props2$defaultIndex;
    var classPrefix = index.usePrefixClass();
    var COMPONENT_NAME = index.usePrefixClass("image-viewer");
    var renderTNodeJSX = index$1.useTNodeJSX();
    var _useConfig = configProvider_hooks_useConfig.useConfig("imageViewer"),
      globalConfig = _useConfig.globalConfig;
    var isExpand = Vue.ref(true);
    var showOverlayValue = Vue.computed(function () {
      return imageViewer_utils_index.getOverlay(props2);
    });
    var _toRefs = Vue.toRefs(props2),
      index$5 = _toRefs.index,
      visible = _toRefs.visible,
      modelValue = _toRefs.modelValue,
      imageReferrerpolicy = _toRefs.imageReferrerpolicy;
    var _useDefaultValue = index$2.useDefaultValue(index$5, (_props2$defaultIndex = props2.defaultIndex) !== null && _props2$defaultIndex !== void 0 ? _props2$defaultIndex : 0, props2.onIndexChange, "index"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
      indexValue = _useDefaultValue2[0],
      setIndexValue = _useDefaultValue2[1];
    var _useVModel = index$3.useVModel(visible, modelValue, props2.defaultVisible, function () {}, "visible"),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      visibleValue = _useVModel2[0],
      setVisibleValue = _useVModel2[1];
    var animationEnd = Vue.ref(true);
    var animationTimer = Vue.ref();
    var teleportElement = loading_loading.useTeleport(function () {
      return props2.attach;
    });
    var wrapClass = Vue.computed(function () {
      return [COMPONENT_NAME.value, "".concat(COMPONENT_NAME.value, "-preview-image"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-is-hide"), !visibleValue.value)];
    });
    var headerClass = Vue.computed(function () {
      return ["".concat(classPrefix.value, "-image-viewer__modal-header"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-is-show"), isExpand.value)];
    });
    var zIndexValue = Vue.computed(function () {
      var _props2$zIndex;
      return (_props2$zIndex = props2.zIndex) !== null && _props2$zIndex !== void 0 ? _props2$zIndex : 3e3;
    });
    var toggleExpand = function toggleExpand() {
      isExpand.value = !isExpand.value;
    };
    var _useMirror = imageViewer_hooks_index.useMirror(),
      mirror = _useMirror.mirror,
      onMirror = _useMirror.onMirror,
      resetMirror = _useMirror.resetMirror;
    var _useScale = imageViewer_hooks_index.useScale(props2.imageScale),
      scale = _useScale.scale,
      onZoomIn = _useScale.onZoomIn,
      onZoomOut = _useScale.onZoomOut,
      resetScale = _useScale.resetScale;
    var _useRotate = imageViewer_hooks_index.useRotate(),
      rotate = _useRotate.rotate,
      onRotate = _useRotate.onRotate,
      resetRotate = _useRotate.resetRotate;
    var onRest = function onRest() {
      resetMirror();
      resetScale();
      resetRotate();
    };
    var images = Vue.computed(function () {
      return formatImages(props2.images);
    });
    var currentImage = Vue.computed(function () {
      var _images$value$indexVa;
      return (_images$value$indexVa = images.value[indexValue.value]) !== null && _images$value$indexVa !== void 0 ? _images$value$indexVa : {
        mainImage: ""
      };
    });
    var _usePopupManager = index$4.usePopupManager("dialog", {
        visible: visibleValue
      }),
      isTopInteractivePopup = _usePopupManager.isTopInteractivePopup;
    var prevImage = function prevImage() {
      var newIndex = indexValue.value - 1;
      onRest();
      setIndexValue(newIndex < 0 ? 0 : newIndex, {
        trigger: "prev"
      });
    };
    var nextImage = function nextImage() {
      var newIndex = indexValue.value + 1;
      onRest();
      setIndexValue(newIndex >= images.value.length ? indexValue.value : newIndex, {
        trigger: "next"
      });
    };
    var onImgClick = function onImgClick(i) {
      setIndexValue(i, {
        trigger: "current"
      });
    };
    var onDownloadClick = function onDownloadClick(url) {
      props2.onDownload ? props2.onDownload(url) : downloadImage(url);
    };
    var openHandler = function openHandler(index2) {
      if (isNumber.isNumber(index2)) {
        onImgClick(index2);
      }
      setVisibleValue(true);
    };
    var onClose = function onClose(ctx) {
      var _props2$onClose;
      setVisibleValue(false);
      (_props2$onClose = props2.onClose) === null || _props2$onClose === void 0 || _props2$onClose.call(props2, ctx);
    };
    var closeBtnAction = function closeBtnAction(e) {
      onClose({
        e: e,
        trigger: "close-btn"
      });
    };
    var clickOverlayHandler = function clickOverlayHandler(e) {
      if (props2.closeOnOverlay) {
        onClose({
          e: e,
          trigger: "overlay"
        });
      }
    };
    var keydownHandler = function keydownHandler(e) {
      e.stopPropagation();
      switch (e.code) {
        case imageViewer_constants_index.EVENT_CODE.left:
          prevImage();
          break;
        case imageViewer_constants_index.EVENT_CODE.right:
          nextImage();
          break;
        case imageViewer_constants_index.EVENT_CODE.up:
          onZoomIn();
          break;
        case imageViewer_constants_index.EVENT_CODE.down:
          onZoomOut();
          break;
        case imageViewer_constants_index.EVENT_CODE.esc:
          if (props2.closeOnEscKeydown && isTopInteractivePopup()) {
            onClose({
              e: e,
              trigger: "esc"
            });
          }
          break;
      }
    };
    var divRef = Vue.ref();
    Vue.watch(function () {
      return visibleValue.value;
    }, function (val) {
      clearTimeout(animationTimer.value);
      if (val) {
        animationEnd.value = false;
        Vue.nextTick().then(function () {
          var _divRef$value, _divRef$value$focus;
          (_divRef$value = divRef.value) === null || _divRef$value === void 0 || (_divRef$value$focus = _divRef$value.focus) === null || _divRef$value$focus === void 0 || _divRef$value$focus.call(_divRef$value);
        });
        onRest();
      } else {
        animationTimer.value = setTimeout(function () {
          animationEnd.value = true;
        }, 200);
      }
    });
    Vue.onBeforeUnmount(function () {
      clearTimeout(animationTimer.value);
    });
    var onWheel = function onWheel(e) {
      e.preventDefault();
      var deltaY = e.deltaY;
      deltaY > 0 ? onZoomOut() : onZoomIn();
    };
    var transStyle = Vue.computed(function () {
      return {
        transform: "translateX(calc(-".concat(indexValue.value, " * (40px / 9 * 16 + 4px)))")
      };
    });
    var renderHeader = function renderHeader() {
      return Vue.createVNode("div", {
        "class": headerClass.value
      }, [Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
        "icon": function icon() {
          return Vue.createVNode(tdesignIconsVueNext.ChevronDownIcon, null, null);
        },
        "class": "".concat(COMPONENT_NAME.value, "__header-pre-bt"),
        "onClick": toggleExpand
      }, null), Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__header-prev")
      }, [Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__header-trans"),
        "style": transStyle.value
      }, [images.value.map(function (image, index2) {
        return Vue.createVNode("div", {
          "key": index2,
          "class": ["".concat(COMPONENT_NAME.value, "__header-box"), _defineProperty__default["default"]({}, "".concat(classPrefix.value, "-is-active"), index2 === indexValue.value)]
        }, [Vue.createVNode(image_index.Image, {
          "src": image.thumbnail || image.mainImage,
          "error": "",
          "class": "".concat(COMPONENT_NAME.value, "__header-img"),
          "onClick": function onClick() {
            return onImgClick(index2);
          }
        }, null)]);
      })])])]);
    };
    var renderTitle = function renderTitle() {
      var titleContent = renderTNodeJSX("title");
      return Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__modal-index")
      }, [titleContent ? titleContent : "".concat(indexValue.value + 1, "/").concat(images.value.length)]);
    };
    var renderNavigationArrow = function renderNavigationArrow(type) {
      var rotateDeg = type === "prev" ? 0 : 180;
      var _icon = renderTNodeJSX("navigationArrow", Vue.createVNode(tdesignIconsVueNext.ChevronLeftIcon, {
        "style": {
          transform: "rotate(".concat(rotateDeg, "deg)")
        },
        "size": "24px"
      }, null));
      return Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
        "class": "".concat(COMPONENT_NAME.value, "__modal-").concat(type, "-bt"),
        "onClick": type === "prev" ? prevImage : nextImage,
        "icon": function icon() {
          return _icon;
        }
      }, null);
    };
    var renderCloseBtn = function renderCloseBtn() {
      if (props2.closeBtn === false) {
        return;
      }
      return Vue.createVNode("div", {
        "class": ["".concat(COMPONENT_NAME.value, "__modal-icon"), "".concat(COMPONENT_NAME.value, "__modal-close-bt")],
        "onClick": closeBtnAction
      }, [renderTNodeJSX("closeBtn", Vue.createVNode(tdesignIconsVueNext.CloseIcon, {
        "size": "24px"
      }, null))]);
    };
    var renderDefaultTrigger = function renderDefaultTrigger() {
      var firstImage = images.value[0] || "";
      var imageSrc = typeof firstImage === "string" ? firstImage : firstImage.mainImage || firstImage.thumbnail;
      return Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__trigger")
      }, [Vue.createVNode(image_index.Image, {
        "src": imageSrc,
        "alt": "preview",
        "fit": "contain",
        "class": "".concat(COMPONENT_NAME.value, "__trigger-img")
      }, null), Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__trigger--hover"),
        "onClick": function onClick() {
          return openHandler();
        }
      }, [Vue.createVNode("span", null, [Vue.createVNode(tdesignIconsVueNext.BrowseIcon, {
        "size": "1.4em",
        "class": "".concat(COMPONENT_NAME.value, "__trigger-icon")
      }, null), globalConfig.value.previewText])])]);
    };
    return function () {
      if (props2.mode === "modeless") {
        return Vue.createVNode(Vue.Fragment, null, [renderTnode.isPropsUsed("trigger") ? renderTNodeJSX("trigger", {
          params: {
            open: openHandler
          }
        }) : renderDefaultTrigger(), Vue.createVNode(imageViewer_base_ImageViewerModal["default"], {
          "zIndex": zIndexValue.value,
          "visible": visibleValue.value,
          "index": indexValue.value,
          "images": images.value,
          "scale": scale.value,
          "rotate": rotate.value,
          "mirror": mirror.value,
          "currentImage": currentImage.value,
          "onRotate": onRotate,
          "onZoomIn": onZoomIn,
          "onZoomOut": onZoomOut,
          "onMirror": onMirror,
          "onReset": onRest,
          "onClose": onClose,
          "onDownload": onDownloadClick,
          "draggable": props2.draggable,
          "showOverlay": showOverlayValue.value,
          "title": renderTitle,
          "imageReferrerpolicy": imageReferrerpolicy.value
        }, null)]);
      }
      return Vue.createVNode(Vue.Fragment, null, [renderTnode.isPropsUsed("trigger") ? renderTNodeJSX("trigger", {
        params: {
          open: openHandler
        }
      }) : renderDefaultTrigger(), Vue.createVNode(Vue.Teleport, {
        "disabled": !props2.attach || !teleportElement.value,
        "to": teleportElement.value
      }, {
        "default": function _default() {
          return [Vue.createVNode(Vue.Transition, null, {
            "default": function _default() {
              return [(visibleValue.value || !animationEnd.value) && Vue.withDirectives(Vue.createVNode("div", {
                "ref": divRef,
                "class": wrapClass.value,
                "style": {
                  zIndex: zIndexValue.value
                },
                "onWheel": onWheel,
                "tabindex": -1,
                "onKeydown": keydownHandler
              }, [!!showOverlayValue.value && Vue.createVNode("div", {
                "class": "".concat(COMPONENT_NAME.value, "__modal-mask"),
                "onClick": clickOverlayHandler
              }, null), images.value.length > 1 && Vue.createVNode(Vue.Fragment, null, [renderHeader(), renderTitle(), renderNavigationArrow("prev"), renderNavigationArrow("next")]), renderCloseBtn(), Vue.createVNode(imageViewer_base_ImageViewerUtils["default"], {
                "zIndex": zIndexValue.value + 1,
                "onZoomIn": onZoomIn,
                "onZoomOut": onZoomOut,
                "onMirror": onMirror,
                "onReset": onRest,
                "onRotate": onRotate,
                "onDownload": onDownloadClick,
                "scale": scale.value,
                "currentImage": currentImage.value
              }, null), Vue.createVNode(imageViewer_base_ImageItem["default"], {
                "scale": scale.value,
                "rotate": rotate.value,
                "mirror": mirror.value,
                "src": currentImage.value.mainImage,
                "placementSrc": currentImage.value.thumbnail,
                "isSvg": currentImage.value.isSvg,
                "imageReferrerpolicy": imageReferrerpolicy.value
              }, null)]), [[Vue.vShow, visibleValue.value]])];
            }
          })];
        }
      })]);
    };
  }
});

exports["default"] = _ImageViewer;
