// tdesign-vue-next/es/breadcrumb/breadcrumb.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, toRefs, createVNode, provide, reactive, mergeProps } from 'vue';
import { _ as _typeof } from '../_chunks/dep-e1c62679.mjs';
import props from './props.mjs';
import _BreadcrumbItem from './breadcrumb-item.mjs';
import { u as usePrefixClass, a as useTNodeJSX, i as useGlobalIcon } from '../_chunks/dep-fbeb2963.mjs';
import './hooks/index.mjs';
import { ChevronRightIcon, EllipsisIcon } from 'tdesign-icons-vue-next';
import { useBreadcrumbOptions } from './hooks/useBreadcrumbOptions.mjs';
import { useEllipsis } from './hooks/useEllipsis.mjs';
import '../_chunks/dep-ba214d75.mjs';
import './breadcrumb-item-props.mjs';
import '../tooltip/index.mjs';
import '../tooltip/tooltip.mjs';
import '../_chunks/dep-f558ebb8.mjs';
import '../_chunks/dep-b0b422fe.mjs';
import 'lodash-es';
import '../tooltip/props.mjs';
import '../popup/props.mjs';
import '../popup/index.mjs';
import '../popup/popup.mjs';
import '@popperjs/core';
import '../_chunks/dep-feed89df.mjs';
import '../_chunks/dep-5b510fcd.mjs';
import '../popup/container.mjs';
import '../_chunks/dep-47c3cae9.mjs';
import './style/css.mjs';
import '../config-provider/hooks/useConfig.mjs';
import '../config-provider/utils/context.mjs';
import '../_chunks/dep-e0ab6d55.mjs';
import '../_chunks/dep-75f265c4.mjs';
import '../_chunks/dep-67ffcfeb.mjs';
import '../_chunks/dep-edab0e6a.mjs';
import '../_chunks/dep-76dddedf.mjs';
import '../_chunks/dep-d92220c4.mjs';
import '../_chunks/dep-466a8f31.mjs';
import '../_chunks/dep-e66c0337.mjs';
import '../tooltip/utils/index.mjs';
import '../_chunks/dep-739ea4cd.mjs';

var _Breadcrumb = defineComponent({
  name: "TBreadcrumb",
  props: props,
  setup: function setup(props2) {
    var _toRefs = toRefs(props2),
      theme = _toRefs.theme,
      maxItemWidth = _toRefs.maxItemWidth;
    var COMPONENT_NAME = usePrefixClass("breadcrumb");
    var renderTNodeJSX = useTNodeJSX();
    var _useGlobalIcon = useGlobalIcon({
        ChevronRightIcon: ChevronRightIcon,
        EllipsisIcon: EllipsisIcon
      }),
      ChevronRightIcon$1 = _useGlobalIcon.ChevronRightIcon,
      EllipsisIcon$1 = _useGlobalIcon.EllipsisIcon;
    var separatorContent = renderTNodeJSX("separator");
    var separator = separatorContent || createVNode(ChevronRightIcon$1, null, null);
    provide("tBreadcrumb", reactive({
      separator: separator,
      theme: theme,
      maxItemWidth: maxItemWidth
    }));
    return function () {
      var _useBreadcrumbOptions = useBreadcrumbOptions(props2),
        breadcrumbOptions = _useBreadcrumbOptions.breadcrumbOptions;
      var ellipsisItems = breadcrumbOptions.value.slice(props2.itemsBeforeCollapse, breadcrumbOptions.value.length - props2.itemsAfterCollapse);
      var ellipsisSlot = renderTNodeJSX("ellipsis", {
        params: {
          items: ellipsisItems,
          separator: separator
        }
      });
      var ellipsisContent = ellipsisSlot || createVNode(EllipsisIcon$1, null, null);
      var _useEllipsis = useEllipsis(props2, breadcrumbOptions, ellipsisContent),
        getDisplayItems = _useEllipsis.getDisplayItems;
      var items = getDisplayItems.value;
      var content = items.map(function (item, index) {
        if (_typeof(item) === "object" && "content" in item) {
          return createVNode(_BreadcrumbItem, mergeProps({
            "key": index
          }, item), {
            "default": function _default() {
              return [item.content];
            }
          });
        }
        return item;
      });
      return createVNode("div", {
        "class": COMPONENT_NAME.value
      }, [content]);
    };
  }
});

export { _Breadcrumb as default };
