// tdesign-vue-next/es/date-picker/hooks/useDisableDate.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { isObject } from 'lodash-es';
import { d as dayjs } from '../../_chunks/dep-67ffcfeb.mjs';
import { b as isEnabledDate } from '../../_chunks/dep-66050381.mjs';
import '../../_chunks/dep-e1c62679.mjs';
import '../../_chunks/dep-5b510fcd.mjs';
import '../../_chunks/dep-f558ebb8.mjs';
import '../../_chunks/dep-b0b422fe.mjs';
import '../../_chunks/dep-ba214d75.mjs';
import '../../_chunks/dep-5d101f94.mjs';
import '../../_chunks/dep-7e7f7e26.mjs';
import '../../_chunks/dep-8c4f181a.mjs';
import '../../_chunks/dep-739ea4cd.mjs';
import '../../_chunks/dep-d92220c4.mjs';

function useDisableDate(props) {
  var _disableDate = props.disableDate,
    format = props.format,
    mode = props.mode,
    start = props.start,
    end = props.end;
  return {
    disableDate: function disableDate(value) {
      return !isEnabledDate({
        disableDate: _disableDate,
        format: format,
        mode: mode,
        value: value
      });
    },
    minDate: isObject(_disableDate) && "before" in _disableDate ? new Date(dayjs(_disableDate.before).startOf("day").format()) : start,
    maxDate: isObject(_disableDate) && "after" in _disableDate ? new Date(dayjs(_disableDate.after).endOf("day").format()) : end
  };
}

export { useDisableDate };
