// tdesign-vue-next/lib/layout/aside.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, inject, onMounted, onUnmounted, createVNode } from 'vue';
import props from './aside-props.js';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../_chunks/dep-c68ea098.js';
import { u as useTNodeJSX } from '../_chunks/dep-7bdccf65.js';
import { u as usePrefixClass } from '../_chunks/dep-e8dd47a9.js';
import '@babel/runtime/helpers/slicedToArray';
import '../_chunks/dep-91fc762d.js';
import '@babel/runtime/helpers/defineProperty';
import '../_chunks/dep-f0f392fb.js';
import '../_chunks/dep-d518fdfb.js';
import '../_chunks/dep-8d4d971b.js';
import '../config-provider/hooks/useConfig.js';
import '../config-provider/utils/context.js';
import '../_chunks/dep-509ddbe3.js';
import 'dayjs';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';

var _Aside = defineComponent({
  name: "TAside",
  props: props,
  setup: function setup(props2) {
    var _inject = inject("layout", /* @__PURE__ */Object.create(null)),
      hasSide = _inject.hasSide;
    var COMPONENT_NAME = usePrefixClass("layout__sider");
    var renderTNodeJSX = useTNodeJSX();
    if (!hasSide) return;
    onMounted(function () {
      hasSide.value = true;
    });
    onUnmounted(function () {
      hasSide.value = false;
    });
    return function () {
      var styles = props2.width ? {
        width: props2.width
      } : {};
      return createVNode("aside", {
        "class": COMPONENT_NAME.value,
        "style": styles
      }, [renderTNodeJSX("default")]);
    };
  }
});

export { _Aside as default };
