// tdesign-vue-next/cjs/table/hooks/useFilter.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _typeof = require('@babel/runtime/helpers/typeof');
var table_hooks_useClassName = require('./useClassName.js');
var button_index = require('../../button/index.js');
require('@babel/runtime/helpers/toConsumableArray');
require('../../_chunks/dep-35e23c2b.js');
var index = require('../../_chunks/dep-1862d466.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
var index$1 = require('../../_chunks/dep-0afce170.js');
require('../../_chunks/dep-0989c3be.js');
var table_components_filterController = require('../components/filter-controller.js');
var utils = require('../../_chunks/dep-2c1ecdb4.js');
var table_hooks_useTableHeader = require('./useTableHeader.js');
var isFunction = require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-0238ed9a.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../button/button.js');
require('../../loading/index.js');
require('../../_chunks/dep-34cd76d4.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-d632b788.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-4ba41627.js');
require('../../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-3d29e358.js');
require('../../_chunks/dep-6212811d.js');
require('../../_chunks/dep-44195fb2.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-38acad3d.js');
require('../../loading/props.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-43dcd2db.js');
require('../../_chunks/dep-392dde71.js');
require('../../_chunks/dep-02ea4f3c.js');
require('../../_chunks/dep-5103206f.js');
require('../../_chunks/dep-7707ffe2.js');
require('../../_chunks/dep-74a5bebe.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-42740075.js');
require('../../_chunks/dep-9ac5c6ca.js');
require('../../_chunks/dep-e3edff33.js');
require('../../button/props.js');
require('../../_chunks/dep-362c5212.js');
require('../../_chunks/dep-74b143e0.js');
require('../../_chunks/dep-719a33c8.js');
require('tdesign-icons-vue-next');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../_chunks/dep-ce9c15f5.js');
require('../../popup/container.js');
require('../../popup/props.js');
require('../../_chunks/dep-e2122882.js');
require('../../_chunks/dep-9bee89d3.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-ae4040c5.js');
require('../../_chunks/dep-79476fcb.js');
require('../../checkbox/index.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-9dbade68.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-b9a88912.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-a4be1915.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-0cda9fe9.js');
require('../../_chunks/dep-b0334c78.js');
require('../../_chunks/dep-a214821d.js');
require('../../radio/index.js');
require('../../radio/radio.js');
require('../../radio/props.js');
require('../../radio/constants/index.js');
require('../../_chunks/dep-5c79e9ef.js');
require('../../_chunks/dep-9a54a293.js');
require('../../_chunks/dep-7870f4cb.js');
require('../../radio/group.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../radio/radio-group-props.js');
require('../../radio/radio-button.js');
require('../../radio/hooks/useKeyboard.js');
require('../../watermark/hooks/index.js');
require('../../_chunks/dep-27729d32.js');
require('../../_chunks/dep-95977a93.js');
require('../../input/index.js');
require('../../input/input.js');
require('../../input/props.js');
require('../../_chunks/dep-c26a7d50.js');
require('../../input/hooks/useInput.js');
require('../../form/constants/index.js');
require('../../input/hooks/useLengthLimit.js');
require('../../_chunks/dep-692bdebe.js');
require('../../input/hooks/useInputEventHandler.js');
require('../../input/hooks/useInputWidth.js');
require('../../input/input-group.js');
require('../../input/input-group-props.js');
require('../../_chunks/dep-7fb34f13.js');
require('./useMultiHeader.js');
require('../components/ellipsis.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../tooltip/utils/index.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _typeof__default = /*#__PURE__*/_interopDefaultLegacy(_typeof);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function isFilterValueExist(value) {
  var isArrayTrue = value instanceof Array && value.length;
  var isObject = _typeof__default["default"](value) === "object" && !(value instanceof Array);
  var isObjectTrue = value !== null && isObject && Object.keys(value).length;
  return isArrayTrue || isObjectTrue || !["null", "", "undefined"].includes(String(value));
}
function filterEmptyData(data) {
  var newFilterValue = {};
  Object.keys(data).forEach(function (key) {
    var item = data[key];
    if (isFilterValueExist(item)) {
      newFilterValue[key] = item;
    }
  });
  return newFilterValue;
}
function useFilter(props, context) {
  var primaryTableRef = Vue.ref(null);
  var _useConfig = configProvider_hooks_useConfig.useConfig("table", props.locale),
    t = _useConfig.t,
    globalConfig = _useConfig.globalConfig;
  var renderTNode = index.useTNodeDefault();
  var _toRefs = Vue.toRefs(props),
    filterValue = _toRefs.filterValue,
    columns = _toRefs.columns;
  var _useClassName = table_hooks_useClassName["default"](),
    tableFilterClasses = _useClassName.tableFilterClasses,
    isFocusClass = _useClassName.isFocusClass;
  var isTableOverflowHidden = Vue.ref();
  var _useDefaultValue = index$1.useDefaultValue(filterValue, props.defaultFilterValue, props.onFilterChange, "filterValue"),
    _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
    tFilterValue = _useDefaultValue2[0],
    setTFilterValue = _useDefaultValue2[1];
  var innerFilterValue = Vue.ref(tFilterValue.value);
  var hasEmptyCondition = Vue.computed(function () {
    var filterEmpty = filterEmptyData(tFilterValue.value || {});
    return !tFilterValue.value || !Object.keys(filterEmpty).length;
  });
  Vue.watch([tFilterValue], function (_ref) {
    var _ref2 = _slicedToArray__default["default"](_ref, 1),
      val = _ref2[0];
    innerFilterValue.value = val;
  });
  function renderFirstFilterRow() {
    var _props$pagination, _props$data;
    if (hasEmptyCondition.value) return null;
    var defaultNode = Vue.createVNode("div", {
      "class": tableFilterClasses.result
    }, [Vue.createVNode("span", null, [t(globalConfig.value.searchResultText, {
      result: getFilterResultContent(),
      count: ((_props$pagination = props.pagination) === null || _props$pagination === void 0 ? void 0 : _props$pagination.total) || ((_props$data = props.data) === null || _props$data === void 0 ? void 0 : _props$data.length)
    })]), Vue.createVNode(button_index.Button, {
      "theme": "primary",
      "variant": "text",
      "onClick": onResetAll
    }, {
      "default": function _default() {
        return [globalConfig.value.clearFilterResultButtonText];
      }
    })]);
    var filterContent = renderTNode("filterRow");
    if (props.filterRow && !filterContent || props.filterRow === null) return null;
    return Vue.createVNode("div", {
      "class": tableFilterClasses.inner
    }, [filterContent || defaultNode]);
  }
  function getFilterResultContent() {
    var arr = [];
    var columns2 = [];
    getAllColumns(props.columns, columns2);
    columns2.filter(function (col) {
      return col.filter;
    }).forEach(function (col, index) {
      var value = tFilterValue.value[col.colKey];
      if (col.filter.list && !["null"].includes(String(value))) {
        var formattedValue = value instanceof Array ? value : [value];
        var label = [];
        col.filter.list.forEach(function (option) {
          if (formattedValue.includes(option.value)) {
            label.push(option.label);
          }
        });
        value = label.join();
      }
      if (isFilterValueExist(value)) {
        var _col$filter, _col$filter2;
        var _label = isFunction.isFunction((_col$filter = col.filter) === null || _col$filter === void 0 ? void 0 : _col$filter.label) ? col.filter.label(Vue.h) : (_col$filter2 = col.filter) === null || _col$filter2 === void 0 ? void 0 : _col$filter2.label;
        var title = table_hooks_useTableHeader.renderTitle(context.slots, col, index);
        arr.push("".concat(_label || title, "\uFF1A").concat(value));
      }
    });
    return arr.join("\uFF1B");
  }
  function getAllColumns(col, columns2) {
    col.forEach(function (column) {
      if (column.children) {
        getAllColumns(column.children, columns2);
      }
      columns2.push(column);
    });
  }
  function onInnerFilterChange(val, column) {
    var filterValue2 = _objectSpread(_objectSpread({}, innerFilterValue.value), {}, _defineProperty__default["default"]({}, column.colKey, val));
    innerFilterValue.value = filterValue2;
    if (!column.filter.showConfirmAndReset) {
      emitFilterChange(filterValue2, "filter-change", column);
    }
  }
  function emitFilterChange(filterValue2, trigger, column) {
    var _props$onChange;
    setTFilterValue(filterValue2, {
      col: column,
      trigger: trigger
    });
    (_props$onChange = props.onChange) === null || _props$onChange === void 0 || _props$onChange.call(props, {
      filter: filterValue2
    }, {
      trigger: "filter"
    });
  }
  function onReset(column) {
    var _ref3, _column$filter$resetV;
    var filterValue2 = _objectSpread(_objectSpread({}, tFilterValue.value), {}, _defineProperty__default["default"]({}, column.colKey, (_ref3 = (_column$filter$resetV = column.filter.resetValue) !== null && _column$filter$resetV !== void 0 ? _column$filter$resetV : {
      single: "",
      multiple: [],
      input: ""
    }[column.filter.type]) !== null && _ref3 !== void 0 ? _ref3 : ""));
    emitFilterChange(filterValue2, "reset", column);
  }
  function onResetAll() {
    var resetValue = utils.getColumnsResetValue(columns.value);
    emitFilterChange(resetValue, "clear", void 0);
  }
  function onConfirm(column) {
    emitFilterChange(innerFilterValue.value, "confirm", column);
  }
  function renderFilterIcon(_ref4) {
    var _primaryTableRef$valu;
    var col = _ref4.col,
      colIndex = _ref4.colIndex;
    return Vue.createVNode(table_components_filterController["default"], {
      "column": col,
      "colIndex": colIndex,
      "filterIcon": props.filterIcon,
      "tFilterValue": tFilterValue.value,
      "innerFilterValue": innerFilterValue.value,
      "tableFilterClasses": tableFilterClasses,
      "isFocusClass": isFocusClass,
      "popupProps": col.filter.popupProps,
      "attach": props.attach,
      "locale": props.locale,
      "onInnerFilterChange": onInnerFilterChange,
      "onReset": onReset,
      "onConfirm": onConfirm,
      "primaryTableElement": (_primaryTableRef$valu = primaryTableRef.value) === null || _primaryTableRef$valu === void 0 ? void 0 : _primaryTableRef$valu.$el,
      "onVisibleChange": onPopupVisibleChange
    }, {
      filterIcon: context.slots.filterIcon
    });
  }
  function setFilterPrimaryTableRef(primaryTableElement) {
    primaryTableRef.value = primaryTableElement;
  }
  function onPopupVisibleChange(visible) {
    if (visible && !isTableOverflowHidden.value) {
      isTableOverflowHidden.value = !visible;
    }
  }
  return {
    hasEmptyCondition: hasEmptyCondition,
    isTableOverflowHidden: isTableOverflowHidden,
    renderFilterIcon: renderFilterIcon,
    renderFirstFilterRow: renderFirstFilterRow,
    setFilterPrimaryTableRef: setFilterPrimaryTableRef
  };
}

exports["default"] = useFilter;
