// tdesign-vue-next/cjs/table/hooks/useTableHeader.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _typeof = require('@babel/runtime/helpers/typeof');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var table_hooks_useMultiHeader = require('./useMultiHeader.js');
var table_hooks_useClassName = require('./useClassName.js');
var table_components_ellipsis = require('../components/ellipsis.js');
var isFunction = require('../../_chunks/dep-2dcf9237.js');
var isString = require('../../_chunks/dep-631750cd.js');
require('@babel/runtime/helpers/toConsumableArray');
require('../../_chunks/dep-35e23c2b.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-0238ed9a.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('../../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-1862d466.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../tooltip/index.js');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../_chunks/dep-38acad3d.js');
require('../../_chunks/dep-4ba41627.js');
require('../../_chunks/dep-ce9c15f5.js');
require('../../popup/container.js');
require('../../_chunks/dep-e2122882.js');
require('../../_chunks/dep-9bee89d3.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-ae4040c5.js');
require('../../_chunks/dep-79476fcb.js');
require('../../_chunks/dep-e3edff33.js');
require('../../tooltip/utils/index.js');
require('../../_chunks/dep-5c79e9ef.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-9a54a293.js');
require('../../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _typeof__default = /*#__PURE__*/_interopDefaultLegacy(_typeof);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
function renderTitle(slots, col, index) {
  var params = {
    col: col,
    colIndex: index
  };
  if (isFunction.isFunction(col.title)) {
    return col.title(Vue.h, params);
  }
  if (isString.isString(col.title) && slots[col.title]) {
    return slots[col.title](params);
  }
  if (isFunction.isFunction(col.render)) {
    return col.render(Vue.h, _objectSpread(_objectSpread({}, params), {}, {
      type: "title",
      row: {},
      rowIndex: -1
    })) || col.title;
  }
  return col.title;
}
function useTableHeader(props) {
  var _useClassName = table_hooks_useClassName["default"](),
    tableSortClasses = _useClassName.tableSortClasses,
    tableFilterClasses = _useClassName.tableFilterClasses;
  var spansAndLeafNodes = Vue.computed(function () {
    return table_hooks_useMultiHeader.getThRowspanAndColspan(props.columns);
  });
  var thList = Vue.computed(function () {
    return table_hooks_useMultiHeader.getThList(props.columns);
  });
  var isMultipleHeader = Vue.computed(function () {
    return thList.value.length > 1;
  });
  var renderTitleWidthIcon = function renderTitleWidthIcon(_ref, col, colIndex, ellipsisTitle, attach, extra) {
    var _ref2 = _slicedToArray__default["default"](_ref, 3),
      title = _ref2[0],
      sortIcon = _ref2[1],
      filterIcon = _ref2[2];
    var classes = _defineProperty__default["default"](_defineProperty__default["default"]({}, tableSortClasses.sortable, sortIcon), tableFilterClasses.filterable, filterIcon);
    var content = isFunction.isFunction(ellipsisTitle) ? ellipsisTitle(Vue.h, {
      col: col,
      colIndex: colIndex
    }) : void 0;
    var isEllipsis = ellipsisTitle !== void 0 ? Boolean(ellipsisTitle) : Boolean(col.ellipsis);
    return Vue.createVNode("div", {
      "class": classes
    }, [Vue.createVNode("div", {
      "class": tableSortClasses.title
    }, [isEllipsis ? Vue.createVNode(table_components_ellipsis["default"], {
      "placement": "bottom",
      "attach": props.attach || (attach ? function () {
        return attach;
      } : void 0),
      "tooltipContent": content && function () {
        return content;
      },
      "tooltipProps": _typeof__default["default"](ellipsisTitle) === "object" ? ellipsisTitle : void 0,
      "classPrefix": extra === null || extra === void 0 ? void 0 : extra.classPrefix,
      "overlayClassName": extra === null || extra === void 0 ? void 0 : extra.ellipsisOverlayClassName
    }, _isSlot(title) ? title : {
      "default": function _default() {
        return [title];
      }
    }) : Vue.createVNode("div", null, [title]), Boolean(sortIcon || filterIcon) && Vue.createVNode("div", {
      "class": tableFilterClasses.iconWrap
    }, [sortIcon, filterIcon])])]);
  };
  return {
    thList: thList,
    isMultipleHeader: isMultipleHeader,
    spansAndLeafNodes: spansAndLeafNodes,
    renderTitleWidthIcon: renderTitleWidthIcon
  };
}

exports["default"] = useTableHeader;
exports.renderTitle = renderTitle;
