// tdesign-vue-next/cjs/space/space.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var space_props = require('./props.js');
var index$2 = require('../_chunks/dep-0cda9fe9.js');
var index$1 = require('../_chunks/dep-1862d466.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('../_chunks/dep-0989c3be.js');
require('../_chunks/dep-35e23c2b.js');
require('@babel/runtime/helpers/toConsumableArray');
var helper = require('../_chunks/dep-2aa3d3cf.js');
var isArray = require('../_chunks/dep-4cb26289.js');
var isNumber = require('../_chunks/dep-44195fb2.js');
var isString = require('../_chunks/dep-631750cd.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('../_chunks/dep-4a7162e7.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var sizeMap = {
  small: "8px",
  medium: "16px",
  large: "24px"
};
var defaultNeedPolyfill = helper.getFlexGapPolyFill();
var _Space = Vue.defineComponent({
  name: "TSpace",
  props: _objectSpread(_objectSpread({}, space_props["default"]), {}, {
    forceFlexGapPolyfill: Boolean
  }),
  setup: function setup(props2) {
    var COMPONENT_NAME = index.usePrefixClass("space");
    var renderTNodeJSX = index$1.useTNodeJSX();
    var getChildSlots = index$2.useChildSlots();
    var getFlatChildren = index$2.useFlatChildrenSlots();
    var needPolyfill = Vue.computed(function () {
      return props2.forceFlexGapPolyfill || defaultNeedPolyfill;
    });
    var renderStyle = Vue.computed(function () {
      var renderGap = "";
      if (isArray.isArray(props2.size)) {
        renderGap = props2.size.map(function (s) {
          if (isNumber.isNumber(s)) return "".concat(s, "px");
          if (isString.isString(s) && ["small", "medium", "large"].includes(s)) return sizeMap[s];
          return s;
        }).join(" ");
      } else if (isString.isString(props2.size)) {
        renderGap = ["small", "medium", "large"].includes(props2.size) ? sizeMap[props2.size] : props2.size;
      } else if (isNumber.isNumber(props2.size)) {
        renderGap = "".concat(props2.size, "px");
      }
      var style = {};
      if (needPolyfill.value) {
        var _renderGap$split = renderGap.split(" "),
          _renderGap$split2 = _slicedToArray__default["default"](_renderGap$split, 2),
          columnGap = _renderGap$split2[0],
          rowGap = _renderGap$split2[1];
        style["--td-space-column-gap"] = columnGap;
        style["--td-space-row-gap"] = rowGap || columnGap;
      } else {
        style.gap = renderGap;
      }
      return style;
    });
    function renderChildren() {
      var children = getFlatChildren(getChildSlots());
      var separatorContent = renderTNodeJSX("separator");
      return children.map(function (child, index) {
        var showSeparator = index + 1 !== children.length && separatorContent;
        return Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
          "class": "".concat(COMPONENT_NAME.value, "-item")
        }, [child]), showSeparator && Vue.createVNode("div", {
          "class": "".concat(COMPONENT_NAME.value, "-item-separator")
        }, [separatorContent])]);
      });
    }
    return function () {
      var spaceClassNames = ["".concat(COMPONENT_NAME.value), _defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "-align-").concat(props2.align), props2.align), "".concat(COMPONENT_NAME.value, "-").concat(props2.direction), props2.direction), "".concat(COMPONENT_NAME.value, "--break-line"), props2.breakLine), "".concat(COMPONENT_NAME.value, "--polyfill"), needPolyfill.value)];
      return Vue.createVNode("div", {
        "class": spaceClassNames,
        "style": renderStyle.value
      }, [renderChildren()]);
    };
  }
});

exports["default"] = _Space;
