// tdesign-vue-next/es/tree/utils/adapt.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import * as Vue from 'vue';
export { TransitionGroup, computed, defineComponent, onMounted, reactive, ref, toRefs, watch } from 'vue';
import { j as useVModel$1, k as useDefaultValue } from '../../_chunks/dep-fbeb2963.mjs';
export { i as useGlobalIcon, y as useLazyLoad, u as usePrefixClass, x as useVirtualScrollNew } from '../../_chunks/dep-fbeb2963.mjs';
import { w as withInstall$1 } from '../../_chunks/dep-feed89df.mjs';
export { CaretRightSmallIcon as TdCaretRightSmallIcon } from 'tdesign-icons-vue-next';
export { Checkbox as TCheckBox } from '../../checkbox/index.mjs';
export { Loading as TLoading } from '../../loading/index.mjs';
export { T as TreeNode, p as privateKey } from '../../_chunks/dep-9e0400d9.mjs';
export { useConfig } from '../../config-provider/hooks/useConfig.mjs';
import 'lodash-es';
import '../../_chunks/dep-47c3cae9.mjs';
import '../../_chunks/dep-b0b422fe.mjs';
import '../../_chunks/dep-e1c62679.mjs';
import '../../_chunks/dep-edab0e6a.mjs';
import '../../_chunks/dep-f558ebb8.mjs';
import '../../_chunks/dep-76dddedf.mjs';
import '../../_chunks/dep-d92220c4.mjs';
import '../../config-provider/utils/context.mjs';
import '../../_chunks/dep-e0ab6d55.mjs';
import '../../_chunks/dep-5b510fcd.mjs';
import '../../_chunks/dep-75f265c4.mjs';
import '../../_chunks/dep-67ffcfeb.mjs';
import '../../_chunks/dep-466a8f31.mjs';
import '../../_chunks/dep-ba214d75.mjs';
import '../../_chunks/dep-e66c0337.mjs';
import '../../checkbox/checkbox.mjs';
import '../../checkbox/props.mjs';
import '../../checkbox/constants/index.mjs';
import '../../checkbox/hooks/useCheckboxLazyLoad.mjs';
import '../../checkbox/hooks/useKeyboardEvent.mjs';
import '../../checkbox/group.mjs';
import '../../checkbox/checkbox-group-props.mjs';
import './style/css.mjs';
import '../../loading/directive.mjs';
import '../../loading/plugin.mjs';
import '../../loading/loading.mjs';
import '../../loading/icon/gradient.mjs';
import '../../_chunks/dep-58979704.mjs';
import '../../_chunks/dep-c231a5b4.mjs';
import '../../loading/props.mjs';
import '../../_chunks/dep-3114e4de.mjs';
import '../../_chunks/dep-739ea4cd.mjs';

var onDrag = {
  "default": void 0
};
var isVueNext = true;
var TreeItemDefinition = {
  name: "TTreeNode",
  inject: {
    onDrag: onDrag
  }
};
function useRipple(el) {}
function withInstall(construct) {
  return withInstall$1(construct);
}
function getCreateElement(h) {
  if (h) {}
  return Vue.h;
}
function getScopedSlots(instance) {
  return instance.$slots;
}
function useVModel(props, refsProps) {
  var propName = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : "value";
  var defaultPropName = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : "defaultValue";
  var eventPropName = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : "onChange";
  var eventName = arguments.length > 5 && arguments[5] !== undefined ? arguments[5] : "change";
  if (eventName) {}
  var modelValue = refsProps.modelValue;
  var vm;
  if (propName === "value") {
    vm = useVModel$1(refsProps[propName], modelValue, props[defaultPropName], props[eventPropName], propName);
  } else {
    vm = useDefaultValue(refsProps[propName], props[defaultPropName], props[eventPropName], propName);
  }
  return vm;
}

export { TreeItemDefinition, getCreateElement, getScopedSlots, isVueNext, useRipple, useVModel, withInstall };
