// tdesign-vue-next/cjs/color-picker/components/trigger/index.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var input_index = require('../../../input/index.js');
var color = require('../../../_chunks/dep-34c8efe7.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/toConsumableArray');
var colorPicker_hooks_index = require('../../hooks/index.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-35e23c2b.js');
var index = require('../../../_chunks/dep-38acad3d.js');
require('../../../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-0989c3be.js');
var index$1 = require('../../../_chunks/dep-a13f0135.js');
require('../../../input/input.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('tdesign-icons-vue-next');
require('../../../input/props.js');
require('../../../_chunks/dep-1862d466.js');
require('../../../_chunks/dep-8d990e66.js');
require('../../../_chunks/dep-2dcf9237.js');
require('../../../_chunks/dep-22dc294c.js');
require('../../../_chunks/dep-bf76dead.js');
require('../../../_chunks/dep-631750cd.js');
require('../../../_chunks/dep-4cb26289.js');
require('../../../_chunks/dep-a6551cc4.js');
require('../../../_chunks/dep-7d9abefd.js');
require('../../../_chunks/dep-2fb8d44f.js');
require('../../../_chunks/dep-80a41429.js');
require('../../../_chunks/dep-91305690.js');
require('../../../_chunks/dep-0b92e695.js');
require('../../../_chunks/dep-b2024592.js');
require('../../../_chunks/dep-9f98bf9b.js');
require('../../../_chunks/dep-0238ed9a.js');
require('../../../_chunks/dep-362c5212.js');
require('../../../_chunks/dep-74b143e0.js');
require('../../../_chunks/dep-c26a7d50.js');
require('../../../_chunks/dep-9dbade68.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../_chunks/dep-4a7162e7.js');
require('../../../_chunks/dep-be1af85d.js');
require('../../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../../_chunks/dep-82fe2026.js');
require('../../../_chunks/dep-33b46a52.js');
require('../../../_chunks/dep-930a2de5.js');
require('../../../_chunks/dep-c3bbd06c.js');
require('../../../_chunks/dep-6d87f74d.js');
require('../../../_chunks/dep-281f7eb2.js');
require('../../../_chunks/dep-f8f7b1af.js');
require('../../../_chunks/dep-66d3f30f.js');
require('../../../_chunks/dep-e1d22111.js');
require('../../../_chunks/dep-754c0523.js');
require('../../../_chunks/dep-7076a08a.js');
require('../../../_chunks/dep-2fc884aa.js');
require('../../../_chunks/dep-75d6da05.js');
require('../../../_chunks/dep-274369eb.js');
require('../../../_chunks/dep-df6e14a0.js');
require('../../../_chunks/dep-ce9c15f5.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-692bdebe.js');
require('../../../_chunks/dep-2aa3d3cf.js');
require('../../../_chunks/dep-3d29e358.js');
require('../../../_chunks/dep-6212811d.js');
require('../../../_chunks/dep-44195fb2.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-e2122882.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../_chunks/dep-e3edff33.js');
require('tinycolor2');
require('../../../_chunks/dep-9d53c643.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var DefaultTrigger = Vue.defineComponent({
  name: "DefaultTrigger",
  inheritAttrs: false,
  props: {
    color: {
      type: String,
      "default": ""
    },
    disabled: {
      type: Boolean,
      "default": false
    },
    borderless: {
      type: Boolean,
      "default": false
    },
    clearable: {
      type: Boolean,
      "default": false
    },
    inputProps: {
      type: Object,
      "default": function _default() {
        return {
          autoWidth: true
        };
      }
    },
    onTriggerChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    },
    onTriggerClear: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    },
    size: {
      type: String,
      "default": "medium"
    }
  },
  setup: function setup(props) {
    var baseClassName = colorPicker_hooks_index.useBaseClassName();
    var value = Vue.ref(props.color);
    var _useCommonClassName = index.useCommonClassName(),
      sizeClassNames = _useCommonClassName.SIZE;
    Vue.watch(function () {
      return [props.color];
    }, function () {
      return value.value = props.color;
    });
    var handleChange = function handleChange(input) {
      if (input !== props.color) {
        props.onTriggerChange(value.value, {
          color: color.getColorObject(new color.Color(input)),
          trigger: "input"
        });
      }
    };
    var handleClear = function handleClear(context) {
      var _props$onTriggerClear;
      return (_props$onTriggerClear = props.onTriggerClear) === null || _props$onTriggerClear === void 0 ? void 0 : _props$onTriggerClear.call(props, context);
    };
    var inputEvents = index$1.useEventForward(props.inputProps, {
      onBlur: handleChange,
      onChange: handleChange,
      onClear: handleClear
    });
    return function () {
      var inputSlots = {
        label: function label() {
          return Vue.createVNode("div", {
            "class": ["".concat(baseClassName.value, "__trigger--default__color"), "".concat(baseClassName.value, "--bg-alpha")]
          }, [Vue.createVNode("span", {
            "class": ["color-inner", _defineProperty__default["default"]({}, sizeClassNames.value[props.size], props.size !== "medium")],
            "style": {
              background: value.value
            }
          }, null)]);
        }
      };
      return Vue.createVNode(input_index.Input, Vue.mergeProps({
        "borderless": props.borderless,
        "clearable": props.clearable,
        "size": props.size,
        "modelValue": value.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return value.value = $event;
        },
        "disabled": props.disabled
      }, inputEvents.value), inputSlots);
    };
  }
});

exports["default"] = DefaultTrigger;
