// tdesign-vue-next/lib/drawer/plugin.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import _defineProperty from '@babel/runtime/helpers/defineProperty';
import { ref, defineComponent, onMounted, nextTick, h, createVNode, render } from 'vue';
import _Drawer from './drawer.js';
import { g as getAttach } from '../_chunks/dep-d518fdfb.js';
import { omit } from 'lodash-es';
import 'tdesign-icons-vue-next';
import '@babel/runtime/helpers/toConsumableArray';
import '@babel/runtime/helpers/typeof';
import '../_chunks/dep-7bdccf65.js';
import '../_chunks/dep-8d4d971b.js';
import '../_chunks/dep-e8dd47a9.js';
import '../config-provider/hooks/useConfig.js';
import '../config-provider/utils/context.js';
import '../_chunks/dep-509ddbe3.js';
import '../_chunks/dep-f0f392fb.js';
import 'dayjs';
import '@babel/runtime/helpers/slicedToArray';
import '../_chunks/dep-52eae58a.js';
import '../_chunks/dep-91fc762d.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../_chunks/dep-2a7145b5.js';
import '../loading/icon/gradient.js';
import '../_chunks/dep-c68ea098.js';
import '../_chunks/dep-ef7a41ce.js';
import '../_chunks/dep-fbf70ecb.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../_chunks/dep-98d89c71.js';
import '../loading/props.js';
import '../_chunks/dep-295b33d1.js';
import './props.js';
import '../dialog/hooks/useAction.js';
import '../button/index.js';
import '../button/button.js';
import '../loading/index.js';
import '../loading/directive.js';
import '../loading/plugin.js';
import '../_chunks/dep-ae3e94b6.js';
import '../button/props.js';
import '../_chunks/dep-2ba9b7d0.js';
import '../_chunks/dep-d67d6f8a.js';
import './hooks/index.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var createDrawer = function createDrawer(props, context) {
  var options = _objectSpread({}, props);
  var wrapper = document.createElement("div");
  var visible = ref(false);
  var style = options.style;
  var updateStyle = function updateStyle(style2) {
    if (style2) {
      wrapper.firstElementChild.style.cssText += style2;
    }
  };
  function destroySelf() {
    render(null, wrapper);
    wrapper.remove();
  }
  var component = defineComponent({
    setup: function setup(props2, _ref) {
      var expose = _ref.expose;
      var drawerOptions = ref(options);
      onMounted(function () {
        nextTick(function () {
          visible.value = true;
        });
        document.activeElement.blur();
        nextTick(function () {
          updateStyle(style);
        });
      });
      var update = function update(newOptions) {
        var _drawerOptions$value;
        drawerOptions.value = _objectSpread(_objectSpread({}, options), {}, {
          drawerClassName: (_drawerOptions$value = drawerOptions.value) === null || _drawerOptions$value === void 0 ? void 0 : _drawerOptions$value.className
        }, omit(newOptions, ["className", "style"]));
      };
      expose({
        update: update
      });
      return function () {
        var _drawerOptions$value2;
        var onClose = options.onClose || function () {
          visible.value = false;
          if (options.destroyOnClose) {
            setTimeout(function () {
              destroySelf();
            }, 300);
          }
        };
        delete options.style;
        return h(_Drawer, _objectSpread({
          onClose: onClose,
          visible: visible.value,
          drawerClassName: (_drawerOptions$value2 = drawerOptions.value) === null || _drawerOptions$value2 === void 0 ? void 0 : _drawerOptions$value2.className
        }, omit(drawerOptions.value, "className", "style")));
      };
    }
  });
  var drawerComponent = createVNode(component);
  if (context !== null && context !== void 0 ? context : DrawerPlugin._context) {
    drawerComponent.appContext = context !== null && context !== void 0 ? context : DrawerPlugin._context;
  }
  var container = getAttach(options.attach);
  if (container) {
    container.appendChild(wrapper);
  } else {
    console.error("attach is not exist");
  }
  render(drawerComponent, wrapper);
  var destroyDrawer = function destroyDrawer() {
    visible.value = false;
    setTimeout(function () {
      destroySelf();
    }, 400);
  };
  var drawerNode = {
    show: function show() {
      visible.value = true;
    },
    hide: function hide() {
      visible.value = false;
    },
    update: function update(newOptions) {
      drawerComponent.component.exposed.update(newOptions);
      updateStyle(newOptions.style);
    },
    destroy: function destroy() {
      destroyDrawer();
    }
  };
  return drawerNode;
};
var DrawerPlugin = createDrawer;
DrawerPlugin.install = function (app) {
  DrawerPlugin._context = app._context;
  app.config.globalProperties.$drawer = createDrawer;
};

export { DrawerPlugin, DrawerPlugin as default };
