// tdesign-vue-next/cjs/cascader/components/Item.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var cascader_utils_className = require('../utils/className.js');
require('@babel/runtime/helpers/slicedToArray');
var cascader_utils_helper = require('../utils/helper.js');
var checkbox_index = require('../../checkbox/index.js');
var loading_index = require('../../loading/index.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var index$3 = require('../../_chunks/dep-38acad3d.js');
var index$1 = require('../../_chunks/dep-0238ed9a.js');
var index$2 = require('../../_chunks/dep-c26a7d50.js');
require('../../_chunks/dep-0989c3be.js');
var index = require('../../_chunks/dep-719a33c8.js');
require('@babel/runtime/helpers/defineProperty');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-44195fb2.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-7fb34f13.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-274369eb.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-1862d466.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-362c5212.js');
require('../../_chunks/dep-74b143e0.js');
require('../../_chunks/dep-9dbade68.js');
require('../../_chunks/dep-ce9c15f5.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-b9a88912.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-a4be1915.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-4ba41627.js');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-0cda9fe9.js');
require('../../_chunks/dep-b0334c78.js');
require('../../_chunks/dep-02ea4f3c.js');
require('../../_chunks/dep-a214821d.js');
require('../../_chunks/dep-3d29e358.js');
require('../../_chunks/dep-e3edff33.js');
require('../../_chunks/dep-34cd76d4.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-d632b788.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6212811d.js');
require('../../loading/props.js');
require('../../_chunks/dep-43dcd2db.js');
require('../../_chunks/dep-392dde71.js');
require('../../_chunks/dep-5103206f.js');
require('../../_chunks/dep-7707ffe2.js');
require('../../_chunks/dep-74a5bebe.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-42740075.js');
require('../../_chunks/dep-9ac5c6ca.js');

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var props = {
  node: {
    type: Object,
    "default": function _default() {
      return {};
    }
  },
  optionChild: {
    type: [Object, Array]
  },
  cascaderContext: {
    type: Object
  },
  onChange: Function,
  onClick: Function,
  onMouseenter: Function
};
var Item = Vue.defineComponent({
  name: "TCascaderItem",
  props: props,
  setup: function setup(props2) {
    var liRef = Vue.ref();
    index.useRipple(liRef);
    var COMPONENT_NAME = index$1.usePrefixClass("cascader__item");
    var classPrefix = index$1.usePrefixClass();
    var _useGlobalIcon = index$2.useGlobalIcon({
        ChevronRightIcon: tdesignIconsVueNext.ChevronRightIcon
      }),
      ChevronRightIcon = _useGlobalIcon.ChevronRightIcon;
    var _useCommonClassName = index$3.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS,
      SIZE = _useCommonClassName.SIZE;
    var itemClass = Vue.computed(function () {
      return cascader_utils_className.getCascaderItemClass(classPrefix.value, props2.node, SIZE.value, STATUS.value, props2.cascaderContext);
    });
    var iconClass = Vue.computed(function () {
      return cascader_utils_className.getCascaderItemIconClass(classPrefix.value, props2.node, STATUS.value, props2.cascaderContext);
    });
    function RenderLabelInner(node, cascaderContext) {
      var inputVal = cascaderContext.inputVal;
      var labelText = inputVal ? cascader_utils_helper.getFullPathLabel(node) : node.label;
      if (inputVal) {
        var texts = labelText.split(inputVal);
        var doms = [];
        for (var index = 0; index < texts.length; index++) {
          doms.push(Vue.createVNode("span", {
            "key": index
          }, [texts[index]]));
          if (index === texts.length - 1) break;
          doms.push(Vue.createVNode("span", {
            "key": "".concat(index, "filter"),
            "class": "".concat(COMPONENT_NAME.value, "-label--filter")
          }, [inputVal]));
        }
        return doms;
      }
      return labelText;
    }
    var renderTitle = function renderTitle(node) {
      if (typeof node.label === "string") return node.label;
      return null;
    };
    function RenderLabelContent(node, cascaderContext) {
      var label = RenderLabelInner(node, cascaderContext);
      var labelCont = Vue.createVNode("span", {
        "title": cascaderContext.inputVal ? cascader_utils_helper.getFullPathLabel(node) : renderTitle(node),
        "class": ["".concat(COMPONENT_NAME.value, "-label"), "".concat(COMPONENT_NAME.value, "-label--ellipsis")],
        "role": "label"
      }, [label]);
      return labelCont;
    }
    function RenderCheckBox(node, cascaderContext) {
      var checkProps = cascaderContext.checkProps,
        value = cascaderContext.value,
        max = cascaderContext.max,
        inputVal = cascaderContext.inputVal;
      var label = RenderLabelInner(node, cascaderContext);
      return Vue.createVNode(checkbox_index.Checkbox, Vue.mergeProps({
        "checked": node.checked,
        "indeterminate": node.indeterminate,
        "disabled": node.isDisabled() || value.length >= max && max !== 0,
        "name": String(node.value),
        "stopLabelTrigger": !!node.children && !props2.cascaderContext.isParentFilterable,
        "title": inputVal ? cascader_utils_helper.getFullPathLabel(node) : renderTitle(node),
        "onChange": function onChange() {
          props2.onChange();
        }
      }, checkProps), _isSlot(label) ? label : {
        "default": function _default() {
          return [label];
        }
      });
    }
    return function () {
      var cascaderContext = props2.cascaderContext,
        node = props2.node,
        optionChild = props2.optionChild;
      return Vue.createVNode("li", {
        "ref": liRef,
        "class": itemClass.value,
        "onClick": props2.onClick,
        "onMouseenter": props2.onMouseenter
      }, [optionChild || (cascaderContext.multiple ? RenderCheckBox(node, cascaderContext) : RenderLabelContent(node, cascaderContext)), node.children && !props2.cascaderContext.isParentFilterable && (node.loading ? Vue.createVNode(loading_index.Loading, {
        "class": iconClass.value,
        "size": "small"
      }, null) : Vue.createVNode(ChevronRightIcon, {
        "class": iconClass.value
      }, null))]);
    };
  }
});

exports["default"] = Item;
