// tdesign-vue-next/cjs/cascader/components/Panel.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var cascader_components_Item = require('./Item.js');
var cascader_props = require('../props.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var renderTnode = require('../../_chunks/dep-a6551cc4.js');
var index = require('../../_chunks/dep-1862d466.js');
var index$1 = require('../../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-0989c3be.js');
var cascader_utils_effect = require('../utils/effect.js');
var cascader_utils_helper = require('../utils/helper.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('tdesign-icons-vue-next');
require('../utils/className.js');
require('../../checkbox/index.js');
require('../../checkbox/checkbox.js');
require('../../checkbox/props.js');
require('../../_chunks/dep-38acad3d.js');
require('../../_chunks/dep-362c5212.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-74b143e0.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-9dbade68.js');
require('../../_chunks/dep-719a33c8.js');
require('../../_chunks/dep-4ba41627.js');
require('../../_chunks/dep-ce9c15f5.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');
require('../../checkbox/constants/index.js');
require('../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../_chunks/dep-b9a88912.js');
require('../../checkbox/hooks/useKeyboardEvent.js');
require('../../_chunks/dep-a4be1915.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../checkbox/group.js');
require('../../checkbox/checkbox-group-props.js');
require('../../_chunks/dep-0cda9fe9.js');
require('../../_chunks/dep-b0334c78.js');
require('../../_chunks/dep-02ea4f3c.js');
require('../../_chunks/dep-a214821d.js');
require('../../_chunks/dep-3d29e358.js');
require('../../_chunks/dep-e3edff33.js');
require('../../loading/index.js');
require('../../_chunks/dep-34cd76d4.js');
require('../../loading/plugin.js');
require('../../_chunks/dep-d632b788.js');
require('../../loading/icon/gradient.js');
require('../../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-6212811d.js');
require('../../_chunks/dep-44195fb2.js');
require('../../loading/props.js');
require('../../_chunks/dep-43dcd2db.js');
require('../../_chunks/dep-392dde71.js');
require('../../_chunks/dep-5103206f.js');
require('../../_chunks/dep-7707ffe2.js');
require('../../_chunks/dep-74a5bebe.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-42740075.js');
require('../../_chunks/dep-9ac5c6ca.js');
require('../../_chunks/dep-c26a7d50.js');
require('../../_chunks/dep-7fb34f13.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var TCascaderSubPanel = Vue.defineComponent({
  name: "TCascaderSubPanel",
  props: {
    option: cascader_props["default"].option,
    options: cascader_props["default"].options,
    empty: cascader_props["default"].empty,
    trigger: cascader_props["default"].trigger,
    onChange: cascader_props["default"].onChange,
    loading: cascader_props["default"].loading,
    loadingText: cascader_props["default"].loadingText,
    cascaderContext: {
      type: Object
    }
  },
  setup: function setup(props) {
    var renderTNodeJSXDefault = index.useTNodeDefault();
    var COMPONENT_NAME = index$1.usePrefixClass("cascader");
    var _useConfig = configProvider_hooks_useConfig.useConfig("cascader"),
      globalConfig = _useConfig.globalConfig;
    var panels = Vue.computed(function () {
      return cascader_utils_helper.getPanels(props.cascaderContext.treeNodes);
    });
    var handleExpand = function handleExpand(node, trigger) {
      var propsTrigger = props.trigger,
        cascaderContext = props.cascaderContext;
      cascader_utils_effect.expandClickEffect(propsTrigger, trigger, node, cascaderContext);
    };
    var renderItem = function renderItem(node, index) {
      var optionChild = node.data.content ? renderTnode.getDefaultNode(node.data.content(Vue.h)) : renderTNodeJSXDefault("option", {
        params: {
          item: node.data,
          index: index,
          onExpand: function onExpand() {
            return handleExpand(node, "click");
          },
          onChange: function onChange() {
            return cascader_utils_effect.valueChangeEffect(node, props.cascaderContext);
          }
        }
      });
      return Vue.createVNode(cascader_components_Item["default"], {
        "key": node.value,
        "node": node,
        "optionChild": optionChild,
        "cascaderContext": props.cascaderContext,
        "onClick": function onClick() {
          handleExpand(node, "click");
        },
        "onMouseenter": function onMouseenter() {
          handleExpand(node, "hover");
        },
        "onChange": function onChange() {
          cascader_utils_effect.valueChangeEffect(node, props.cascaderContext);
        }
      }, null);
    };
    var renderList = function renderList(treeNodes) {
      var isFilter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
      var segment = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
      var index = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : 1;
      return Vue.createVNode("ul", {
        "class": ["".concat(COMPONENT_NAME.value, "__menu"), "narrow-scrollbar", _defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__menu--segment"), segment), "".concat(COMPONENT_NAME.value, "__menu--filter"), isFilter)],
        "key": "".concat(COMPONENT_NAME, "__menu").concat(index)
      }, [treeNodes.map(function (node) {
        return renderItem(node, index);
      })]);
    };
    var renderPanels = function renderPanels() {
      var _props$cascaderContex = props.cascaderContext,
        inputVal = _props$cascaderContex.inputVal,
        treeNodes = _props$cascaderContex.treeNodes;
      return inputVal ? renderList(treeNodes, true) : panels.value.map(function (treeNodes2, index) {
        return renderList(treeNodes2, false, index !== panels.value.length - 1, index);
      });
    };
    return function () {
      var content;
      if (props.loading) {
        content = renderTNodeJSXDefault("loadingText", Vue.createVNode("div", {
          "class": "".concat(COMPONENT_NAME.value, "__panel--empty")
        }, [globalConfig.value.loadingText]));
      } else {
        content = panels.value.length ? renderPanels() : renderTNodeJSXDefault("empty", Vue.createVNode("div", {
          "class": "".concat(COMPONENT_NAME.value, "__panel--empty")
        }, [globalConfig.value.empty]));
      }
      return Vue.createVNode("div", {
        "class": ["".concat(COMPONENT_NAME.value, "__panel"), _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--normal"), panels.value.length && !props.loading)]
      }, [content]);
    };
  }
});

exports["default"] = TCascaderSubPanel;
