// tdesign-vue-next/cjs/message/plugin.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var Vue = require('vue');
var dom = require('../_chunks/dep-8d990e66.js');
var message_messageList = require('./message-list.js');
var isString = require('../_chunks/dep-631750cd.js');
var isObject = require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-35e23c2b.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-81e71c43.js');
require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/toConsumableArray');
require('../_chunks/dep-1862d466.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-0238ed9a.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-c26a7d50.js');
require('../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../loading/index.js');
require('../_chunks/dep-34cd76d4.js');
require('../loading/plugin.js');
require('../_chunks/dep-d632b788.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');
require('../_chunks/dep-38acad3d.js');
require('../loading/props.js');
require('../_chunks/dep-43dcd2db.js');
require('../_chunks/dep-392dde71.js');
require('../_chunks/dep-02ea4f3c.js');
require('../_chunks/dep-5103206f.js');
require('../_chunks/dep-7707ffe2.js');
require('../_chunks/dep-74a5bebe.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-42740075.js');
require('../_chunks/dep-9ac5c6ca.js');
require('../_chunks/dep-e3edff33.js');
require('./props.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var instanceMap = /* @__PURE__ */new Map();
function handleParams(params) {
  var options = _objectSpread({
    duration: 3e3,
    attach: "body",
    zIndex: message_messageList.DEFAULT_Z_INDEX,
    placement: "top"
  }, params);
  options.content = params.content;
  return options;
}
var MessageFunction = function MessageFunction(props, context) {
  var options = handleParams(props);
  var attach = options.attach,
    placement = options.placement;
  var attachDom = dom.getAttach(attach);
  if (!instanceMap.get(attachDom)) {
    instanceMap.set(attachDom, {});
  }
  var p = instanceMap.get(attachDom)[placement];
  var mgKey;
  if (!p || !attachDom.contains(p.el)) {
    var wrapper = document.createElement("div");
    var instance = Vue.createVNode(message_messageList.MessageList, {
      zIndex: options.zIndex,
      placement: options.placement
    });
    if (context !== null && context !== void 0 ? context : MessagePlugin._context) {
      instance.appContext = context !== null && context !== void 0 ? context : MessagePlugin._context;
    }
    attachDom.appendChild(wrapper);
    Vue.render(instance, wrapper);
    mgKey = instance.component.exposed.add(options);
    instanceMap.get(attachDom)[placement] = instance;
  } else {
    mgKey = p.component.exposed.add(options);
  }
  return new Promise(function (resolve) {
    var ins = instanceMap.get(attachDom)[placement];
    Vue.nextTick(function () {
      var msg = ins.component.exposed.messageList.value;
      resolve(msg === null || msg === void 0 ? void 0 : msg.find(function (mg) {
        var _mg$$;
        return ((_mg$$ = mg.$) === null || _mg$$ === void 0 || (_mg$$ = _mg$$.vnode) === null || _mg$$ === void 0 ? void 0 : _mg$$.key) === mgKey;
      }));
    });
  });
};
var showThemeMessage = function showThemeMessage(theme, params, duration, context) {
  var options = {
    theme: theme
  };
  if (Vue.isVNode(params)) {
    options.content = function () {
      return params;
    };
  } else if (isString.isString(params)) {
    options.content = params;
  } else if (isObject.isObject(params) && !(params instanceof Array)) {
    options = _objectSpread(_objectSpread({}, options), params);
  }
  (duration || duration === 0) && (options.duration = duration);
  return MessageFunction(options, context);
};
var extraApi = {
  info: function info(params, duration, context) {
    return showThemeMessage("info", params, duration, context);
  },
  success: function success(params, duration, context) {
    return showThemeMessage("success", params, duration, context);
  },
  warning: function warning(params, duration, context) {
    return showThemeMessage("warning", params, duration, context);
  },
  error: function error(params, duration, context) {
    return showThemeMessage("error", params, duration, context);
  },
  question: function question(params, duration, context) {
    return showThemeMessage("question", params, duration, context);
  },
  loading: function loading(params, duration, context) {
    return showThemeMessage("loading", params, duration, context);
  },
  close: function close(promise) {
    promise.then(function (instance) {
      return instance === null || instance === void 0 ? void 0 : instance.close();
    });
  },
  closeAll: function closeAll() {
    if (instanceMap instanceof Map) {
      instanceMap.forEach(function (attach) {
        Object.keys(attach).forEach(function (placement) {
          var instance = attach[placement];
          instance.component.exposed.removeAll();
        });
      });
    }
  }
};
var MessagePlugin = showThemeMessage;
MessagePlugin.install = function (app) {
  app.config.globalProperties.$message = showThemeMessage;
  Object.keys(extraApi).forEach(function (funcName) {
    app.config.globalProperties.$message[funcName] = extraApi[funcName];
  });
  MessagePlugin._context = app._context;
};
Object.keys(extraApi).forEach(function (funcName) {
  MessagePlugin[funcName] = extraApi[funcName];
});

exports.MessagePlugin = MessagePlugin;
exports["default"] = MessagePlugin;
