// tdesign-vue-next/cjs/list/list.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$2 = require('../_chunks/dep-1862d466.js');
var index$1 = require('../_chunks/dep-38acad3d.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
var loading_index = require('../loading/index.js');
var list_listItem = require('./list-item.js');
var list_props = require('./props.js');
var list_constants_index = require('./constants/index.js');
var list_hooks_useListItems = require('./hooks/useListItems.js');
var list_hooks_useListVirtualScroll = require('./hooks/useListVirtualScroll.js');
var isString = require('../_chunks/dep-631750cd.js');
var omit = require('../_chunks/dep-5c79e9ef.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-34cd76d4.js');
require('../loading/plugin.js');
require('../_chunks/dep-d632b788.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');
require('../loading/props.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-43dcd2db.js');
require('../_chunks/dep-392dde71.js');
require('../_chunks/dep-02ea4f3c.js');
require('../_chunks/dep-5103206f.js');
require('../_chunks/dep-7707ffe2.js');
require('../_chunks/dep-74a5bebe.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-42740075.js');
require('../_chunks/dep-9ac5c6ca.js');
require('../_chunks/dep-e3edff33.js');
require('./list-item-props.js');
require('../_chunks/dep-0cda9fe9.js');
require('../_chunks/dep-692bdebe.js');
require('../_chunks/dep-0349078b.js');
require('../_chunks/dep-e2122882.js');
require('../_chunks/dep-27729d32.js');
require('../_chunks/dep-79476fcb.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var _List = Vue.defineComponent({
  name: "TList",
  props: list_props["default"],
  setup: function setup(props2, _ref) {
    var expose = _ref.expose;
    var listRef = Vue.ref();
    var _useConfig = configProvider_hooks_useConfig.useConfig("list"),
      globalConfig = _useConfig.globalConfig;
    var COMPONENT_NAME = index.usePrefixClass("list");
    var _useCommonClassName = index$1.useCommonClassName(),
      SIZE = _useCommonClassName.SIZE;
    var renderTNodeJSX = index$2.useTNodeJSX();
    var _useListItems = list_hooks_useListItems.useListItems(),
      listItems = _useListItems.listItems;
    var _useListVirtualScroll = list_hooks_useListVirtualScroll.useListVirtualScroll(props2.scroll, listRef, listItems),
      virtualConfig = _useListVirtualScroll.virtualConfig,
      cursorStyle = _useListVirtualScroll.cursorStyle,
      listStyle = _useListVirtualScroll.listStyle,
      isVirtualScroll = _useListVirtualScroll.isVirtualScroll,
      onInnerVirtualScroll = _useListVirtualScroll.onInnerVirtualScroll,
      scrollToElement = _useListVirtualScroll.scrollToElement;
    var listClass = Vue.computed(function () {
      return ["".concat(COMPONENT_NAME.value), SIZE.value[props2.size], _defineProperty__default["default"](_defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--split"), props2.split), "".concat(COMPONENT_NAME.value, "--stripe"), props2.stripe), "".concat(COMPONENT_NAME.value, "--vertical-action"), props2.layout === "vertical")];
    });
    var renderContent = function renderContent() {
      var propsHeaderContent = renderTNodeJSX("header");
      var propsFooterContent = renderTNodeJSX("footer");
      var isVirtualScroll2 = virtualConfig.isVirtualScroll.value;
      return Vue.createVNode(Vue.Fragment, null, [propsHeaderContent ? Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__header")
      }, [propsHeaderContent]) : null, isVirtualScroll2 ? Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
        "style": cursorStyle.value
      }, null), Vue.createVNode("ul", {
        "class": "".concat(COMPONENT_NAME.value, "__inner"),
        "style": listStyle.value
      }, [virtualConfig.visibleData.value.map(function (item) {
        return Vue.createVNode(Vue.Fragment, null, [Vue.createVNode(list_listItem["default"], omit.omit(item, "slots"), item.slots)]);
      })])]) : Vue.createVNode("ul", {
        "class": "".concat(COMPONENT_NAME.value, "__inner")
      }, [renderTNodeJSX("default")]), propsFooterContent ? Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "__footer")
      }, [propsFooterContent]) : null]);
    };
    var handleScroll = function handleScroll(e) {
      var _props2$onScroll;
      var listElement = e.target;
      var scrollTop = listElement.scrollTop,
        scrollHeight = listElement.scrollHeight,
        clientHeight = listElement.clientHeight;
      if (isVirtualScroll.value) onInnerVirtualScroll(e);
      (_props2$onScroll = props2.onScroll) === null || _props2$onScroll === void 0 || _props2$onScroll.call(props2, {
        e: e,
        scrollTop: scrollTop,
        scrollBottom: scrollHeight - clientHeight - scrollTop
      });
    };
    var loadingClass = Vue.computed(function () {
      return isString.isString(props2.asyncLoading) && ["loading", "load-more"].includes(props2.asyncLoading) ? "".concat(COMPONENT_NAME.value, "__load ").concat(COMPONENT_NAME.value, "__load--").concat(props2.asyncLoading) : "".concat(COMPONENT_NAME.value, "__load");
    });
    var renderLoading = function renderLoading() {
      if (props2.asyncLoading && isString.isString(props2.asyncLoading)) {
        if (props2.asyncLoading === list_constants_index.LOADING) {
          return Vue.createVNode("div", null, [Vue.createVNode(loading_index.Loading, null, null), Vue.createVNode("span", null, [globalConfig.value.loadingText])]);
        }
        if (props2.asyncLoading === list_constants_index.LOAD_MORE) {
          return Vue.createVNode("span", null, [globalConfig.value.loadingMoreText]);
        }
      }
      return renderTNodeJSX("asyncLoading");
    };
    var handleLoadMore = function handleLoadMore(e) {
      var _props2$onLoadMore;
      if (isString.isString(props2.asyncLoading) && props2.asyncLoading !== list_constants_index.LOAD_MORE) return;
      (_props2$onLoadMore = props2.onLoadMore) === null || _props2$onLoadMore === void 0 || _props2$onLoadMore.call(props2, {
        e: e
      });
    };
    expose({
      scrollTo: scrollToElement
    });
    return function () {
      var listContent = [renderContent(), Vue.createVNode("div", {
        "class": loadingClass.value,
        "onClick": handleLoadMore
      }, [renderLoading()])];
      return Vue.createVNode("div", {
        "class": listClass.value,
        "onScroll": handleScroll,
        "ref": listRef,
        "style": isVirtualScroll.value ? "position:relative" : void 0
      }, [listContent]);
    };
  }
});

exports["default"] = _List;
