// tdesign-vue-next/es/tabs/tab-nav-bar.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, ref, computed, onMounted, nextTick, watch, createVNode } from 'vue';
import { _ as _defineProperty } from '../_chunks/dep-ba214d75.mjs';
import { _ as _slicedToArray } from '../_chunks/dep-f558ebb8.mjs';
import props from './props.mjs';
import { u as usePrefixClass, l as useResizeObserver } from '../_chunks/dep-fbeb2963.mjs';
import { debounce } from 'lodash-es';
import '../_chunks/dep-e1c62679.mjs';
import '../_chunks/dep-b0b422fe.mjs';
import '../_chunks/dep-47c3cae9.mjs';
import '../_chunks/dep-feed89df.mjs';
import '../_chunks/dep-5b510fcd.mjs';
import '../config-provider/hooks/useConfig.mjs';
import '../config-provider/utils/context.mjs';
import '../_chunks/dep-e0ab6d55.mjs';
import '../_chunks/dep-75f265c4.mjs';
import '../_chunks/dep-67ffcfeb.mjs';
import '../_chunks/dep-edab0e6a.mjs';
import '../_chunks/dep-76dddedf.mjs';
import '../_chunks/dep-d92220c4.mjs';
import '../_chunks/dep-466a8f31.mjs';
import '../_chunks/dep-e66c0337.mjs';

var TTabNavBar = defineComponent({
  props: {
    navs: {
      type: Array
    },
    placement: props.placement,
    value: props.value
  },
  setup: function setup(props) {
    var COMPONENT_NAME = usePrefixClass("tabs");
    var classPrefix = usePrefixClass();
    var barRef = ref();
    var navBarClass = computed(function () {
      return ["".concat(COMPONENT_NAME.value, "__bar"), "".concat(classPrefix.value, "-is-").concat(props.placement)];
    });
    var navBarStyle = ref(null);
    var getStyle = function getStyle() {
      var isVertical = ["left", "right"].includes(props.placement.toLowerCase());
      var _ref = isVertical ? ["height", "top"] : ["width", "left"],
        _ref2 = _slicedToArray(_ref, 2),
        sizePropName = _ref2[0],
        offsetPropName = _ref2[1];
      var offset = 0;
      var i = 0;
      for (; i < props.navs.length; i++) {
        var _props$navs$i;
        if (props.navs[i].props.value === props.value) {
          break;
        }
        if ((_props$navs$i = props.navs[i]) !== null && _props$navs$i !== void 0 && _props$navs$i.el) {
          var sizeWithUnit = getComputedStyle(props.navs[i].el)[sizePropName];
          var size = parseFloat(sizeWithUnit);
          offset += size;
        }
      }
      if (!props.navs[i]) return {};
      return _defineProperty(_defineProperty({}, offsetPropName, "".concat(offset, "px")), sizePropName, props.navs[i].el ? getComputedStyle(props.navs[i].el)[sizePropName] : "0px");
    };
    var update = function update() {
      return navBarStyle.value = getStyle();
    };
    onMounted(function () {
      nextTick(function () {
        update();
      });
    });
    watch([function () {
      return props.navs;
    }, function () {
      return props.value;
    }, function () {
      return props.placement;
    }], function () {
      nextTick(function () {
        update();
      });
    });
    useResizeObserver(barRef, debounce(function () {
      update();
    }, 35));
    return function () {
      return createVNode("div", {
        "class": navBarClass.value,
        "style": navBarStyle.value,
        "ref": barRef
      }, null);
    };
  }
});

export { TTabNavBar as default };
