// tdesign-vue-next/cjs/typography/text.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _typeof = require('@babel/runtime/helpers/typeof');
require('@babel/runtime/helpers/toConsumableArray');
require('../_chunks/dep-35e23c2b.js');
var index$1 = require('../_chunks/dep-1862d466.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/defineProperty');
var typography_textProps = require('./text-props.js');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var typography_components_ellipsis = require('./components/ellipsis.js');
var tooltip_index = require('../tooltip/index.js');
var button_index = require('../button/index.js');
var configProvider_hooks_useConfig = require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-4a7162e7.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('./paragraph-props.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-38acad3d.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-ce9c15f5.js');
require('../popup/container.js');
require('../_chunks/dep-e2122882.js');
require('../_chunks/dep-9bee89d3.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-79476fcb.js');
require('../_chunks/dep-e3edff33.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../tooltip/utils/index.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');
require('../button/button.js');
require('../loading/index.js');
require('../_chunks/dep-34cd76d4.js');
require('../loading/plugin.js');
require('../_chunks/dep-d632b788.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');
require('../loading/props.js');
require('../_chunks/dep-43dcd2db.js');
require('../_chunks/dep-392dde71.js');
require('../_chunks/dep-02ea4f3c.js');
require('../_chunks/dep-5103206f.js');
require('../_chunks/dep-7707ffe2.js');
require('../_chunks/dep-74a5bebe.js');
require('../_chunks/dep-42740075.js');
require('../_chunks/dep-9ac5c6ca.js');
require('../button/props.js');
require('../_chunks/dep-362c5212.js');
require('../_chunks/dep-74b143e0.js');
require('../_chunks/dep-719a33c8.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _typeof__default = /*#__PURE__*/_interopDefaultLegacy(_typeof);

var defaultMessage = "Copy to clipboard: #{key}, Enter";
var format = function format(message) {
  var copyKey = (/mac os x/i.test(navigator.userAgent) ? "\u2318" : "Ctrl") + "+C";
  return message.replace(/#{\s*key\s*}/g, copyKey);
};
var deselectCurrent = function deselectCurrent() {
  var selection = document.getSelection();
  if (!selection.rangeCount) {
    return function () {};
  }
  var active = document.activeElement;
  var ranges = [];
  for (var i = 0; i < selection.rangeCount; i++) {
    ranges.push(selection.getRangeAt(i));
  }
  var tagName = active.tagName.toUpperCase();
  switch (tagName) {
    case "INPUT":
    case "TEXTAREA":
      active.blur();
      break;
    default:
      active = null;
      break;
  }
  selection.removeAllRanges();
  return function () {
    selection.type === "Caret" && selection.removeAllRanges();
    if (!selection.rangeCount) {
      ranges.forEach(function (range) {
        selection.addRange(range);
      });
    }
    active && active.focus();
  };
};
var copy = function copy(text, options) {
  var message,
    reselectPrevious,
    range,
    selection,
    mark,
    success = false;
  if (!options) {
    options = {};
  }
  try {
    reselectPrevious = deselectCurrent();
    range = document.createRange();
    selection = document.getSelection();
    mark = document.createElement("span");
    mark.textContent = text;
    mark.style.all = "unset";
    mark.style.position = "fixed";
    mark.style.top = "0";
    mark.style.clip = "rect(0, 0, 0, 0)";
    mark.style.whiteSpace = "pre";
    mark.style.webkitUserSelect = "text";
    mark.style.userSelect = "text";
    mark.addEventListener("copy", function (e) {
      e.stopPropagation();
      if (options.format) {
        e.preventDefault();
        e.clipboardData.clearData();
        e.clipboardData.setData(options.format, text);
      }
      if (options.onCopy) {
        e.preventDefault();
        options.onCopy(e.clipboardData);
      }
    });
    document.body.appendChild(mark);
    range.selectNodeContents(mark);
    selection.addRange(range);
    var successful = document.execCommand("copy");
    if (!successful) {
      throw new Error("copy command was unsuccessful");
    }
    success = true;
  } catch (err) {
    try {
      window.clipboardData.setData(options.format || "text", text);
      options.onCopy && options.onCopy(window.clipboardData);
      success = true;
    } catch (err2) {
      message = format("message" in options ? options.message : defaultMessage);
      window.prompt(message, text);
    }
  } finally {
    if (selection) {
      if (typeof selection.removeRange == "function") {
        selection.removeRange(range);
      } else {
        selection.removeAllRanges();
      }
    }
    if (mark) {
      document.body.removeChild(mark);
    }
    reselectPrevious();
  }
  return success;
};

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var _Text = Vue.defineComponent({
  name: "TTypographyText",
  props: typography_textProps["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var COMPONENT_NAME = index.usePrefixClass("typography");
    var _useConfig = configProvider_hooks_useConfig.useConfig("typography"),
      globalConfig = _useConfig.globalConfig;
    var isCopied = Vue.ref(false);
    var renderContent = index$1.useContent();
    var wrapperDecorations = function wrapperDecorations(_ref2, content2) {
      var code = _ref2.code,
        underline = _ref2.underline,
        del = _ref2["delete"],
        strong = _ref2.strong,
        keyboard = _ref2.keyboard,
        mark = _ref2.mark,
        italic = _ref2.italic;
      var currentContent = content2;
      function wrap(needed, Tag) {
        var styles = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
        if (!needed) return;
        currentContent = Vue.h(Tag, {
          style: styles
        }, [currentContent]);
      }
      wrap(strong, "strong");
      wrap(underline, "u");
      wrap(del, "del");
      wrap(code, "code");
      wrap(mark !== false, "mark", mark ? {
        backgroundColor: mark
      } : {});
      wrap(keyboard, "kbd");
      wrap(italic, "i");
      return currentContent;
    };
    var classList = Vue.computed(function () {
      var theme = props2.theme,
        disabled = props2.disabled;
      var prefix = COMPONENT_NAME.value;
      var list = [prefix];
      if (disabled) {
        list.push("".concat(prefix, "--disabled"));
      } else if (theme && ["primary", "secondary", "success", "warning", "error"].includes(theme)) {
        list.push("".concat(prefix, "--").concat(theme));
      }
      return list;
    });
    var tooltipText = Vue.computed(function () {
      var _copyable$tooltipProp;
      var copyable = props2.copyable;
      if (isCopied.value) return globalConfig.value.copiedText;else if (_typeof__default["default"](copyable) === "object") return (_copyable$tooltipProp = copyable.tooltipProps) === null || _copyable$tooltipProp === void 0 ? void 0 : _copyable$tooltipProp.content;
      return null;
    });
    var content = Vue.computed(function () {
      return props2.content || (slots === null || slots === void 0 ? void 0 : slots["default"]());
    });
    var renderCopy = function renderCopy(afterEllipsis) {
      var copyable = props2.copyable;
      var icon = isCopied.value ? function () {
        return Vue.createVNode(tdesignIconsVueNext.CheckIcon, null, null);
      } : function () {
        return Vue.createVNode(tdesignIconsVueNext.CopyIcon, null, null);
      };
      var tooltipConf = {
        theme: "default"
      };
      var onCopy = function onCopy() {};
      if (_typeof__default["default"](copyable) === "object") {
        if (copyable.suffix && !isCopied.value) {
          icon = copyable.suffix;
        }
        if (copyable.tooltipProps) {
          tooltipConf = copyable.tooltipProps;
        }
        if (typeof copyable.onCopy === "function") {
          onCopy = copyable.onCopy;
        }
      }
      return Vue.createVNode(tooltip_index.Tooltip, Vue.mergeProps(tooltipConf, {
        "content": tooltipText.value
      }), {
        "default": function _default() {
          return [afterEllipsis ? Vue.createVNode("span", {
            "onClick": function onClick(e) {
              return onCopyClick(e, onCopy);
            }
          }, [icon()]) : Vue.createVNode(button_index.Button, {
            "icon": icon,
            "shape": "square",
            "theme": "primary",
            "variant": "text",
            "onClick": function onClick(e) {
              return onCopyClick(e, onCopy);
            }
          }, null)];
        }
      });
    };
    var getChildrenText = function getChildrenText() {
      var copyable = props2.copyable;
      if (_typeof__default["default"](copyable) === "object" && copyable !== null && copyable !== void 0 && copyable.text) {
        return copyable.text;
      }
      if (typeof content.value === "string") {
        return content.value;
      } else if (Array.isArray(content.value)) {
        return content.value.map(function (v) {
          return v.children;
        }).join("");
      }
    };
    var onCopyClick = function onCopyClick(e, cb) {
      e.preventDefault();
      e.stopPropagation();
      isCopied.value = true;
      setTimeout(function () {
        isCopied.value = false;
      }, 1500);
      copy(getChildrenText());
      cb === null || cb === void 0 || cb();
    };
    return function () {
      var _slot;
      var content2 = renderContent("default", "content");
      return props2.ellipsis ? Vue.createVNode(typography_components_ellipsis["default"], Vue.mergeProps(props2, {
        "class": classList.value,
        "renderCopy": props2.copyable ? function () {
          return renderCopy(true);
        } : null
      }), _isSlot(_slot = wrapperDecorations(props2, content2)) ? _slot : {
        "default": function _default() {
          return [_slot];
        }
      }) : Vue.createVNode("span", {
        "class": classList.value
      }, [wrapperDecorations(props2, content2), props2.copyable ? renderCopy() : null]);
    };
  }
});

exports["default"] = _Text;
