// tdesign-vue-next/cjs/color-picker/components/panel/index.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var color = require('../../../_chunks/dep-34c8efe7.js');
var constants = require('../../../_chunks/dep-9d53c643.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
var format = require('../../../_chunks/dep-15083282.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-35e23c2b.js');
var index = require('../../../_chunks/dep-38acad3d.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
var index$2 = require('../../../_chunks/dep-0afce170.js');
require('../../../_chunks/dep-0989c3be.js');
var index$1 = require('../../../_chunks/dep-ce9c15f5.js');
var colorPicker_colorPickerPanelProps = require('../../color-picker-panel-props.js');
var colorPicker_hooks_index = require('../../hooks/index.js');
var colorPicker_components_format_index = require('../format/index.js');
var colorPicker_components_panel_alpha = require('./alpha.js');
var colorPicker_components_panel_header = require('./header.js');
var colorPicker_components_panel_hue = require('./hue.js');
var colorPicker_components_panel_linearGradient = require('./linear-gradient.js');
var colorPicker_components_panel_saturation = require('./saturation.js');
var colorPicker_components_panel_swatches = require('./swatches.js');
var isUndefined = require('../../../_chunks/dep-3d29e358.js');
var isNull = require('../../../_chunks/dep-6212811d.js');
var cloneDeep = require('../../../_chunks/dep-2fc884aa.js');
require('tinycolor2');
require('../../../_chunks/dep-631750cd.js');
require('../../../_chunks/dep-22dc294c.js');
require('../../../_chunks/dep-4cb26289.js');
require('../../../_chunks/dep-4a7162e7.js');
require('../../../_chunks/dep-be1af85d.js');
require('../../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../../_chunks/dep-82fe2026.js');
require('../../../_chunks/dep-33b46a52.js');
require('../../../_chunks/dep-930a2de5.js');
require('../../../_chunks/dep-2dcf9237.js');
require('../../../_chunks/dep-bf76dead.js');
require('../../../_chunks/dep-c3bbd06c.js');
require('../../../_chunks/dep-6d87f74d.js');
require('../../../_chunks/dep-281f7eb2.js');
require('../../../_chunks/dep-f8f7b1af.js');
require('../../../_chunks/dep-66d3f30f.js');
require('../../../_chunks/dep-e1d22111.js');
require('../../../_chunks/dep-754c0523.js');
require('../../../_chunks/dep-7076a08a.js');
require('../../../_chunks/dep-75d6da05.js');
require('../../../_chunks/dep-274369eb.js');
require('../../../_chunks/dep-df6e14a0.js');
require('../../../_chunks/dep-9f98bf9b.js');
require('../../../_chunks/dep-b2024592.js');
require('../../../_chunks/dep-2fb8d44f.js');
require('../../../_chunks/dep-80a41429.js');
require('../../../_chunks/dep-0238ed9a.js');
require('../../props.js');
require('../../../select/index.js');
require('../../../select/select.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../../common-components/fake-arrow.js');
require('../../../_chunks/dep-2aa3d3cf.js');
require('../../../_chunks/dep-44195fb2.js');
require('../../../select-input/index.js');
require('../../../select-input/select-input.js');
require('../../../popup/index.js');
require('../../../popup/popup.js');
require('@popperjs/core');
require('../../../_chunks/dep-8d990e66.js');
require('../../../_chunks/dep-1862d466.js');
require('../../../_chunks/dep-a6551cc4.js');
require('../../../_chunks/dep-7d9abefd.js');
require('../../../_chunks/dep-91305690.js');
require('../../../_chunks/dep-0b92e695.js');
require('../../../_chunks/dep-4ba41627.js');
require('../../../popup/container.js');
require('../../../popup/props.js');
require('../../../_chunks/dep-e2122882.js');
require('../../../_chunks/dep-9bee89d3.js');
require('../../../_chunks/dep-027cd76f.js');
require('../../../_chunks/dep-ae4040c5.js');
require('../../../_chunks/dep-79476fcb.js');
require('../../../_chunks/dep-e3edff33.js');
require('../../../select-input/props.js');
require('../../../select-input/hooks/useMultiple.js');
require('../../../tag-input/index.js');
require('../../../tag-input/tag-input.js');
require('tdesign-icons-vue-next');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-362c5212.js');
require('../../../_chunks/dep-74b143e0.js');
require('../../../_chunks/dep-c26a7d50.js');
require('../../../_chunks/dep-9dbade68.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-692bdebe.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../tag-input/props.js');
require('../../../_chunks/dep-a13f0135.js');
require('../../../tag-input/hooks/useDragSorter.js');
require('../../../tag-input/hooks/useHover.js');
require('../../../tag-input/hooks/useTagScroll.js');
require('../../../tag-input/hooks/useTagList.js');
require('../../../tag/index.js');
require('../../../tag/tag.js');
require('../../../tag/props.js');
require('../../../tag/check-tag.js');
require('../../../tag/check-tag-props.js');
require('../../../_chunks/dep-a4be1915.js');
require('../../../tag/check-tag-group.js');
require('../../../tag/check-tag-group-props.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-34cd76d4.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-d632b788.js');
require('../../../loading/icon/gradient.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-43dcd2db.js');
require('../../../_chunks/dep-392dde71.js');
require('../../../_chunks/dep-02ea4f3c.js');
require('../../../_chunks/dep-5103206f.js');
require('../../../_chunks/dep-7707ffe2.js');
require('../../../_chunks/dep-74a5bebe.js');
require('../../../_chunks/dep-4eca9add.js');
require('../../../_chunks/dep-42740075.js');
require('../../../_chunks/dep-9ac5c6ca.js');
require('../../../select-input/hooks/useOverlayInnerStyle.js');
require('../../../select-input/hooks/useSingle.js');
require('../../../_chunks/dep-eb3114d9.js');
require('../../../_chunks/dep-690970a4.js');
require('../../../_chunks/dep-7870f4cb.js');
require('../../../select/components/select-panel.js');
require('../../../select/option.js');
require('../../../select/option-props.js');
require('../../../checkbox/index.js');
require('../../../checkbox/checkbox.js');
require('../../../checkbox/props.js');
require('../../../_chunks/dep-719a33c8.js');
require('../../../checkbox/constants/index.js');
require('../../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../../_chunks/dep-b9a88912.js');
require('../../../checkbox/hooks/useKeyboardEvent.js');
require('../../../checkbox/group.js');
require('../../../checkbox/checkbox-group-props.js');
require('../../../_chunks/dep-0cda9fe9.js');
require('../../../_chunks/dep-b0334c78.js');
require('../../../_chunks/dep-a214821d.js');
require('../../../select/utils/index.js');
require('../../../select/constants/index.js');
require('../../../select/option-group.js');
require('../../../select/option-group-props.js');
require('../../../select/props.js');
require('../../../select/hooks/usePanelVirtualScroll.js');
require('../../../_chunks/dep-5c79e9ef.js');
require('../../../_chunks/dep-9a54a293.js');
require('../../../select/hooks/useKeyboardControl.js');
require('../../../select/hooks/useSelectOptions.js');
require('../../../_chunks/dep-1c1b8615.js');
require('../format/inputs.js');
require('../../../input-number/index.js');
require('../../../input-number/input-number.js');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../button/props.js');
require('../../../input-number/props.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-6bac596c.js');
require('../../../_chunks/dep-73734701.js');
require('../../../_chunks/dep-27729d32.js');
require('./slider.js');
require('../../../_chunks/dep-f0f81082.js');
require('./base-props.js');
require('../../../radio/index.js');
require('../../../radio/radio.js');
require('../../../radio/props.js');
require('../../../radio/constants/index.js');
require('../../../radio/group.js');
require('../../../radio/radio-group-props.js');
require('../../../radio/radio-button.js');
require('../../../radio/hooks/useKeyboard.js');
require('../../../watermark/hooks/index.js');
require('../../../_chunks/dep-95977a93.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ColorPanel = Vue.defineComponent({
  name: "ColorPanel",
  props: _objectSpread({}, colorPicker_colorPickerPanelProps["default"]),
  setup: function setup(props2) {
    var baseClassName = colorPicker_hooks_index.useBaseClassName();
    var _useCommonClassName = index.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var _useConfig = configProvider_hooks_useConfig.useConfig("colorPicker"),
      t = _useConfig.t,
      globalConfig = _useConfig.globalConfig;
    var statusClassNames = STATUS.value;
    var _toRefs = Vue.toRefs(props2),
      inputValue = _toRefs.value,
      modelValue = _toRefs.modelValue,
      recentColors = _toRefs.recentColors;
    var _useVModel = index$1.useVModel(inputValue, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      innerValue = _useVModel2[0],
      setInnerValue = _useVModel2[1];
    var _useDefaultValue = index$2.useDefaultValue(recentColors, props2.defaultRecentColors, props2.onRecentColorsChange, "recentColors"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 2),
      innerRecentColors = _useDefaultValue2[0],
      setInnerRecentColors = _useDefaultValue2[1];
    var getModeByColor = function getModeByColor(input) {
      if (props2.colorModes.length === 1) return props2.colorModes[0];
      return props2.colorModes.includes("linear-gradient") && color.Color.isGradientColor(input) ? "linear-gradient" : "monochrome";
    };
    var mode = Vue.ref(getModeByColor(innerValue.value));
    var isGradient = Vue.computed(function () {
      return mode.value === "linear-gradient";
    });
    var defaultEmptyColor = Vue.computed(function () {
      return isGradient.value ? constants.DEFAULT_LINEAR_GRADIENT : constants.DEFAULT_COLOR;
    });
    var color$1 = Vue.ref(new color.Color(innerValue.value || defaultEmptyColor.value));
    var formatModel = Vue.ref(format.initColorFormat(props2.format, props2.enableAlpha));
    var addRecentlyUsedColor = function addRecentlyUsedColor() {
      if (innerRecentColors.value === null || innerRecentColors.value === false) {
        return;
      }
      var colors = cloneDeep.cloneDeep(innerRecentColors.value) || [];
      var currentColor = color$1.value.isGradient ? color$1.value.linearGradient : color$1.value.rgba;
      var index = colors.indexOf(currentColor);
      if (index > -1) {
        colors.splice(index, 1);
      }
      colors.unshift(currentColor);
      if (colors.length > constants.TD_COLOR_USED_COLORS_MAX_SIZE) {
        colors.length = constants.TD_COLOR_USED_COLORS_MAX_SIZE;
      }
      handleRecentlyUsedColorsChange(colors);
    };
    var handleRecentlyUsedColorsChange = function handleRecentlyUsedColorsChange(colors) {
      setInnerRecentColors(colors);
    };
    var emitColorChange = function emitColorChange(trigger) {
      var value = color$1.value.getFormattedColor(props2.format, props2.enableAlpha);
      setInnerValue(value, {
        color: color.getColorObject(color$1.value),
        trigger: trigger || "palette-saturation-brightness"
      });
    };
    Vue.watch(function () {
      return innerValue.value;
    }, function (newColor) {
      var newMode = getModeByColor(newColor);
      mode.value = newMode;
      color$1.value.isGradient = newMode === "linear-gradient";
      var currentColor = color$1.value.getFormattedColor(props2.format, props2.enableAlpha);
      if (currentColor !== newColor) {
        color$1.value.update(newColor);
      }
    });
    var handleModeChange = function handleModeChange(newMode) {
      mode.value = newMode;
      var isGradientMode = newMode === "linear-gradient";
      color$1.value.isGradient = isGradientMode;
      if (isGradientMode) {
        color$1.value.update(color$1.value.gradientColors.length > 0 ? color$1.value.linearGradient : constants.DEFAULT_LINEAR_GRADIENT);
      } else {
        color$1.value.update(color$1.value.rgba);
      }
      emitColorChange();
    };
    var handleSatAndValueChange = function handleSatAndValueChange(_ref) {
      var saturation = _ref.saturation,
        value = _ref.value;
      var _color$value = color$1.value,
        sat = _color$value.saturation,
        val = _color$value.value;
      var changeTrigger = "palette-saturation-brightness";
      if (value !== val && saturation !== sat) {
        color$1.value.saturation = saturation;
        color$1.value.value = value;
        changeTrigger = "palette-saturation-brightness";
      } else if (saturation !== sat) {
        color$1.value.saturation = saturation;
        changeTrigger = "palette-saturation";
      } else if (value !== val) {
        color$1.value.value = value;
        changeTrigger = "palette-brightness";
      } else {
        return;
      }
      emitColorChange(changeTrigger);
    };
    var handleHueChange = function handleHueChange(hue) {
      var _props2$onPaletteBarC;
      color$1.value.hue = hue;
      emitColorChange("palette-hue-bar");
      (_props2$onPaletteBarC = props2.onPaletteBarChange) === null || _props2$onPaletteBarC === void 0 || _props2$onPaletteBarC.call(props2, {
        color: color.getColorObject(color$1.value)
      });
    };
    var handleAlphaChange = function handleAlphaChange(alpha) {
      color$1.value.alpha = alpha;
      emitColorChange("palette-alpha-bar");
    };
    var handleInputChange = function handleInputChange(input, alpha) {
      color$1.value.update(input);
      color$1.value.alpha = alpha;
      emitColorChange("input");
    };
    var handleGradientChange = function handleGradientChange(_ref2) {
      var key = _ref2.key,
        payload = _ref2.payload;
      var trigger = "palette-saturation-brightness";
      switch (key) {
        case "degree":
          color$1.value.gradientDegree = payload;
          trigger = "input";
          break;
        case "selectedId":
          color$1.value.gradientSelectedId = payload;
          break;
        case "colors":
          color$1.value.gradientColors = payload;
          break;
      }
      emitColorChange(trigger);
    };
    var handleSetColor = function handleSetColor(value, trigger) {
      var newMode = getModeByColor(value);
      mode.value = newMode;
      color$1.value.isGradient = newMode === "linear-gradient";
      color$1.value.update(value);
      emitColorChange(trigger);
    };
    return function () {
      var baseProps = {
        color: color$1.value,
        disabled: props2.disabled
      };
      var onlySupportGradient = props2.colorModes.length === 1 && props2.colorModes.includes("linear-gradient");
      var recentColors2 = innerRecentColors.value;
      if (onlySupportGradient && Array.isArray(recentColors2)) {
        recentColors2 = recentColors2.filter(function (color2) {
          return color.Color.isGradientColor(color2);
        });
      }
      var showUsedColors = !!Array.isArray(recentColors2);
      var systemColors = props2.swatchColors;
      if (isUndefined.isUndefined(systemColors)) {
        systemColors = _toConsumableArray__default["default"](constants.DEFAULT_SYSTEM_SWATCH_COLORS);
      }
      if (isNull.isNull(systemColors)) {
        systemColors = [];
      }
      if (onlySupportGradient) {
        systemColors = systemColors.filter(function (color2) {
          return color.Color.isGradientColor(color2);
        });
      }
      var showSystemColors = Array.isArray(systemColors) && systemColors.length;
      var renderSwatches = function renderSwatches() {
        if (!showSystemColors && !showUsedColors) return null;
        return Vue.createVNode(Vue.Fragment, null, [Vue.createVNode("div", {
          "class": "".concat(baseClassName.value, "__swatches-wrap")
        }, [showUsedColors ? Vue.createVNode(colorPicker_components_panel_swatches["default"], Vue.mergeProps(baseProps, {
          "title": t(globalConfig.value.recentColorTitle),
          "editable": true,
          "colors": recentColors2,
          "handleAddColor": addRecentlyUsedColor,
          "onSetColor": function onSetColor(color2) {
            return handleSetColor(color2, "recent");
          },
          "onChange": handleRecentlyUsedColorsChange
        }), null) : null, showSystemColors ? Vue.createVNode(colorPicker_components_panel_swatches["default"], Vue.mergeProps(baseProps, {
          "title": t(globalConfig.value.swatchColorTitle),
          "colors": systemColors,
          "onSetColor": function onSetColor(color2) {
            return handleSetColor(color2, "preset");
          }
        }), null) : null])]);
      };
      return Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__panel"), props2.disabled ? statusClassNames.disabled : false]
      }, [Vue.createVNode(colorPicker_components_panel_header["default"], Vue.mergeProps(props2, {
        "mode": mode.value,
        "onModeChange": handleModeChange
      }), null), Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__body")]
      }, [isGradient.value ? Vue.createVNode(colorPicker_components_panel_linearGradient["default"], Vue.mergeProps(baseProps, {
        "onChange": handleGradientChange,
        "enableMultipleGradient": props2.enableMultipleGradient
      }), null) : null, Vue.createVNode(colorPicker_components_panel_saturation["default"], Vue.mergeProps(baseProps, {
        "onChange": handleSatAndValueChange
      }), null), Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__sliders-wrapper")]
      }, [Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__sliders")]
      }, [Vue.createVNode(colorPicker_components_panel_hue["default"], Vue.mergeProps(baseProps, {
        "onChange": handleHueChange
      }), null), props2.enableAlpha ? Vue.createVNode(colorPicker_components_panel_alpha["default"], Vue.mergeProps(baseProps, {
        "onChange": handleAlphaChange
      }), null) : null]), props2.showPrimaryColorPreview ? Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__sliders-preview"), "".concat(baseClassName.value, "--bg-alpha")]
      }, [Vue.createVNode("span", {
        "class": "".concat(baseClassName.value, "__sliders-preview-inner"),
        "style": {
          background: isGradient.value ? color$1.value.linearGradient : color$1.value.rgba
        }
      }, null)]) : null]), Vue.createVNode(colorPicker_components_format_index["default"], Vue.mergeProps(props2, {
        "color": color$1.value,
        "format": formatModel.value,
        "onInputChange": handleInputChange
      }), null), renderSwatches()])]);
    };
  }
});

exports["default"] = ColorPanel;
