// tdesign-vue-next/cjs/color-picker/components/panel/linear-gradient.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var constants = require('../../../_chunks/dep-9d53c643.js');
var color = require('../../../_chunks/dep-34c8efe7.js');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/toConsumableArray');
var inputNumber_index = require('../../../input-number/index.js');
var colorPicker_hooks_index = require('../../hooks/index.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-35e23c2b.js');
var index = require('../../../_chunks/dep-38acad3d.js');
require('../../../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-0989c3be.js');
var colorPicker_components_panel_baseProps = require('./base-props.js');
var cloneDeep = require('../../../_chunks/dep-2fc884aa.js');
require('tinycolor2');
require('../../../_chunks/dep-631750cd.js');
require('../../../_chunks/dep-22dc294c.js');
require('../../../_chunks/dep-4cb26289.js');
require('../../../_chunks/dep-6212811d.js');
require('../../../input-number/input-number.js');
require('tdesign-icons-vue-next');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-34cd76d4.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-d632b788.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-0238ed9a.js');
require('../../../_chunks/dep-4ba41627.js');
require('../../../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-3d29e358.js');
require('../../../_chunks/dep-44195fb2.js');
require('../../../_chunks/dep-4a7162e7.js');
require('../../../_chunks/dep-be1af85d.js');
require('../../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../../_chunks/dep-82fe2026.js');
require('../../../_chunks/dep-33b46a52.js');
require('../../../_chunks/dep-930a2de5.js');
require('../../../_chunks/dep-2dcf9237.js');
require('../../../_chunks/dep-bf76dead.js');
require('../../../_chunks/dep-c3bbd06c.js');
require('../../../_chunks/dep-6d87f74d.js');
require('../../../_chunks/dep-281f7eb2.js');
require('../../../_chunks/dep-f8f7b1af.js');
require('../../../_chunks/dep-66d3f30f.js');
require('../../../_chunks/dep-e1d22111.js');
require('../../../_chunks/dep-754c0523.js');
require('../../../_chunks/dep-7076a08a.js');
require('../../../_chunks/dep-75d6da05.js');
require('../../../_chunks/dep-274369eb.js');
require('../../../_chunks/dep-df6e14a0.js');
require('../../../_chunks/dep-8d990e66.js');
require('../../../_chunks/dep-1862d466.js');
require('../../../_chunks/dep-a6551cc4.js');
require('../../../_chunks/dep-7d9abefd.js');
require('../../../_chunks/dep-2fb8d44f.js');
require('../../../_chunks/dep-80a41429.js');
require('../../../_chunks/dep-91305690.js');
require('../../../_chunks/dep-0b92e695.js');
require('../../../_chunks/dep-b2024592.js');
require('../../../_chunks/dep-9f98bf9b.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-43dcd2db.js');
require('../../../_chunks/dep-392dde71.js');
require('../../../_chunks/dep-02ea4f3c.js');
require('../../../_chunks/dep-5103206f.js');
require('../../../_chunks/dep-7707ffe2.js');
require('../../../_chunks/dep-74a5bebe.js');
require('../../../_chunks/dep-4eca9add.js');
require('../../../_chunks/dep-42740075.js');
require('../../../_chunks/dep-9ac5c6ca.js');
require('../../../_chunks/dep-e3edff33.js');
require('../../../button/props.js');
require('../../../_chunks/dep-362c5212.js');
require('../../../_chunks/dep-74b143e0.js');
require('../../../_chunks/dep-719a33c8.js');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-c26a7d50.js');
require('../../../_chunks/dep-9dbade68.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../_chunks/dep-ce9c15f5.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-692bdebe.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-e2122882.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../input-number/props.js');
require('../../../_chunks/dep-a13f0135.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-6bac596c.js');
require('../../../_chunks/dep-73734701.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var DELETE_KEYS = ["delete", "backspace"];
var LinearGradient = Vue.defineComponent({
  name: "LinearGradient",
  inheritAttrs: false,
  props: _objectSpread(_objectSpread({}, colorPicker_components_panel_baseProps["default"]), {}, {
    enableMultipleGradient: {
      type: Boolean,
      "default": true
    }
  }),
  setup: function setup(props) {
    var baseClassName = colorPicker_hooks_index.useBaseClassName();
    var _useCommonClassName = index.useCommonClassName(),
      STATUS = _useCommonClassName.STATUS;
    var statusClassNames = STATUS.value;
    var refSlider = Vue.ref(null);
    var sliderRect = Vue.reactive({
      left: 0,
      width: constants.GRADIENT_SLIDER_DEFAULT_WIDTH
    });
    var isDragging = Vue.ref(false);
    var isMoved = Vue.ref(false);
    var degree = Vue.ref(props.color.gradientDegree);
    var selectedId = Vue.ref(props.color.gradientSelectedId);
    var colors = Vue.ref(cloneDeep.cloneDeep(props.color.gradientColors));
    Vue.watch(function () {
      return props.color.gradientDegree;
    }, function (value) {
      return degree.value = value;
    });
    Vue.watch(function () {
      return props.color.gradientSelectedId;
    }, function (value) {
      return selectedId.value = value;
    });
    Vue.watch(function () {
      return props.color.gradientColors;
    }, function (value) {
      colors.value = cloneDeep.cloneDeep(value);
    }, {
      deep: true
    });
    var handleChange = function handleChange(key, payload, addUsedColor) {
      if (props.disabled) {
        return;
      }
      props.onChange({
        key: key,
        payload: payload,
        addUsedColor: addUsedColor
      });
    };
    var handleDegreeChange = function handleDegreeChange(value) {
      if (props.disabled || value === props.color.gradientDegree) {
        return;
      }
      degree.value = value;
      handleChange("degree", value, true);
    };
    var handleSelectedIdChange = function handleSelectedIdChange(value) {
      if (props.disabled) {
        return;
      }
      selectedId.value = value;
      handleChange("selectedId", value);
    };
    var handleColorsChange = function handleColorsChange(value, isEnded) {
      if (props.disabled) {
        return;
      }
      colors.value = value;
      handleChange("colors", value, isEnded);
    };
    var updateActiveThumbLeft = function updateActiveThumbLeft(left) {
      var index = colors.value.findIndex(function (c) {
        return c.id === selectedId.value;
      });
      if (index === -1) {
        return;
      }
      var point = colors.value[index];
      left = Math.max(0, Math.min(sliderRect.width, left));
      var percentLeft = left / sliderRect.width * 100;
      colors.value.splice(index, 1, {
        color: point.color,
        left: percentLeft,
        id: point.id
      });
      handleColorsChange(colors.value);
    };
    var handleStart = function handleStart(id) {
      var rect = refSlider.value.getBoundingClientRect();
      sliderRect.left = rect.left;
      sliderRect.width = rect.width || constants.GRADIENT_SLIDER_DEFAULT_WIDTH;
      if (isDragging.value || props.disabled) {
        return;
      }
      isMoved.value = false;
      isDragging.value = true;
      handleSelectedIdChange(id);
      refSlider.value.focus();
      window.addEventListener("mousemove", handleMove, false);
      window.addEventListener("mouseup", _handleEnd, false);
      window.addEventListener("contextmenu", _handleEnd, false);
    };
    var handleMove = function handleMove(e) {
      if (!isDragging.value || props.disabled) {
        return;
      }
      var left = e.clientX - sliderRect.left;
      isMoved.value = true;
      updateActiveThumbLeft(left);
    };
    var _handleEnd = function handleEnd() {
      if (!isDragging.value) {
        return;
      }
      setTimeout(function () {
        isDragging.value = false;
      }, 0);
      if (isMoved.value) {
        handleColorsChange(colors.value, true);
        isMoved.value = false;
      }
      window.removeEventListener("mousemove", handleMove, false);
      window.removeEventListener("mouseup", _handleEnd, false);
      window.removeEventListener("contextmenu", _handleEnd, false);
    };
    var handleKeyup = function handleKeyup(e) {
      if (props.disabled) {
        return;
      }
      var points = colors.value;
      var pos = points.findIndex(function (c) {
        return c.id === selectedId.value;
      });
      var length = points.length;
      if (DELETE_KEYS.includes(e.key.toLocaleLowerCase()) && length > 2 && pos >= 0 && pos <= length - 1) {
        points.splice(pos, 1);
        if (!points[pos]) {
          pos = points[pos + 1] ? pos + 1 : points[pos - 1] ? pos - 1 : 0;
        }
        var current = points[pos];
        handleColorsChange(points, true);
        handleSelectedIdChange(current === null || current === void 0 ? void 0 : current.id);
      }
    };
    var handleThumbBarClick = function handleThumbBarClick(e) {
      if (props.disabled || !props.enableMultipleGradient) {
        return;
      }
      var left = e.offsetX;
      var percentLeft = left / sliderRect.width * 100;
      var newPoint = color.genGradientPoint(percentLeft, props.color.rgba);
      colors.value.push(newPoint);
      handleColorsChange(colors.value, true);
      handleSelectedIdChange(newPoint.id);
    };
    Vue.onMounted(function () {
      var rect = refSlider.value.getBoundingClientRect();
      sliderRect.left = rect.left;
      sliderRect.width = rect.width || constants.GRADIENT_SLIDER_DEFAULT_WIDTH;
    });
    Vue.onBeforeUnmount(function () {
      window.removeEventListener("mousemove", handleMove, false);
      window.removeEventListener("mouseup", _handleEnd, false);
      window.removeEventListener("contextmenu", _handleEnd, false);
    });
    return function () {
      var _props$color = props.color,
        linearGradient = _props$color.linearGradient,
        gradientColors = _props$color.gradientColors;
      var thumbBackground = color.gradientColors2string({
        points: gradientColors,
        degree: 90
      });
      return Vue.createVNode("div", {
        "class": "".concat(baseClassName.value, "__gradient")
      }, [Vue.createVNode("div", {
        "class": "".concat(baseClassName.value, "__gradient-slider")
      }, [Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__slider"), "".concat(baseClassName.value, "--bg-alpha")],
        "onKeyup": handleKeyup,
        "tabindex": 0,
        "ref": refSlider
      }, [Vue.createVNode("ul", {
        "class": "gradient-thumbs",
        "onClick": handleThumbBarClick,
        "style": {
          background: thumbBackground
        }
      }, [colors.value.map(function (t) {
        var left = "".concat(Math.round(t.left * 100) / 100, "%");
        return Vue.createVNode("li", {
          "class": ["".concat(baseClassName.value, "__thumb"), "gradient-thumbs__item", selectedId.value === t.id ? statusClassNames.active : ""],
          "key": t.id,
          "title": "".concat(t.color, " ").concat(left),
          "style": {
            color: t.color,
            left: left
          },
          "onClick": function onClick(e) {
            return e.stopPropagation();
          },
          "onMousedown": function onMousedown() {
            return handleStart(t.id);
          }
        }, [Vue.createVNode("span", {
          "class": ["gradient-thumbs__item-inner", "".concat(baseClassName.value, "--bg-alpha")]
        }, null)]);
      })])])]), Vue.createVNode("div", {
        "class": "".concat(baseClassName.value, "__gradient-degree"),
        "title": "".concat(degree.value, "deg")
      }, [Vue.createVNode(inputNumber_index.InputNumber, {
        "size": "small",
        "theme": "normal",
        "min": 0,
        "max": 360,
        "step": 1,
        "format": function format(value) {
          return "".concat(value, "\xB0");
        },
        "modelValue": degree.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return degree.value = $event;
        },
        "onBlur": handleDegreeChange,
        "onEnter": handleDegreeChange,
        "disabled": props.disabled
      }, null)]), Vue.createVNode("div", {
        "class": ["".concat(baseClassName.value, "__gradient-preview"), "".concat(baseClassName.value, "--bg-alpha")]
      }, [Vue.createVNode("span", {
        "class": "preview-inner",
        "style": {
          background: linearGradient
        }
      }, null)])]);
    };
  }
});

exports["default"] = LinearGradient;
