// tdesign-vue-next/esm/transfer/components/transfer-search.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, createVNode, mergeProps } from 'vue';
import _typeof from '@babel/runtime/helpers/typeof';
import { SearchIcon } from 'tdesign-icons-vue-next';
import { Input } from '../../input/index.js';
import 'lodash-es';
import '@babel/runtime/helpers/toConsumableArray';
import '../../_chunks/dep-c68ea098.js';
import { u as usePrefixClass } from '../../_chunks/dep-e8dd47a9.js';
import { u as useGlobalIcon } from '../../_chunks/dep-52eae58a.js';
import '@babel/runtime/helpers/slicedToArray';
import '../../_chunks/dep-91fc762d.js';
import '@babel/runtime/helpers/defineProperty';
import { u as useEventForward } from '../../_chunks/dep-cd9622b1.js';
import '../../input/input.js';
import '@babel/runtime/helpers/objectWithoutProperties';
import '../../input/props.js';
import '../../_chunks/dep-7bdccf65.js';
import '../../_chunks/dep-d518fdfb.js';
import '../../_chunks/dep-8d4d971b.js';
import '../../_chunks/dep-98d89c71.js';
import '../../config-provider/hooks/useConfig.js';
import '../../config-provider/utils/context.js';
import '../../_chunks/dep-509ddbe3.js';
import '../../_chunks/dep-f0f392fb.js';
import 'dayjs';
import '../../_chunks/dep-2ba9b7d0.js';
import '../../_chunks/dep-8be9c790.js';
import '../../input/hooks/useInput.js';
import '../../form/constants/index.js';
import '@babel/runtime/helpers/createClass';
import '@babel/runtime/helpers/classCallCheck';
import '../../_chunks/dep-e332908e.js';
import '../../input/hooks/useLengthLimit.js';
import '../../_chunks/dep-06660d60.js';
import '../../_chunks/dep-fbf70ecb.js';
import '../../input/hooks/useInputEventHandler.js';
import '../../input/hooks/useInputWidth.js';
import '../../_chunks/dep-d5fc4caa.js';
import '../../input/input-group.js';
import '../../input/input-group-props.js';
import '../../_chunks/dep-ae3e94b6.js';
import 'tdesign-vue-next/esm/common/style/web/components/input/_index.less';

var Search = defineComponent({
  name: "TTransferSearch",
  props: {
    value: {
      type: String,
      "default": ""
    },
    search: {
      type: [Boolean, Object],
      "default": false
    },
    placeholder: {
      type: String,
      "default": ""
    },
    onChange: Function
  },
  setup: function setup(props) {
    var classPrefix = usePrefixClass();
    var _useGlobalIcon = useGlobalIcon({
        SearchIcon: SearchIcon
      }),
      SearchIcon$1 = _useGlobalIcon.SearchIcon;
    var handleChange = function handleChange(value, changeCtx) {
      var _props$onChange;
      var e = changeCtx.e,
        trigger = changeCtx.trigger;
      (_props$onChange = props.onChange) === null || _props$onChange === void 0 || _props$onChange.call(props, {
        value: value,
        e: e,
        trigger: trigger
      });
    };
    var inputProps = _typeof(props.search) === "object" ? props.search : {
      clearable: true
    };
    var inputEvents = useEventForward(inputProps, {
      onChange: handleChange
    });
    return function () {
      return createVNode("div", {
        "class": "".concat(classPrefix.value, "-transfer__search-wrapper")
      }, [createVNode(Input, mergeProps({
        "defaultValue": props.value,
        "placeholder": props.placeholder
      }, inputEvents.value), {
        "default": function _default() {
          return [createVNode(SearchIcon$1, {
            "slot": "suffix-icon"
          }, null)];
        }
      })]);
    };
  }
});

export { Search as default };
