// tdesign-vue-next/cjs/table/components/tfoot.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var index = require('../../_chunks/dep-1862d466.js');
require('../../config-provider/hooks/useConfig.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-0989c3be.js');
var table_utils_index = require('../utils/index.js');
var table_hooks_useFixed = require('../hooks/useFixed.js');
var table_hooks_useRowspanAndColspan = require('../hooks/useRowspanAndColspan.js');
var table_hooks_useClassName = require('../hooks/useClassName.js');
var isFunction = require('../../_chunks/dep-2dcf9237.js');
var isString = require('../../_chunks/dep-631750cd.js');
var get = require('../../_chunks/dep-74a5bebe.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-dc6f0c3f.js');
require('../../_chunks/dep-2aa3d3cf.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../_chunks/dep-3d29e358.js');
require('../../_chunks/dep-6212811d.js');
require('../../_chunks/dep-44195fb2.js');
require('../../_chunks/dep-0af49915.js');
require('../../_chunks/dep-02ea4f3c.js');
require('../../_chunks/dep-a214821d.js');
require('../../_chunks/dep-7870f4cb.js');
require('../../_chunks/dep-1c1b8615.js');
require('../../_chunks/dep-5103206f.js');
require('../../_chunks/dep-9a54a293.js');
require('../../_chunks/dep-eb3114d9.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-690970a4.js');
require('../../_chunks/dep-42740075.js');
require('../../_chunks/dep-79476fcb.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-0238ed9a.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TFoot = Vue.defineComponent({
  name: "TFoot",
  props: {
    rowKey: String,
    isFixedHeader: Boolean,
    rowAndColFixedPosition: Map,
    footData: Array,
    columns: Array,
    rowAttributes: [Array, Object, Function],
    rowClassName: [Array, String, Object, Function],
    thWidthList: [Object],
    footerSummary: [String, Function],
    rowspanAndColspanInFooter: Function,
    virtualScroll: Boolean
  },
  setup: function setup(props, context) {
    var renderTNode = index.useTNodeJSX();
    var classnames = table_hooks_useClassName["default"]();
    var _toRefs = Vue.toRefs(props),
      footData = _toRefs.footData,
      columns = _toRefs.columns,
      rowKey = _toRefs.rowKey,
      rowspanAndColspanInFooter = _toRefs.rowspanAndColspanInFooter;
    var _useRowspanAndColspan = table_hooks_useRowspanAndColspan["default"](footData, columns, rowKey, rowspanAndColspanInFooter),
      skipSpansMap = _useRowspanAndColspan.skipSpansMap;
    var renderTFootCell = function renderTFootCell(p) {
      var col = p.col,
        row = p.row;
      if (isFunction.isFunction(col.foot)) {
        return col.foot(Vue.h, p);
      }
      if (isString.isString(col.foot) && context.slots[col.foot]) {
        return context.slots[col.foot](p) || col.foot;
      }
      return col.foot || get.get(row, col.colKey);
    };
    return function () {
      var _props$footData;
      if (!columns.value) return null;
      var theadClasses = [classnames.tableFooterClasses.footer, _defineProperty__default["default"]({}, classnames.tableFooterClasses.fixed, props.isFixedHeader)];
      var footerDomList = (_props$footData = props.footData) === null || _props$footData === void 0 ? void 0 : _props$footData.map(function (row, rowIndex) {
        var trAttributes = table_utils_index.formatRowAttributes(props.rowAttributes, {
          row: row,
          rowIndex: rowIndex,
          type: "foot"
        });
        var customClasses = table_utils_index.formatRowClassNames(props.rowClassName, {
          row: row,
          rowIndex: rowIndex,
          type: "foot"
        }, rowKey.value || "id");
        return Vue.createVNode("tr", Vue.mergeProps(trAttributes, {
          "key": rowIndex,
          "class": customClasses
        }), [columns.value.map(function (col, colIndex) {
          var _props$thWidthList;
          var cellSpans = {};
          var spanState = null;
          if (skipSpansMap.value.size) {
            var _spanState, _spanState2;
            var cellKey = table_hooks_useRowspanAndColspan.getCellKey(row, rowKey.value, col.colKey, colIndex, rowIndex);
            spanState = skipSpansMap.value.get(cellKey) || {};
            ((_spanState = spanState) === null || _spanState === void 0 ? void 0 : _spanState.rowspan) > 1 && (cellSpans.rowspan = spanState.rowspan);
            ((_spanState2 = spanState) === null || _spanState2 === void 0 ? void 0 : _spanState2.colspan) > 1 && (cellSpans.colspan = spanState.colspan);
            if (spanState.skipped) return null;
          }
          var tdStyles = table_hooks_useFixed.getColumnFixedStyles(col, colIndex, props.rowAndColFixedPosition, classnames.tableColFixedClasses);
          var style = _objectSpread({}, tdStyles.style);
          if ((_props$thWidthList = props.thWidthList) !== null && _props$thWidthList !== void 0 && _props$thWidthList[col.colKey]) {
            style.width = "".concat(props.thWidthList[col.colKey], "px");
          }
          return Vue.createVNode("td", Vue.mergeProps(_objectSpread({
            key: col.colKey
          }, cellSpans), {
            "class": tdStyles.classes,
            "style": style
          }), [renderTFootCell({
            row: row,
            rowIndex: rowIndex,
            col: col,
            colIndex: colIndex
          })]);
        })]);
      });
      var footerSummary = renderTNode("footerSummary");
      if (!footerSummary && (!props.footData || !props.footData.length)) return null;
      return Vue.createVNode("tfoot", {
        "class": theadClasses,
        "style": {
          visibility: props.virtualScroll ? "hidden" : "visible"
        }
      }, [footerSummary && Vue.createVNode("tr", {
        "class": classnames.tableFullRowClasses.base
      }, [Vue.createVNode("td", {
        "colspan": columns.value.length
      }, [Vue.createVNode("div", {
        "class": classnames.tableFullRowClasses.innerFullElement
      }, [footerSummary])])]), footerDomList]);
    };
  }
});

exports["default"] = TFoot;
