// tdesign-vue-next/es/color-picker/utils/index.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

export { C as Color, D as Draggable, d as cmyk2rgb, e as cmykInputToColor, b as genGradientPoint, h as genId, k as getColorFormatInputs, j as getColorFormatMap, a as getColorFormatOptions, g as getColorObject, f as getColorWithoutAlpha, c as gradientColors2string, i as initColorFormat, l as isGradientColor, p as parseGradientString, r as rgb2cmyk } from '../../_chunks/dep-1a80bfd8.mjs';
export { A as ALPHA_FORMAT_MAP, C as COLOR_FORMAT_INPUTS, c as COLOR_MODES, a as DEFAULT_COLOR, D as DEFAULT_LINEAR_GRADIENT, b as DEFAULT_SYSTEM_SWATCH_COLORS, F as FORMATS, G as GRADIENT_SLIDER_DEFAULT_WIDTH, d as SATURATION_PANEL_DEFAULT_HEIGHT, S as SATURATION_PANEL_DEFAULT_WIDTH, e as SLIDER_DEFAULT_WIDTH, T as TD_COLOR_USED_COLORS_MAX_SIZE } from '../../_chunks/dep-063c3304.mjs';
import '../../_chunks/dep-f558ebb8.mjs';
import '../../_chunks/dep-b0b422fe.mjs';
import '../../_chunks/dep-466a8f31.mjs';
import '../../_chunks/dep-ba214d75.mjs';
import '../../_chunks/dep-e1c62679.mjs';
import '../../_chunks/dep-40798bc5.mjs';
import 'lodash-es';
import '../../_chunks/dep-47c3cae9.mjs';
