// tdesign-vue-next/cjs/date-picker/components/base/Header.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var pagination_index = require('../../../pagination/index.js');
var select_index = require('../../../select/index.js');
require('@babel/runtime/helpers/typeof');
require('../../../_chunks/dep-35e23c2b.js');
var index = require('../../../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/defineProperty');
require('dayjs');
require('../../../_chunks/dep-1035a9b6.js');
require('tdesign-icons-vue-next');
require('../../../_chunks/dep-4d6191e3.js');
var datePicker_hooks_useSelectRange = require('../../hooks/useSelectRange.js');
var configProvider_hooks_useConfig = require('../../../config-provider/hooks/useConfig.js');
require('../../../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../../../_chunks/dep-1862d466.js');
require('../../../_chunks/dep-8d990e66.js');
require('../../../_chunks/dep-2dcf9237.js');
require('../../../_chunks/dep-22dc294c.js');
require('../../../_chunks/dep-bf76dead.js');
require('../../../_chunks/dep-631750cd.js');
require('../../../_chunks/dep-4cb26289.js');
require('../../../_chunks/dep-a6551cc4.js');
require('../../../_chunks/dep-7d9abefd.js');
require('../../../_chunks/dep-2fb8d44f.js');
require('../../../_chunks/dep-80a41429.js');
require('../../../_chunks/dep-91305690.js');
require('../../../_chunks/dep-0b92e695.js');
require('../../../_chunks/dep-b2024592.js');
require('../../../_chunks/dep-9f98bf9b.js');
require('../../../_chunks/dep-0afce170.js');
require('../../../_chunks/dep-c26a7d50.js');
require('../../../_chunks/dep-ce9c15f5.js');
require('../../../input-number/index.js');
require('../../../input-number/input-number.js');
require('../../../button/index.js');
require('../../../button/button.js');
require('../../../loading/index.js');
require('../../../_chunks/dep-34cd76d4.js');
require('../../../loading/plugin.js');
require('../../../_chunks/dep-d632b788.js');
require('../../../loading/icon/gradient.js');
require('../../../_chunks/dep-4ba41627.js');
require('../../../_chunks/dep-2aa3d3cf.js');
require('../../../_chunks/dep-3d29e358.js');
require('../../../_chunks/dep-6212811d.js');
require('../../../_chunks/dep-44195fb2.js');
require('../../../_chunks/dep-4a7162e7.js');
require('../../../_chunks/dep-be1af85d.js');
require('../../../_chunks/dep-db9c85b1.js');
require('../../../_chunks/dep-82fe2026.js');
require('../../../_chunks/dep-33b46a52.js');
require('../../../_chunks/dep-930a2de5.js');
require('../../../_chunks/dep-c3bbd06c.js');
require('../../../_chunks/dep-6d87f74d.js');
require('../../../_chunks/dep-281f7eb2.js');
require('../../../_chunks/dep-f8f7b1af.js');
require('../../../_chunks/dep-66d3f30f.js');
require('../../../_chunks/dep-e1d22111.js');
require('../../../_chunks/dep-754c0523.js');
require('../../../_chunks/dep-7076a08a.js');
require('../../../_chunks/dep-2fc884aa.js');
require('../../../_chunks/dep-75d6da05.js');
require('../../../_chunks/dep-274369eb.js');
require('../../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../../_chunks/dep-38acad3d.js');
require('../../../loading/props.js');
require('../../../_chunks/dep-43dcd2db.js');
require('../../../_chunks/dep-392dde71.js');
require('../../../_chunks/dep-02ea4f3c.js');
require('../../../_chunks/dep-5103206f.js');
require('../../../_chunks/dep-7707ffe2.js');
require('../../../_chunks/dep-74a5bebe.js');
require('../../../_chunks/dep-4eca9add.js');
require('../../../_chunks/dep-42740075.js');
require('../../../_chunks/dep-9ac5c6ca.js');
require('../../../_chunks/dep-e3edff33.js');
require('../../../button/props.js');
require('../../../_chunks/dep-362c5212.js');
require('../../../_chunks/dep-74b143e0.js');
require('../../../_chunks/dep-719a33c8.js');
require('../../../input/index.js');
require('../../../input/input.js');
require('../../../input/props.js');
require('../../../_chunks/dep-9dbade68.js');
require('../../../input/hooks/useInput.js');
require('../../../form/constants/index.js');
require('../../../input/hooks/useLengthLimit.js');
require('../../../_chunks/dep-692bdebe.js');
require('../../../input/hooks/useInputEventHandler.js');
require('../../../input/hooks/useInputWidth.js');
require('../../../_chunks/dep-e2122882.js');
require('../../../input/input-group.js');
require('../../../input/input-group-props.js');
require('../../../input-number/props.js');
require('../../../_chunks/dep-a13f0135.js');
require('../../../input-number/hooks/useInputNumber.js');
require('../../../_chunks/dep-6bac596c.js');
require('../../../_chunks/dep-73734701.js');
require('../../../input-adornment/index.js');
require('../../../input-adornment/input-adornment.js');
require('../../../input-adornment/props.js');
require('../../../pagination/props.js');
require('../../../pagination/hooks/useMoreAction.js');
require('../../../pagination/hooks/usePaginationClasses.js');
require('../../../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../../../common-components/fake-arrow.js');
require('../../../select-input/index.js');
require('../../../select-input/select-input.js');
require('../../../popup/index.js');
require('../../../popup/popup.js');
require('@popperjs/core');
require('../../../popup/container.js');
require('../../../popup/props.js');
require('../../../_chunks/dep-9bee89d3.js');
require('../../../_chunks/dep-027cd76f.js');
require('../../../_chunks/dep-ae4040c5.js');
require('../../../_chunks/dep-79476fcb.js');
require('../../../select-input/props.js');
require('../../../select-input/hooks/useMultiple.js');
require('../../../tag-input/index.js');
require('../../../tag-input/tag-input.js');
require('../../../tag-input/props.js');
require('../../../tag-input/hooks/useDragSorter.js');
require('../../../tag-input/hooks/useHover.js');
require('../../../tag-input/hooks/useTagScroll.js');
require('../../../tag-input/hooks/useTagList.js');
require('../../../tag/index.js');
require('../../../tag/tag.js');
require('tinycolor2');
require('../../../tag/props.js');
require('../../../tag/check-tag.js');
require('../../../tag/check-tag-props.js');
require('../../../_chunks/dep-a4be1915.js');
require('../../../tag/check-tag-group.js');
require('../../../tag/check-tag-group-props.js');
require('../../../select-input/hooks/useOverlayInnerStyle.js');
require('../../../select-input/hooks/useSingle.js');
require('../../../_chunks/dep-eb3114d9.js');
require('../../../_chunks/dep-690970a4.js');
require('../../../_chunks/dep-7870f4cb.js');
require('../../../select/components/select-panel.js');
require('../../../select/option.js');
require('../../../select/option-props.js');
require('../../../checkbox/index.js');
require('../../../checkbox/checkbox.js');
require('../../../checkbox/props.js');
require('../../../checkbox/constants/index.js');
require('../../../checkbox/hooks/useCheckboxLazyLoad.js');
require('../../../_chunks/dep-b9a88912.js');
require('../../../checkbox/hooks/useKeyboardEvent.js');
require('../../../checkbox/group.js');
require('../../../checkbox/checkbox-group-props.js');
require('../../../_chunks/dep-0cda9fe9.js');
require('../../../_chunks/dep-b0334c78.js');
require('../../../_chunks/dep-a214821d.js');
require('../../../select/utils/index.js');
require('../../../select/constants/index.js');
require('../../../select/option-group.js');
require('../../../select/option-group-props.js');
require('../../../select/props.js');
require('../../../select/hooks/usePanelVirtualScroll.js');
require('../../../_chunks/dep-5c79e9ef.js');
require('../../../_chunks/dep-9a54a293.js');
require('../../../select/hooks/useKeyboardControl.js');
require('../../../select/hooks/useSelectOptions.js');
require('../../../_chunks/dep-1c1b8615.js');
require('../../../pagination/pagination-mini.js');
require('../../../pagination/pagination-mini-props.js');
require('../../../tooltip/index.js');
require('../../../tooltip/tooltip.js');
require('../../../tooltip/props.js');
require('../../../tooltip/utils/index.js');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('../../../_chunks/dep-58022b88.js');
require('../../../_chunks/dep-484d5d62.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);

var TDateHeader = Vue.defineComponent({
  name: "TDatePickerHeader",
  props: {
    mode: {
      type: String,
      "default": "date"
    },
    range: [Array, Function],
    year: Number,
    month: Number,
    internalYear: Array,
    partial: String,
    onMonthChange: Function,
    onYearChange: Function,
    onJumperClick: Function
  },
  setup: function setup(props) {
    var _toRefs = Vue.toRefs(props),
      year = _toRefs.year,
      month = _toRefs.month;
    var _useConfig = configProvider_hooks_useConfig.useConfig("classPrefix"),
      classPrefix = _useConfig.classPrefix;
    var COMPONENT_NAME = index.usePrefixClass("date-picker__header");
    var _useConfig2 = configProvider_hooks_useConfig.useConfig("datePicker"),
      globalConfig = _useConfig2.globalConfig;
    var _useSelectRange = datePicker_hooks_useSelectRange.useSelectRange({
        range: props.range,
        mode: props.mode,
        year: year,
        month: month
      }),
      monthHasAnyAllowed = _useSelectRange.monthHasAnyAllowed,
      yearHasAnyAllowed = _useSelectRange.yearHasAnyAllowed,
      decadeHasAnyAllowed = _useSelectRange.decadeHasAnyAllowed,
      paginationDisabled = _useSelectRange.paginationDisabled,
      canLoadMoreTop = _useSelectRange.canLoadMoreTop,
      canLoadMoreBottom = _useSelectRange.canLoadMoreBottom;
    var yearOptions = Vue.ref(initOptions(props.year));
    var showMonthPicker = Vue.computed(function () {
      return props.mode === "date" || props.mode === "week";
    });
    var nearestYear = Vue.computed(function () {
      var _yearOptions$value$fi;
      var extraYear = props.partial === "end" && props.mode === "year" && Number(props.internalYear[1]) - Number(props.internalYear[0]) <= 9 ? 9 : 0;
      return ((_yearOptions$value$fi = yearOptions.value.find(function (option) {
        return option.value - (props.year + extraYear) <= 9 && option.value - (props.year + extraYear) >= 0;
      })) === null || _yearOptions$value$fi === void 0 ? void 0 : _yearOptions$value$fi.value) || props.year;
    });
    var monthOptions = Vue.computed(function () {
      return globalConfig.value.months.map(function (item, index) {
        return {
          label: item,
          value: index,
          disabled: !monthHasAnyAllowed(props.year, index)
        };
      });
    });
    var showPanelTop = Vue.computed(function () {
      var options = yearOptions.value;
      if (!options.length) return false;
      var first = options[0].value;
      return canLoadMoreTop(first);
    });
    var showPanelBottom = Vue.computed(function () {
      var options = yearOptions.value;
      if (!options.length) return false;
      var last = options[options.length - 1].value;
      return canLoadMoreBottom(last);
    });
    function initOptions(year2) {
      var options = [];
      if (props.mode === "year") {
        var extraYear = year2 % 10;
        var minYear = year2 - extraYear - 100;
        var maxYear = year2 - extraYear + 100;
        for (var i = minYear; i <= maxYear; i += 10) {
          var end = i + 9;
          if (decadeHasAnyAllowed(end)) {
            options.push({
              label: "".concat(i, " - ").concat(end),
              value: end,
              disabled: false
            });
          }
        }
      } else {
        if (yearHasAnyAllowed(year2)) options.push({
          label: "".concat(year2),
          value: year2,
          disabled: false
        });
        for (var _i = 1; _i <= 10; _i++) {
          if (yearHasAnyAllowed(year2 + _i)) options.push({
            label: "".concat(year2 + _i),
            value: year2 + _i,
            disabled: false
          });
          if (yearHasAnyAllowed(year2 - _i)) options.unshift({
            label: "".concat(year2 - _i),
            value: year2 - _i,
            disabled: false
          });
        }
      }
      return options;
    }
    function loadMoreYear(year2, type) {
      var options = [];
      if (props.mode === "year") {
        var extraYear = year2 % 10;
        if (type === "add") {
          for (var i = year2 - extraYear + 10; i <= year2 - extraYear + 50; i += 10) {
            var end = i + 9;
            if (decadeHasAnyAllowed(end)) options.push({
              label: "".concat(i, " - ").concat(end),
              value: end,
              disabled: false
            });
          }
        } else {
          for (var _i2 = year2 - extraYear - 1; _i2 > year2 - extraYear - 50; _i2 -= 10) {
            if (decadeHasAnyAllowed(_i2)) options.unshift({
              label: "".concat(_i2 - 9, " - ").concat(_i2),
              value: _i2,
              disabled: false
            });
          }
        }
      } else if (type === "add") {
        for (var _i3 = year2 + 1; _i3 <= year2 + 10; _i3++) {
          if (yearHasAnyAllowed(_i3)) options.push({
            label: "".concat(_i3),
            value: _i3,
            disabled: false
          });
        }
      } else {
        for (var _i4 = year2 - 1; _i4 > year2 - 10; _i4--) {
          if (yearHasAnyAllowed(_i4)) options.unshift({
            label: "".concat(_i4),
            value: _i4,
            disabled: false
          });
        }
      }
      return options;
    }
    var labelMap = Vue.computed(function () {
      return {
        year: {
          prev: globalConfig.value.preDecade,
          current: globalConfig.value.now,
          next: globalConfig.value.nextDecade
        },
        month: {
          prev: globalConfig.value.preYear,
          current: globalConfig.value.now,
          next: globalConfig.value.nextYear
        },
        date: {
          prev: globalConfig.value.preMonth,
          current: globalConfig.value.now,
          next: globalConfig.value.nextMonth
        },
        quarter: {
          prev: globalConfig.value.preYear,
          current: globalConfig.value.now,
          next: globalConfig.value.nextYear
        },
        week: {
          prev: globalConfig.value.preMonth,
          current: globalConfig.value.now,
          next: globalConfig.value.nextMonth
        }
      };
    });
    function handleScroll(_ref) {
      var e = _ref.e;
      if (e.target.scrollTop === 0) {
        if (showPanelTop.value) handlePanelTopClick(e);
      } else if (e.target.scrollTop === e.target.scrollHeight - e.target.clientHeight) {
        if (showPanelBottom.value) handlePanelBottomClick(e);
      }
    }
    function handlePanelTopClick(e) {
      var _e$stopPropagation;
      e === null || e === void 0 || (_e$stopPropagation = e.stopPropagation) === null || _e$stopPropagation === void 0 || _e$stopPropagation.call(e);
      var firstYear = yearOptions.value[0].value;
      var options = loadMoreYear(firstYear, "reduce");
      yearOptions.value = [].concat(_toConsumableArray__default["default"](options), _toConsumableArray__default["default"](yearOptions.value));
    }
    function handlePanelBottomClick(e) {
      var _e$stopPropagation2;
      e === null || e === void 0 || (_e$stopPropagation2 = e.stopPropagation) === null || _e$stopPropagation2 === void 0 || _e$stopPropagation2.call(e);
      var lastYear = yearOptions.value.slice(-1)[0].value;
      var options = loadMoreYear(lastYear, "add");
      yearOptions.value = [].concat(_toConsumableArray__default["default"](yearOptions.value), _toConsumableArray__default["default"](options));
    }
    Vue.watch(function () {
      return props.mode;
    }, function () {
      yearOptions.value = initOptions(props.year);
    });
    return function () {
      return Vue.createVNode("div", {
        "class": COMPONENT_NAME.value
      }, [Vue.createVNode("div", {
        "class": "".concat(COMPONENT_NAME.value, "-controller")
      }, [showMonthPicker.value && Vue.createVNode(select_index.Select, {
        "class": "".concat(COMPONENT_NAME.value, "-controller-month"),
        "value": props.month,
        "options": monthOptions.value,
        "onChange": function onChange(val) {
          var _props$onMonthChange;
          (_props$onMonthChange = props.onMonthChange) === null || _props$onMonthChange === void 0 || _props$onMonthChange.call(props, val);
        },
        "popupProps": {
          attach: function attach(triggerElement) {
            return triggerElement.parentNode;
          },
          overlayClassName: "".concat(COMPONENT_NAME.value, "-controller-month-popup")
        }
      }, null), Vue.createVNode(select_index.Select, {
        "class": "".concat(COMPONENT_NAME.value, "-controller-year"),
        "value": props.mode === "year" ? nearestYear.value : props.year,
        "options": yearOptions.value,
        "onChange": function onChange(val) {
          var _props$onYearChange;
          (_props$onYearChange = props.onYearChange) === null || _props$onYearChange === void 0 || _props$onYearChange.call(props, val);
        },
        "popupProps": {
          onScroll: handleScroll,
          attach: function attach(triggerElement) {
            return triggerElement.parentNode;
          },
          overlayClassName: "".concat(COMPONENT_NAME.value, "-controller-year-popup")
        },
        "panelTopContent": function panelTopContent() {
          return showPanelTop.value && Vue.createVNode("div", {
            "class": "".concat(classPrefix.value, "-select-option"),
            "onClick": handlePanelTopClick
          }, [Vue.createTextVNode("...")]);
        },
        "panelBottomContent": function panelBottomContent() {
          return showPanelBottom.value && Vue.createVNode("div", {
            "class": "".concat(classPrefix.value, "-select-option"),
            "onClick": handlePanelBottomClick
          }, [Vue.createTextVNode("...")]);
        }
      }, null)]), Vue.createVNode(pagination_index.PaginationMini, {
        "tips": labelMap.value[props.mode],
        "size": "small",
        "onChange": props.onJumperClick,
        "disabled": paginationDisabled.value
      }, null)]);
    };
  }
});

exports["default"] = TDateHeader;
