// tdesign-vue-next/cjs/date-picker/hooks/useDisableDate.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var dayjs = require('dayjs');
var utils = require('../../_chunks/dep-1035a9b6.js');
var isObject = require('../../_chunks/dep-bf76dead.js');
require('@babel/runtime/helpers/slicedToArray');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('@babel/runtime/helpers/defineProperty');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-58022b88.js');
require('../../_chunks/dep-9bee89d3.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-80a41429.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-484d5d62.js');
require('../../_chunks/dep-4d6191e3.js');
require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function useDisableDate(props) {
  var _disableDate = props.disableDate,
    format = props.format,
    mode = props.mode,
    start = props.start,
    end = props.end;
  return {
    disableDate: function disableDate(value) {
      return !utils.isEnabledDate({
        disableDate: _disableDate,
        format: format,
        mode: mode,
        value: value
      });
    },
    minDate: isObject.isObject(_disableDate) && "before" in _disableDate ? new Date(dayjs__default["default"](_disableDate.before).startOf("day").format()) : start,
    maxDate: isObject.isObject(_disableDate) && "after" in _disableDate ? new Date(dayjs__default["default"](_disableDate.after).endOf("day").format()) : end
  };
}

exports.useDisableDate = useDisableDate;
