// tdesign-vue-next/cjs/date-picker/hooks/useRange.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var index$2 = require('../../_chunks/dep-1862d466.js');
var index = require('../../_chunks/dep-0238ed9a.js');
var index$1 = require('../../_chunks/dep-c26a7d50.js');
require('@babel/runtime/helpers/slicedToArray');
require('../../_chunks/dep-0989c3be.js');
var index$3 = require('../../_chunks/dep-9dbade68.js');
var format = require('../../_chunks/dep-4d6191e3.js');
var datePicker_hooks_useRangeValue = require('./useRangeValue.js');
var omit = require('../../_chunks/dep-5c79e9ef.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-74b143e0.js');
require('../../_chunks/dep-484d5d62.js');
require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-ce9c15f5.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-9a54a293.js');
require('../../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var PARTIAL_MAP = {
  first: "start",
  second: "end"
};
function useRange(props) {
  var COMPONENT_NAME = index.usePrefixClass("date-range-picker");
  var _useConfig = configProvider_hooks_useConfig.useConfig("datePicker"),
    globalConfig = _useConfig.globalConfig;
  var _useGlobalIcon = index$1.useGlobalIcon({
      CalendarIcon: tdesignIconsVueNext.CalendarIcon
    }),
    CalendarIcon = _useGlobalIcon.CalendarIcon;
  var renderTNodeJSX = index$2.useTNodeJSX();
  var isMountedRef = Vue.ref(false);
  var inputRef = Vue.ref();
  var _useRangeValue = datePicker_hooks_useRangeValue.useRangeValue(props),
    value = _useRangeValue.value,
    onChange = _useRangeValue.onChange,
    time = _useRangeValue.time,
    month = _useRangeValue.month,
    year = _useRangeValue.year,
    cacheValue = _useRangeValue.cacheValue,
    isFirstValueSelected = _useRangeValue.isFirstValueSelected;
  var formatRef = Vue.computed(function () {
    return format.getDefaultFormat({
      mode: props.mode,
      format: props.format,
      valueType: props.valueType,
      enableTimePicker: props.enableTimePicker
    });
  });
  var popupVisible = Vue.ref(false);
  var isHoverCell = Vue.ref(false);
  var activeIndex = Vue.ref(0);
  var inputValue = Vue.ref(format.formatDate(props.value, {
    format: formatRef.value.format
  }));
  var isReadOnly = index$3.useReadonly();
  var rangeInputProps = Vue.computed(function () {
    return _objectSpread(_objectSpread({}, props.rangeInputProps), {}, {
      size: props.size,
      ref: inputRef,
      borderless: props.borderless,
      clearable: props.clearable && !isReadOnly.value,
      prefixIcon: function prefixIcon() {
        return renderTNodeJSX("prefixIcon");
      },
      readonly: isReadOnly.value || !props.allowInput,
      separator: props.separator || globalConfig.value.rangeSeparator,
      placeholder: props.placeholder || globalConfig.value.placeholder[props.mode],
      suffixIcon: function suffixIcon() {
        return renderTNodeJSX("suffixIcon") || Vue.createVNode(CalendarIcon, null, null);
      },
      "class": _defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "__input--placeholder"), isHoverCell.value),
      onClick: function onClick(_ref) {
        var position = _ref.position;
        activeIndex.value = position === "first" ? 0 : 1;
      },
      onClear: function onClear(context) {
        if (context instanceof MouseEvent) context.stopPropagation();else context.e.stopPropagation();
        popupVisible.value = false;
        onChange === null || onChange === void 0 || onChange([], {
          dayjsValue: [],
          trigger: "clear"
        });
      },
      onBlur: function onBlur(newVal, _ref2) {
        var _props$onBlur;
        var e = _ref2.e,
          position = _ref2.position;
        (_props$onBlur = props.onBlur) === null || _props$onBlur === void 0 || _props$onBlur.call(props, {
          value: newVal,
          partial: PARTIAL_MAP[position],
          e: e
        });
      },
      onFocus: function onFocus(newVal, _ref3) {
        var _props$onFocus;
        var e = _ref3.e,
          position = _ref3.position;
        (_props$onFocus = props.onFocus) === null || _props$onFocus === void 0 || _props$onFocus.call(props, {
          value: newVal,
          partial: PARTIAL_MAP[position],
          e: e
        });
        activeIndex.value = position === "first" ? 0 : 1;
      },
      onChange: function onChange(newVal, _ref4) {
        _ref4.e;
          _ref4.position;
        inputValue.value = newVal;
        if (!format.isValidDate(newVal, formatRef.value.format)) return;
        cacheValue.value = newVal;
        var newYear = [];
        var newMonth = [];
        var newTime = [];
        newVal.forEach(function (v) {
          newYear.push(format.parseToDayjs(v, formatRef.value.format).year());
          newMonth.push(format.parseToDayjs(v, formatRef.value.format).month());
          newTime.push(format.parseToDayjs(v, formatRef.value.format).format(formatRef.value.timeFormat));
        });
        year.value = newYear;
        month.value = newMonth;
        time.value = newTime;
      },
      onEnter: function onEnter(newVal) {
        if (!format.isValidDate(newVal, formatRef.value.format) && !format.isValidDate(value.value, formatRef.value.format)) return;
        popupVisible.value = false;
        if (format.isValidDate(newVal, formatRef.value.format)) {
          onChange === null || onChange === void 0 || onChange(format.formatDate(newVal, {
            format: formatRef.value.format,
            targetFormat: formatRef.value.valueType,
            autoSwap: true
          }), {
            dayjsValue: newVal.map(function (v) {
              return format.parseToDayjs(v, formatRef.value.format);
            }),
            trigger: "enter"
          });
        } else if (format.isValidDate(value.value, formatRef.value.format)) {
          inputValue.value = format.formatDate(value.value, {
            format: formatRef.value.format
          });
        } else {
          inputValue.value = [];
        }
      }
    });
  });
  var popupProps = Vue.computed(function () {
    var _props$popupProps$ove, _props$popupProps, _props$popupProps2;
    return _objectSpread(_objectSpread({
      expandAnimation: true
    }, omit.omit(props.popupProps, "on-visible-change")), {}, {
      overlayInnerStyle: (_props$popupProps$ove = (_props$popupProps = props.popupProps) === null || _props$popupProps === void 0 ? void 0 : _props$popupProps.overlayInnerStyle) !== null && _props$popupProps$ove !== void 0 ? _props$popupProps$ove : {
        width: "auto"
      },
      overlayClassName: [(_props$popupProps2 = props.popupProps) === null || _props$popupProps2 === void 0 ? void 0 : _props$popupProps2.overlayClassName, "".concat(COMPONENT_NAME.value, "__panel-container")],
      onVisibleChange: function onVisibleChange(visible, context) {
        var _props$popupProps3, _props$popupProps3$on, _props$popupProps4, _props$popupProps4$on;
        if (isReadOnly.value) return;
        (_props$popupProps3 = props.popupProps) === null || _props$popupProps3 === void 0 || (_props$popupProps3$on = _props$popupProps3.onVisibleChange) === null || _props$popupProps3$on === void 0 || _props$popupProps3$on.call(_props$popupProps3, visible, context);
        (_props$popupProps4 = props.popupProps) === null || _props$popupProps4 === void 0 || (_props$popupProps4$on = _props$popupProps4["on-visible-change"]) === null || _props$popupProps4$on === void 0 || _props$popupProps4$on.call(_props$popupProps4, visible, context);
        if (context.trigger === "trigger-element-click") {
          var indexMap = {
            0: "first",
            1: "second"
          };
          inputRef.value.focus({
            position: indexMap[activeIndex.value]
          });
          popupVisible.value = true;
          return;
        }
        popupVisible.value = visible;
      }
    });
  });
  Vue.watch(value, function (value2) {
    if (!value2) {
      inputValue.value = [];
      return;
    }
    if (!format.isValidDate(value2, formatRef.value.format)) return;
    inputValue.value = format.formatDate(value2, {
      format: formatRef.value.valueType,
      targetFormat: formatRef.value.format
    });
  }, {
    immediate: true
  });
  Vue.watch(activeIndex, function (index) {
    var _inputRef$value, _inputRef$value$focus;
    if (!isMountedRef.value) {
      isMountedRef.value = true;
      return;
    }
    if (!popupVisible.value) return;
    var indexMap = {
      0: "first",
      1: "second"
    };
    (_inputRef$value = inputRef.value) === null || _inputRef$value === void 0 || (_inputRef$value$focus = _inputRef$value.focus) === null || _inputRef$value$focus === void 0 || _inputRef$value$focus.call(_inputRef$value, {
      position: indexMap[index]
    });
  }, {
    immediate: true
  });
  return {
    year: year,
    month: month,
    value: value,
    time: time,
    inputValue: inputValue,
    popupVisible: popupVisible,
    rangeInputProps: rangeInputProps,
    popupProps: popupProps,
    isHoverCell: isHoverCell,
    activeIndex: activeIndex,
    isFirstValueSelected: isFirstValueSelected,
    cacheValue: cacheValue,
    onChange: onChange
  };
}

exports.PARTIAL_MAP = PARTIAL_MAP;
exports.useRange = useRange;
