// tdesign-vue-next/cjs/date-picker/hooks/useSelectRange.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var utils = require('../../_chunks/dep-1035a9b6.js');
var isArray = require('../../_chunks/dep-4cb26289.js');
var isFunction = require('../../_chunks/dep-2dcf9237.js');
require('@babel/runtime/helpers/slicedToArray');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('@babel/runtime/helpers/defineProperty');
require('dayjs');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-58022b88.js');
require('../../_chunks/dep-9bee89d3.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-bf76dead.js');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-484d5d62.js');
require('../../_chunks/dep-4d6191e3.js');
require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');

function useSelectRange(props) {
  var rangeBounds = Vue.computed(function () {
    return utils.getRangeBounds(props.range);
  });
  var monthAllowed = function monthAllowed(year, month) {
    return utils.monthHasAnyAllowed(props.range, year, month);
  };
  var yearAllowed = function yearAllowed(year) {
    return utils.yearHasAnyAllowed(props.range, year);
  };
  var decadeAllowed = function decadeAllowed(decadeEndYear) {
    return utils.decadeHasAnyAllowed(props.range, decadeEndYear);
  };
  var paginationDisabled = Vue.computed(function () {
    var _props$month;
    return utils.computePaginationDisabled(props.range, props.mode, props.year.value, (_props$month = props.month) === null || _props$month === void 0 ? void 0 : _props$month.value);
  });
  var canLoadMoreTop = function canLoadMoreTop(firstValue) {
    var min = rangeBounds.value.min;
    if (isArray.isArray(props.range)) {
      if (!min) return true;
      var minYear = min.getFullYear();
      if (props.mode === "year") {
        var minDecadeEnd = Math.floor(minYear / 10) * 10 + 9;
        return firstValue > minDecadeEnd;
      }
      return firstValue > minYear;
    }
    if (isFunction.isFunction(props.range)) {
      if (props.mode === "year") {
        for (var i = firstValue - 10; i >= firstValue - 50; i -= 10) {
          if (decadeAllowed(i)) return true;
        }
        return false;
      }
      for (var _i = firstValue - 1; _i > firstValue - 10; _i--) {
        if (yearAllowed(_i)) return true;
      }
      return false;
    }
    return true;
  };
  var canLoadMoreBottom = function canLoadMoreBottom(lastValue) {
    var max = rangeBounds.value.max;
    if (isArray.isArray(props.range)) {
      if (!max) return true;
      var maxYear = max.getFullYear();
      if (props.mode === "year") {
        var maxDecadeEnd = Math.floor(maxYear / 10) * 10 + 9;
        return lastValue < maxDecadeEnd;
      }
      return lastValue < maxYear;
    }
    if (isFunction.isFunction(props.range)) {
      if (props.mode === "year") {
        for (var i = lastValue + 10; i <= lastValue + 50; i += 10) {
          if (decadeAllowed(i)) return true;
        }
        return false;
      }
      for (var _i2 = lastValue + 1; _i2 <= lastValue + 10; _i2++) {
        if (yearAllowed(_i2)) return true;
      }
      return false;
    }
    return true;
  };
  return {
    rangeBounds: rangeBounds,
    monthHasAnyAllowed: monthAllowed,
    yearHasAnyAllowed: yearAllowed,
    decadeHasAnyAllowed: decadeAllowed,
    paginationDisabled: paginationDisabled,
    canLoadMoreTop: canLoadMoreTop,
    canLoadMoreBottom: canLoadMoreBottom
  };
}

exports.useSelectRange = useSelectRange;
