// tdesign-vue-next/cjs/date-picker/hooks/useSingleValue.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var Vue = require('vue');
var format = require('../../_chunks/dep-4d6191e3.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-0989c3be.js');
var index = require('../../_chunks/dep-ce9c15f5.js');
require('@babel/runtime/helpers/defineProperty');
require('dayjs');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-484d5d62.js');
require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function useSingleValue(props) {
  var _value$value, _value$value2, _value$value3, _value$value4;
  var _toRefs = Vue.toRefs(props),
    valueFromProps = _toRefs.value,
    modelValue = _toRefs.modelValue;
  var _useVModel = index.useVModel(valueFromProps, modelValue, props.defaultValue, props.onChange),
    _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
    value = _useVModel2[0],
    onChange = _useVModel2[1];
  var formatRef = Vue.computed(function () {
    return format.getDefaultFormat({
      mode: props.mode,
      format: props.format,
      enableTimePicker: props.multiple ? false : props.enableTimePicker
    });
  });
  if (props.enableTimePicker) {
    if (!format.extractTimeFormat(formatRef.value.format)) console.error("format: ".concat(formatRef.value.format, " \u4E0D\u89C4\u8303\uFF0C\u5305\u542B\u65F6\u95F4\u9009\u62E9\u5FC5\u987B\u8981\u6709\u65F6\u95F4\u683C\u5F0F\u5316 HH:mm:ss"));
  }
  var time = Vue.ref(format.formatTime(props.multiple ? (_value$value = value.value) === null || _value$value === void 0 ? void 0 : _value$value[0] : value.value, formatRef.value.format, formatRef.value.timeFormat, props.defaultTime));
  var month = Vue.ref(format.parseToDayjs(props.multiple ? (_value$value2 = value.value) === null || _value$value2 === void 0 ? void 0 : _value$value2[0] : value.value, formatRef.value.format).month());
  var year = Vue.ref(format.parseToDayjs(props.multiple ? (_value$value3 = value.value) === null || _value$value3 === void 0 ? void 0 : _value$value3[0] : value.value, formatRef.value.format).year());
  var cacheValue = Vue.ref(format.formatDate(props.multiple ? (_value$value4 = value.value) === null || _value$value4 === void 0 ? void 0 : _value$value4[0] : value.value, {
    format: formatRef.value.format
  }));
  Vue.watchEffect(function () {
    if (!value.value) {
      cacheValue.value = "";
      return;
    }
    if (!format.isValidDate(value.value, formatRef.value.format)) return;
    cacheValue.value = format.formatDate(value.value, {
      format: formatRef.value.format
    });
    time.value = format.formatTime(value.value, formatRef.value.format, formatRef.value.timeFormat, props.defaultTime);
  });
  return {
    year: year,
    month: month,
    value: value,
    time: time,
    cacheValue: cacheValue,
    onChange: onChange
  };
}

exports.useSingleValue = useSingleValue;
