// tdesign-vue-next/cjs/transfer/transfer.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _toConsumableArray = require('@babel/runtime/helpers/toConsumableArray');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var _slicedToArray = require('@babel/runtime/helpers/slicedToArray');
var transfer_components_transferList = require('./components/transfer-list.js');
var transfer_components_transferOperations = require('./components/transfer-operations.js');
var transfer_utils_index = require('./utils/index.js');
var transfer_props = require('./props.js');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index$1 = require('../_chunks/dep-0238ed9a.js');
var index$3 = require('../_chunks/dep-0afce170.js');
var index = require('../_chunks/dep-362c5212.js');
require('../_chunks/dep-0989c3be.js');
var index$2 = require('../_chunks/dep-ce9c15f5.js');
var isFunction = require('../_chunks/dep-2dcf9237.js');
var pick = require('../_chunks/dep-eb3114d9.js');
var isEqual = require('../_chunks/dep-43dcd2db.js');
require('../pagination/index.js');
require('../pagination/pagination.js');
require('@babel/runtime/helpers/objectWithoutProperties');
require('tdesign-icons-vue-next');
require('../_chunks/dep-1862d466.js');
require('../_chunks/dep-8d990e66.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-c26a7d50.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('../_chunks/dep-4a7162e7.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../input-number/index.js');
require('../input-number/input-number.js');
require('../button/index.js');
require('../button/button.js');
require('../loading/index.js');
require('../_chunks/dep-34cd76d4.js');
require('../loading/plugin.js');
require('../_chunks/dep-d632b788.js');
require('../loading/icon/gradient.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-2aa3d3cf.js');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');
require('../_chunks/dep-38acad3d.js');
require('../loading/props.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../_chunks/dep-7707ffe2.js');
require('../_chunks/dep-392dde71.js');
require('../_chunks/dep-02ea4f3c.js');
require('../_chunks/dep-5103206f.js');
require('../_chunks/dep-74a5bebe.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-42740075.js');
require('../_chunks/dep-9ac5c6ca.js');
require('../_chunks/dep-e3edff33.js');
require('../button/props.js');
require('../_chunks/dep-719a33c8.js');
require('../_chunks/dep-74b143e0.js');
require('../input/index.js');
require('../input/input.js');
require('../input/props.js');
require('../_chunks/dep-9dbade68.js');
require('../input/hooks/useInput.js');
require('../form/constants/index.js');
require('../input/hooks/useLengthLimit.js');
require('../_chunks/dep-692bdebe.js');
require('../input/hooks/useInputEventHandler.js');
require('../input/hooks/useInputWidth.js');
require('../_chunks/dep-e2122882.js');
require('../input/input-group.js');
require('../input/input-group-props.js');
require('../input-number/props.js');
require('../_chunks/dep-a13f0135.js');
require('../input-number/hooks/useInputNumber.js');
require('../_chunks/dep-6bac596c.js');
require('../_chunks/dep-73734701.js');
require('../select/index.js');
require('../select/select.js');
require('@babel/runtime/helpers/asyncToGenerator');
require('@babel/runtime/regenerator');
require('../common-components/fake-arrow.js');
require('../select-input/index.js');
require('../select-input/select-input.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../popup/container.js');
require('../popup/props.js');
require('../_chunks/dep-9bee89d3.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-79476fcb.js');
require('../select-input/props.js');
require('../select-input/hooks/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/tag-input.js');
require('../tag-input/props.js');
require('../tag-input/hooks/useDragSorter.js');
require('../tag-input/hooks/useHover.js');
require('../tag-input/hooks/useTagScroll.js');
require('../tag-input/hooks/useTagList.js');
require('../tag/index.js');
require('../tag/tag.js');
require('tinycolor2');
require('../tag/props.js');
require('../tag/check-tag.js');
require('../tag/check-tag-props.js');
require('../_chunks/dep-a4be1915.js');
require('../tag/check-tag-group.js');
require('../tag/check-tag-group-props.js');
require('../select-input/hooks/useOverlayInnerStyle.js');
require('../select-input/hooks/useSingle.js');
require('../_chunks/dep-690970a4.js');
require('../_chunks/dep-7870f4cb.js');
require('../select/components/select-panel.js');
require('../select/option.js');
require('../select/option-props.js');
require('../checkbox/index.js');
require('../checkbox/checkbox.js');
require('../checkbox/props.js');
require('../checkbox/constants/index.js');
require('../checkbox/hooks/useCheckboxLazyLoad.js');
require('../_chunks/dep-b9a88912.js');
require('../checkbox/hooks/useKeyboardEvent.js');
require('../checkbox/group.js');
require('../checkbox/checkbox-group-props.js');
require('../_chunks/dep-0cda9fe9.js');
require('../_chunks/dep-b0334c78.js');
require('../_chunks/dep-a214821d.js');
require('../select/utils/index.js');
require('../select/constants/index.js');
require('../select/option-group.js');
require('../select/option-group-props.js');
require('../select/props.js');
require('../select/hooks/usePanelVirtualScroll.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-9a54a293.js');
require('../select/hooks/useKeyboardControl.js');
require('../select/hooks/useSelectOptions.js');
require('../_chunks/dep-1c1b8615.js');
require('../input-adornment/index.js');
require('../input-adornment/input-adornment.js');
require('../input-adornment/props.js');
require('../pagination/props.js');
require('../pagination/hooks/useMoreAction.js');
require('../pagination/hooks/usePaginationClasses.js');
require('../pagination/pagination-mini.js');
require('../pagination/pagination-mini-props.js');
require('../tooltip/index.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../tooltip/utils/index.js');
require('./components/transfer-search.js');
require('./hooks/useDragSort.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _toConsumableArray__default = /*#__PURE__*/_interopDefaultLegacy(_toConsumableArray);
var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);
var _slicedToArray__default = /*#__PURE__*/_interopDefaultLegacy(_slicedToArray);

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var _Transfer = Vue.defineComponent({
  name: transfer_utils_index.TRANSFER_NAME,
  props: transfer_props["default"],
  setup: function setup(props2, _ref) {
    var slots = _ref.slots;
    var disabled = index.useDisabled();
    var classPrefix = index$1.usePrefixClass();
    var _toRefs = Vue.toRefs(props2),
      value = _toRefs.value,
      modelValue = _toRefs.modelValue,
      checked = _toRefs.checked;
    var _useVModel = index$2.useVModel(value, modelValue, props2.defaultValue, props2.onChange),
      _useVModel2 = _slicedToArray__default["default"](_useVModel, 2),
      innerValue = _useVModel2[0],
      setInnerValue = _useVModel2[1];
    var _useDefaultValue = index$3.useDefaultValue(checked, props2.defaultChecked, props2.onCheckedChange, "checked"),
      _useDefaultValue2 = _slicedToArray__default["default"](_useDefaultValue, 1),
      innerChecked = _useDefaultValue2[0];
    var valueList = Vue.computed(function () {
      return innerValue.value;
    });
    var isTreeMode = Vue.computed(function () {
      var treeSlot = slots.tree;
      return isFunction.isFunction(treeSlot);
    });
    var leftButtonDisabled = Vue.computed(function () {
      return props2.direction === "right";
    });
    var rightButtonDisabled = Vue.computed(function () {
      return props2.direction === "left";
    });
    var transferData = Vue.computed(function () {
      return transfer_utils_index.getTransferData(props2.data, props2.keys, isTreeMode.value);
    });
    var sourceList = Vue.computed(function () {
      return transfer_utils_index.filterTransferData(transferData.value, valueList.value, false, isTreeMode.value);
    });
    var targetList = Vue.computed(function () {
      return transfer_utils_index.filterTransferData(transferData.value, valueList.value, true, isTreeMode.value);
    });
    var checkedValue = Vue.computed(function () {
      return _defineProperty__default["default"](_defineProperty__default["default"]({}, transfer_utils_index.SOURCE, transfer_utils_index.getDataValues(sourceList.value, innerChecked.value, {
        isTreeMode: isTreeMode.value
      })), transfer_utils_index.TARGET, transfer_utils_index.getDataValues(targetList.value, innerChecked.value, {
        isTreeMode: isTreeMode.value
      }));
    });
    var hasFooter = Vue.computed(function () {
      return !!slots.footer || !!props2.footer;
    });
    var showPagination = Vue.computed(function () {
      return !!props2.pagination && !slots.content;
    });
    var showSearch = Vue.computed(function () {
      return !!props2.search;
    });
    var footerOption = Vue.computed(function () {
      var footer = props2.footer || "";
      return transfer_utils_index.getTransferListOption(footer);
    });
    var emptyOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.empty);
    });
    var searchOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.search);
    });
    var checkAllOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.showCheckAll);
    });
    var disabledOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(disabled.value);
    });
    var titleOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.title);
    });
    var paginationOption = Vue.computed(function () {
      return transfer_utils_index.getTransferListOption(props2.pagination);
    });
    var handleCheckedChange = function handleCheckedChange(val, listType) {
      var _props2$onCheckedChan;
      var sourceChecked = listType === transfer_utils_index.SOURCE ? val : checkedValue.value[transfer_utils_index.SOURCE];
      var targetChecked = listType === transfer_utils_index.TARGET ? val : checkedValue.value[transfer_utils_index.TARGET];
      var checked2 = [].concat(_toConsumableArray__default["default"](sourceChecked), _toConsumableArray__default["default"](targetChecked));
      var event = {
        checked: checked2,
        sourceChecked: sourceChecked,
        targetChecked: targetChecked,
        type: listType
      };
      innerChecked.value = checked2;
      (_props2$onCheckedChan = props2.onCheckedChange) === null || _props2$onCheckedChan === void 0 || _props2$onCheckedChan.call(props2, event);
    };
    var transferTo = function transferTo(toDirection) {
      var oldTargetValue = JSON.parse(JSON.stringify(valueList.value));
      var newTargetValue;
      var selfCheckedValue = toDirection === transfer_utils_index.TARGET ? checkedValue.value[transfer_utils_index.SOURCE] : checkedValue.value[transfer_utils_index.TARGET];
      if (toDirection === transfer_utils_index.SOURCE) {
        newTargetValue = oldTargetValue.filter(function (v) {
          return !selfCheckedValue.some(function (cv) {
            return isEqual.isEqual(cv, v);
          });
        });
      } else if (props2.targetSort === "original") {
        var remainValue = transferData.value.reduce(function (acc, data) {
          if (oldTargetValue.some(function (v) {
            return isEqual.isEqual(v, data.value);
          }) && data.disabled) {
            return acc.concat(data.value);
          }
          return acc;
        }, []);
        newTargetValue = transfer_utils_index.getDataValues(transferData.value, oldTargetValue.concat(selfCheckedValue), {
          isTreeMode: isTreeMode.value,
          remainValue: remainValue
        });
      } else if (props2.targetSort === "unshift") {
        newTargetValue = selfCheckedValue.concat(oldTargetValue);
      } else {
        newTargetValue = oldTargetValue.concat(selfCheckedValue);
      }
      handleCheckedChange([], toDirection === transfer_utils_index.SOURCE ? transfer_utils_index.TARGET : transfer_utils_index.SOURCE);
      var params = {
        type: toDirection,
        movedValue: selfCheckedValue
      };
      setInnerValue(newTargetValue, params);
    };
    var transferToRight = function transferToRight() {
      transferTo(transfer_utils_index.TARGET);
    };
    var transferToLeft = function transferToLeft() {
      transferTo(transfer_utils_index.SOURCE);
    };
    var handleScroll = function handleScroll(e, listType) {
      var _props2$onScroll;
      var target = e.target;
      var bottomDistance = target.scrollHeight - target.scrollTop - target.clientHeight;
      var event = {
        e: e,
        bottomDistance: bottomDistance,
        type: listType
      };
      (_props2$onScroll = props2.onScroll) === null || _props2$onScroll === void 0 || _props2$onScroll.call(props2, event);
    };
    var handleSearch = function handleSearch(e) {
      var _props2$onSearch;
      (_props2$onSearch = props2.onSearch) === null || _props2$onSearch === void 0 || _props2$onSearch.call(props2, e);
    };
    var handlePageChange = function handlePageChange(pageInfo, listType) {
      var _props2$onPageChange;
      (_props2$onPageChange = props2.onPageChange) === null || _props2$onPageChange === void 0 || _props2$onPageChange.call(props2, pageInfo, {
        type: listType
      });
    };
    var handleDataChange = function handleDataChange(data, movedValue) {
      setInnerValue(data, {
        type: transfer_utils_index.TARGET,
        movedValue: movedValue
      });
    };
    var renderTransferList = function renderTransferList(listType) {
      var scopedSlots = pick.pick(slots, ["title", "empty", "footer", "operation", "transferItem", "default", "tree"]);
      return Vue.createVNode(transfer_components_transferList["default"], {
        "checkboxProps": props2.checkboxProps,
        "transferItem": props2.transferItem,
        "list-type": listType,
        "title": titleOption.value[listType],
        "data-source": listType === transfer_utils_index.SOURCE ? sourceList.value : targetList.value,
        "checked-value": checkedValue.value[listType],
        "disabled": disabledOption.value[listType],
        "search": searchOption.value[listType],
        "pagination": paginationOption.value[listType],
        "check-all": checkAllOption.value[listType],
        "footer": footerOption.value[listType],
        "empty": emptyOption.value[listType],
        "onCheckedChange": function onCheckedChange($event) {
          return handleCheckedChange($event, listType);
        },
        "onScroll": function onScroll($event) {
          return handleScroll($event, listType);
        },
        "onSearch": handleSearch,
        "onPageChange": function onPageChange($event) {
          return handlePageChange($event, listType);
        },
        "isTreeMode": isTreeMode.value,
        "onDataChange": handleDataChange,
        "currentValue": valueList.value,
        "draggable": props2.targetDraggable && listType === transfer_utils_index.TARGET
      }, _isSlot(scopedSlots) ? scopedSlots : {
        "default": function _default() {
          return [scopedSlots];
        }
      });
    };
    return function () {
      return Vue.createVNode("div", {
        "class": ["".concat(classPrefix.value, "-transfer"), showSearch.value ? "".concat(classPrefix.value, "-transfer__search") : "", hasFooter.value ? "".concat(classPrefix.value, "-transfer__footer") : "", showPagination.value ? "".concat(classPrefix.value, "-transfer__pagination") : "", isTreeMode.value ? "".concat(classPrefix.value, "-transfer--with-tree") : ""]
      }, [renderTransferList(transfer_utils_index.SOURCE), Vue.createVNode(transfer_components_transferOperations["default"], {
        "leftDisabled": disabledOption.value[transfer_utils_index.TARGET] || leftButtonDisabled.value || checkedValue.value[transfer_utils_index.TARGET].length === 0,
        "rightDisabled": disabledOption.value[transfer_utils_index.SOURCE] || rightButtonDisabled.value || checkedValue.value[transfer_utils_index.SOURCE].length === 0,
        "operation": props2.operation,
        "onMoveToRight": transferToRight,
        "onMoveToLeft": transferToLeft
      }, {
        operation: slots.operation
      }), renderTransferList(transfer_utils_index.TARGET)]);
    };
  }
});

exports["default"] = _Transfer;
