// tdesign-vue-next/cjs/image-viewer/base/ImageViewerUtils.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var tdesignIconsVueNext = require('tdesign-icons-vue-next');
var imageViewer_base_ImageModalIcon = require('./ImageModalIcon.js');
var tooltip_index = require('../../tooltip/index.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../../_chunks/dep-35e23c2b.js');
var index = require('../../_chunks/dep-0238ed9a.js');
var index$1 = require('../../_chunks/dep-03ea53c4.js');
require('../../_chunks/dep-0989c3be.js');
require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/slicedToArray');
var largeNumber = require('../../_chunks/dep-73734701.js');
var configProvider_hooks_useConfig = require('../../config-provider/hooks/useConfig.js');
require('../../_chunks/dep-1862d466.js');
require('../../_chunks/dep-8d990e66.js');
require('../../_chunks/dep-2dcf9237.js');
require('../../_chunks/dep-22dc294c.js');
require('../../_chunks/dep-bf76dead.js');
require('../../_chunks/dep-631750cd.js');
require('../../_chunks/dep-4cb26289.js');
require('../../_chunks/dep-a6551cc4.js');
require('../../_chunks/dep-7d9abefd.js');
require('../../_chunks/dep-2fb8d44f.js');
require('../../_chunks/dep-80a41429.js');
require('../../_chunks/dep-91305690.js');
require('../../_chunks/dep-0b92e695.js');
require('../../_chunks/dep-b2024592.js');
require('../../_chunks/dep-9f98bf9b.js');
require('../../_chunks/dep-4a7162e7.js');
require('../../_chunks/dep-be1af85d.js');
require('../../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../../_chunks/dep-82fe2026.js');
require('../../_chunks/dep-33b46a52.js');
require('../../_chunks/dep-930a2de5.js');
require('../../_chunks/dep-c3bbd06c.js');
require('../../_chunks/dep-6d87f74d.js');
require('../../_chunks/dep-281f7eb2.js');
require('../../_chunks/dep-f8f7b1af.js');
require('../../_chunks/dep-66d3f30f.js');
require('../../_chunks/dep-e1d22111.js');
require('../../_chunks/dep-754c0523.js');
require('../../_chunks/dep-7076a08a.js');
require('../../_chunks/dep-2fc884aa.js');
require('../../_chunks/dep-75d6da05.js');
require('../../_chunks/dep-274369eb.js');
require('../../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../../tooltip/tooltip.js');
require('../../tooltip/props.js');
require('../../popup/props.js');
require('../../popup/index.js');
require('../../popup/popup.js');
require('@popperjs/core');
require('../../_chunks/dep-38acad3d.js');
require('../../_chunks/dep-4ba41627.js');
require('../../_chunks/dep-ce9c15f5.js');
require('../../popup/container.js');
require('../../_chunks/dep-e2122882.js');
require('../../_chunks/dep-9bee89d3.js');
require('../../_chunks/dep-027cd76f.js');
require('../../_chunks/dep-ae4040c5.js');
require('../../_chunks/dep-79476fcb.js');
require('../../_chunks/dep-e3edff33.js');
require('../../tooltip/utils/index.js');
require('../../_chunks/dep-5c79e9ef.js');
require('../../_chunks/dep-4eca9add.js');
require('../../_chunks/dep-9a54a293.js');
require('../../_chunks/dep-7870f4cb.js');
require('../../_chunks/dep-50d6b5ee.js');
require('../../_chunks/dep-692bdebe.js');
require('../../_chunks/dep-44195fb2.js');

var TImageViewerUtils = Vue.defineComponent({
  name: "TImageViewerUtils",
  props: {
    zIndex: Number,
    scale: Number,
    onRotate: Function,
    onZoomIn: Function,
    onZoomOut: Function,
    onMirror: Function,
    onReset: Function,
    onDownload: Function,
    currentImage: {
      type: Object,
      "default": function _default() {
        return {};
      }
    }
  },
  setup: function setup(props) {
    var classPrefix = index.usePrefixClass();
    var imageUrl = Vue.computed(function () {
      return props.currentImage.mainImage;
    });
    var _useImagePreviewUrl = index$1.useImagePreviewUrl(imageUrl),
      previewUrl = _useImagePreviewUrl.previewUrl;
    var _useConfig = configProvider_hooks_useConfig.useConfig("imageViewer"),
      globalConfig = _useConfig.globalConfig;
    return function () {
      return Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image-viewer__utils")
      }, [Vue.createVNode("div", {
        "class": "".concat(classPrefix.value, "-image-viewer__utils-content")
      }, [Vue.createVNode(tooltip_index.Tooltip, {
        "overlayClassName": "".concat(classPrefix.value, "-image-viewer__utils--tip"),
        "content": globalConfig.value.mirrorTipText,
        "destroyOnClose": true,
        "placement": "top",
        "showArrow": true,
        "theme": "default",
        "zIndex": props.zIndex
      }, {
        "default": function _default() {
          return [Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
            "onClick": props.onMirror,
            "icon": function icon() {
              return Vue.createVNode(tdesignIconsVueNext.MirrorIcon, {
                "size": "medium"
              }, null);
            }
          }, null)];
        }
      }), Vue.createVNode(tooltip_index.Tooltip, {
        "overlayClassName": "".concat(classPrefix.value, "-image-viewer__utils--tip"),
        "content": globalConfig.value.rotateTipText,
        "destroyOnClose": true,
        "placement": "top",
        "showArrow": true,
        "theme": "default",
        "zIndex": props.zIndex
      }, {
        "default": function _default() {
          return [Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
            "onClick": props.onRotate,
            "icon": function icon() {
              return Vue.createVNode(tdesignIconsVueNext.RotationIcon, {
                "size": "medium"
              }, null);
            }
          }, null)];
        }
      }), Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
        "icon": function icon() {
          return Vue.createVNode(tdesignIconsVueNext.ZoomOutIcon, {
            "size": "medium"
          }, null);
        },
        "onClick": props.onZoomOut
      }, null), Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
        "class": "".concat(classPrefix.value, "-image-viewer__utils-scale"),
        "label": "".concat(largeNumber.largeNumberToFixed(String(props.scale * 100)), "%")
      }, null), Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
        "icon": function icon() {
          return Vue.createVNode(tdesignIconsVueNext.ZoomInIcon, {
            "size": "medium"
          }, null);
        },
        "onClick": props.onZoomIn
      }, null), Vue.createVNode(tooltip_index.Tooltip, {
        "overlayClassName": "".concat(classPrefix.value, "-image-viewer__utils--tip"),
        "content": globalConfig.value.originalSizeTipText,
        "destroyOnClose": true,
        "placement": "top",
        "showArrow": true,
        "theme": "default",
        "zIndex": props.zIndex
      }, {
        "default": function _default() {
          return [Vue.createVNode("div", {
            "class": "".concat(classPrefix.value, "-image-viewer__modal-icon")
          }, [Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
            "icon": function icon() {
              return Vue.createVNode(tdesignIconsVueNext.ImageIcon, {
                "size": "medium"
              }, null);
            },
            "onClick": props.onReset
          }, null)])];
        }
      }), props.currentImage.download && Vue.createVNode(imageViewer_base_ImageModalIcon["default"], {
        "icon": function icon() {
          return Vue.createVNode(tdesignIconsVueNext.DownloadIcon, {
            "size": "medium"
          }, null);
        },
        "onClick": function onClick() {
          props.onDownload(previewUrl.value);
        }
      }, null)])]);
    };
  }
});

exports["default"] = TImageViewerUtils;
