// tdesign-vue-next/cjs/breadcrumb/breadcrumb-item.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
var breadcrumb_breadcrumbItemProps = require('./breadcrumb-item-props.js');
var tooltip_index = require('../tooltip/index.js');
var dom = require('../_chunks/dep-8d990e66.js');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
var index = require('../_chunks/dep-1862d466.js');
var index$1 = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
require('../tooltip/tooltip.js');
require('../tooltip/props.js');
require('../popup/props.js');
require('../popup/index.js');
require('../popup/popup.js');
require('@popperjs/core');
require('../_chunks/dep-38acad3d.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('../_chunks/dep-4a7162e7.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('../_chunks/dep-4ba41627.js');
require('../_chunks/dep-ce9c15f5.js');
require('../_chunks/dep-9f98bf9b.js');
require('../_chunks/dep-b2024592.js');
require('../_chunks/dep-2fb8d44f.js');
require('../_chunks/dep-80a41429.js');
require('../popup/container.js');
require('../_chunks/dep-e2122882.js');
require('../_chunks/dep-9bee89d3.js');
require('../_chunks/dep-027cd76f.js');
require('../_chunks/dep-ae4040c5.js');
require('../_chunks/dep-79476fcb.js');
require('../_chunks/dep-35e23c2b.js');
require('../_chunks/dep-e3edff33.js');
require('../_chunks/dep-a6551cc4.js');
require('../_chunks/dep-7d9abefd.js');
require('../_chunks/dep-91305690.js');
require('../_chunks/dep-0b92e695.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('../tooltip/utils/index.js');
require('../_chunks/dep-5c79e9ef.js');
require('../_chunks/dep-4eca9add.js');
require('../_chunks/dep-9a54a293.js');
require('../_chunks/dep-7870f4cb.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty__default["default"](e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !Vue.isVNode(s);
}
var localTBreadcrumbOrigin = {
  separator: "",
  theme: "light",
  maxItemWidth: void 0
};
var _BreadcrumbItem = Vue.defineComponent({
  name: "TBreadcrumbItem",
  inheritAttrs: false,
  props: _objectSpread(_objectSpread({}, breadcrumb_breadcrumbItemProps["default"]), {}, {
    isEllipsisItem: Boolean
  }),
  setup: function setup(props2, _ref) {
    var attrs = _ref.attrs;
    var renderContent = index.useContent();
    var renderTNodeJSX = index.useTNodeJSX();
    var breadcrumbText = Vue.ref();
    var localTBreadcrumb = Vue.inject("tBreadcrumb", localTBreadcrumbOrigin);
    var themeClassName = Vue.ref(localTBreadcrumb === null || localTBreadcrumb === void 0 ? void 0 : localTBreadcrumb.theme);
    var isCutOff = Vue.ref(false);
    var COMPONENT_NAME = index$1.usePrefixClass("breadcrumb__item");
    var separatorClass = index$1.usePrefixClass("breadcrumb__separator");
    var disableClass = index$1.usePrefixClass("is-disabled");
    var linkClass = index$1.usePrefixClass("link");
    var maxLengthClass = index$1.usePrefixClass("breadcrumb__inner");
    var textFlowClass = index$1.usePrefixClass("breadcrumb--text-overflow");
    var maxWithStyle = Vue.computed(function () {
      var maxItemWidth = localTBreadcrumb === null || localTBreadcrumb === void 0 ? void 0 : localTBreadcrumb.maxItemWidth;
      var maxWith = props2.maxWidth || maxItemWidth || "120";
      return {
        maxWidth: "".concat(maxWith, "px")
      };
    });
    var _getCurrentInstance = Vue.getCurrentInstance(),
      proxy = _getCurrentInstance.proxy;
    Vue.onMounted(function () {
      if (breadcrumbText.value) {
        isCutOff.value = dom.isTextEllipsis(breadcrumbText.value);
      }
    });
    Vue.onBeforeUpdate(function () {
      if (breadcrumbText.value) {
        isCutOff.value = dom.isTextEllipsis(breadcrumbText.value);
      }
    });
    var handleClick = function handleClick() {
      var router = props2.router || proxy.$router;
      if (props2.to && router) {
        props2.replace ? router.replace(props2.to) : router.push(props2.to);
      } else {
        window.location.href = props2.href;
      }
    };
    var bindEvent = function bindEvent(e) {
      if (!props2.disabled) {
        e.preventDefault();
        if (props2.target === "_blank") {
          props2.href ? window.open(props2.href) : window.open(props2.to);
        } else {
          handleClick();
        }
      }
    };
    return function () {
      var itemClass = [COMPONENT_NAME.value, themeClassName.value];
      var textClass = [textFlowClass.value];
      if (props2.disabled) {
        textClass.push(disableClass.value);
      }
      var listeners = {
        onClick: function onClick(e) {
          if (props2.disabled) {
            e.stopPropagation();
            return;
          }
        }
      };
      var _content = renderContent("default", "content");
      var textContent = Vue.createVNode("span", {
        "class": maxLengthClass.value,
        style: maxWithStyle.value
      }, [renderTNodeJSX("icon"), Vue.createVNode("span", {
        "ref": breadcrumbText,
        "class": "".concat(maxLengthClass.value, "-text")
      }, [_content])]);
      var itemContent = Vue.createVNode("span", _objectSpread({
        "class": textClass
      }, listeners), [textContent]);
      if ((props2.href || props2.to) && !props2.disabled) {
        textClass.push(linkClass.value);
        itemContent = Vue.createVNode("a", Vue.mergeProps({
          "class": textClass,
          "href": props2.href,
          "target": props2.target
        }, listeners, {
          "onClick": bindEvent
        }), [textContent]);
      }
      if (props2 !== null && props2 !== void 0 && props2.isEllipsisItem) {
        itemContent = Vue.createVNode("div", {
          "style": "display: flex"
        }, [_content]);
      }
      return Vue.createVNode("div", Vue.mergeProps({
        "class": itemClass
      }, attrs, {
        "onClick": props2.disabled ? void 0 : props2.onClick
      }), [isCutOff.value ? Vue.createVNode(tooltip_index.Tooltip, {
        "content": function content() {
          return _content;
        }
      }, _isSlot(itemContent) ? itemContent : {
        "default": function _default() {
          return [itemContent];
        }
      }) : itemContent, Vue.createVNode("span", {
        "class": separatorClass.value,
        "style": {
          textOverflow: isCutOff.value ? "ellipsis" : "clip"
        }
      }, [localTBreadcrumb.separator])]);
    };
  }
});

exports["default"] = _BreadcrumbItem;
