// tdesign-vue-next/cjs/common-components/fake-arrow.js
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var _defineProperty = require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/helpers/toConsumableArray');
require('@babel/runtime/helpers/typeof');
require('../_chunks/dep-35e23c2b.js');
var index = require('../_chunks/dep-0238ed9a.js');
require('@babel/runtime/helpers/slicedToArray');
require('../_chunks/dep-0989c3be.js');
var helper = require('../_chunks/dep-2aa3d3cf.js');
require('../_chunks/dep-4a7162e7.js');
require('../config-provider/hooks/useConfig.js');
require('../_chunks/dep-be1af85d.js');
require('../_chunks/dep-db9c85b1.js');
require('dayjs');
require('../_chunks/dep-82fe2026.js');
require('../_chunks/dep-33b46a52.js');
require('../_chunks/dep-930a2de5.js');
require('../_chunks/dep-2dcf9237.js');
require('../_chunks/dep-22dc294c.js');
require('../_chunks/dep-bf76dead.js');
require('../_chunks/dep-c3bbd06c.js');
require('../_chunks/dep-6d87f74d.js');
require('../_chunks/dep-4cb26289.js');
require('../_chunks/dep-281f7eb2.js');
require('../_chunks/dep-f8f7b1af.js');
require('../_chunks/dep-66d3f30f.js');
require('../_chunks/dep-e1d22111.js');
require('../_chunks/dep-754c0523.js');
require('../_chunks/dep-7076a08a.js');
require('../_chunks/dep-631750cd.js');
require('../_chunks/dep-2fc884aa.js');
require('../_chunks/dep-75d6da05.js');
require('../_chunks/dep-274369eb.js');
require('../_chunks/dep-df6e14a0.js');
require('@babel/runtime/helpers/createClass');
require('@babel/runtime/helpers/classCallCheck');
require('@babel/runtime/helpers/objectWithoutProperties');
require('../_chunks/dep-3d29e358.js');
require('../_chunks/dep-6212811d.js');
require('../_chunks/dep-44195fb2.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var _defineProperty__default = /*#__PURE__*/_interopDefaultLegacy(_defineProperty);

var FakeArrow = Vue.defineComponent({
  name: "TFakeArrow",
  props: {
    isActive: {
      type: Boolean
    },
    overlayClassName: {
      type: [String, Object, Array]
    },
    overlayStyle: {
      type: [Object, String]
    }
  },
  setup: function setup(props) {
    var COMPONENT_NAME = index.usePrefixClass("fake-arrow");
    var classes = Vue.computed(function () {
      return [COMPONENT_NAME.value, _defineProperty__default["default"](_defineProperty__default["default"]({}, "".concat(COMPONENT_NAME.value, "--transform"), helper.isSafari()), "".concat(COMPONENT_NAME.value, "--active"), props.isActive), props.overlayClassName];
    });
    return function () {
      return Vue.createVNode("svg", {
        "class": classes.value,
        "width": "16",
        "height": "16",
        "viewBox": "0 0 16 16",
        "fill": "none",
        "xmlns": "http://www.w3.org/2000/svg",
        "style": props.overlayStyle
      }, [Vue.createVNode("path", {
        "d": "M3.75 5.7998L7.99274 10.0425L12.2361 5.79921",
        "stroke": "black",
        "stroke-opacity": "0.9",
        "stroke-width": "1.3"
      }, null)]);
    };
  }
});

exports["default"] = FakeArrow;
