// tdesign-vue-next/es/color-picker/components/trigger/index.mjs
/**
 * tdesign v1.19.2
 * (c) 2026 tdesign
 * @license MIT
 */

import { defineComponent, ref, watch, createVNode, mergeProps } from 'vue';
import { _ as _defineProperty } from '../../../_chunks/dep-ba214d75.mjs';
import { Input } from '../../../input/index.mjs';
import '../../utils/index.mjs';
import { useBaseClassName } from '../../hooks/index.mjs';
import { c as useCommonClassName, p as useEventForward } from '../../../_chunks/dep-fbeb2963.mjs';
import { g as getColorObject, C as Color } from '../../../_chunks/dep-1a80bfd8.mjs';
import '../../../_chunks/dep-e1c62679.mjs';
import '../../../input/input.mjs';
import '../../../_chunks/dep-c231a5b4.mjs';
import 'tdesign-icons-vue-next';
import '../../../input/props.mjs';
import '../../../input/hooks/index.mjs';
import '../../../input/hooks/useInput.mjs';
import '../../../_chunks/dep-f558ebb8.mjs';
import '../../../_chunks/dep-b0b422fe.mjs';
import '../../../form/constants/index.mjs';
import 'lodash-es';
import '../../../_chunks/dep-47c3cae9.mjs';
import '../../../_chunks/dep-feed89df.mjs';
import '../../../_chunks/dep-5b510fcd.mjs';
import '../../../config-provider/hooks/useConfig.mjs';
import '../../../config-provider/utils/context.mjs';
import '../../../_chunks/dep-e0ab6d55.mjs';
import '../../../_chunks/dep-75f265c4.mjs';
import '../../../_chunks/dep-67ffcfeb.mjs';
import '../../../_chunks/dep-edab0e6a.mjs';
import '../../../_chunks/dep-76dddedf.mjs';
import '../../../_chunks/dep-d92220c4.mjs';
import '../../../_chunks/dep-466a8f31.mjs';
import '../../../_chunks/dep-e66c0337.mjs';
import '../../../input/hooks/useLengthLimit.mjs';
import '../../../_chunks/dep-739ea4cd.mjs';
import '../../../_chunks/dep-58979704.mjs';
import '../../../input/hooks/useInputEventHandler.mjs';
import '../../../input/hooks/useInputWidth.mjs';
import '../../../input/input-group.mjs';
import '../../../input/input-group-props.mjs';
import './style/css.mjs';
import '../../../_chunks/dep-063c3304.mjs';
import '../../../_chunks/dep-40798bc5.mjs';

var DefaultTrigger = defineComponent({
  name: "DefaultTrigger",
  inheritAttrs: false,
  props: {
    color: {
      type: String,
      "default": ""
    },
    disabled: {
      type: Boolean,
      "default": false
    },
    borderless: {
      type: Boolean,
      "default": false
    },
    clearable: {
      type: Boolean,
      "default": false
    },
    inputProps: {
      type: Object,
      "default": function _default() {
        return {
          autoWidth: true
        };
      }
    },
    onTriggerChange: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    },
    onTriggerClear: {
      type: Function,
      "default": function _default() {
        return function () {};
      }
    },
    size: {
      type: String,
      "default": "medium"
    }
  },
  setup: function setup(props) {
    var baseClassName = useBaseClassName();
    var value = ref(props.color);
    var _useCommonClassName = useCommonClassName(),
      sizeClassNames = _useCommonClassName.SIZE;
    watch(function () {
      return [props.color];
    }, function () {
      return value.value = props.color;
    });
    var handleChange = function handleChange(input) {
      if (input !== props.color) {
        props.onTriggerChange(value.value, {
          color: getColorObject(new Color(input)),
          trigger: "input"
        });
      }
    };
    var handleClear = function handleClear(context) {
      var _props$onTriggerClear;
      return (_props$onTriggerClear = props.onTriggerClear) === null || _props$onTriggerClear === void 0 ? void 0 : _props$onTriggerClear.call(props, context);
    };
    var inputEvents = useEventForward(props.inputProps, {
      onBlur: handleChange,
      onChange: handleChange,
      onClear: handleClear
    });
    return function () {
      var inputSlots = {
        label: function label() {
          return createVNode("div", {
            "class": ["".concat(baseClassName.value, "__trigger--default__color"), "".concat(baseClassName.value, "--bg-alpha")]
          }, [createVNode("span", {
            "class": ["color-inner", _defineProperty({}, sizeClassNames.value[props.size], props.size !== "medium")],
            "style": {
              background: value.value
            }
          }, null)]);
        }
      };
      return createVNode(Input, mergeProps({
        "borderless": props.borderless,
        "clearable": props.clearable,
        "size": props.size,
        "modelValue": value.value,
        "onUpdate:modelValue": function onUpdateModelValue($event) {
          return value.value = $event;
        },
        "disabled": props.disabled
      }, inputEvents.value), inputSlots);
    };
  }
});

export { DefaultTrigger as default };
